---
title: "EVITA Manuscript - Final Verified"
output: html_document
date: "2025-04-25"
---

```{r setup, include=FALSE}
#Prevent messages, warnings, and code from showing up in the report to keep things tidy. 
#All code chunks have been run independently to verify model fits
knitr::opts_chunk$set(echo = FALSE,warning = FALSE,message = FALSE)

#Turn off scientific notation
options(scipen=999)

#Helper function for taking the last n characters of a string
substrRight <- function(x, n){
  substr(x, nchar(x)-n+1, nchar(x))
}

pacman::p_load(readxl,ggplot2,table1,dplyr,ggprism,MASS,emmeans,ggbeeswarm,ggsci,
               cowplot, ggmosaic,robustlmm,ggpubr,survival,survminer,patchwork,
               forestmodel,tidycmprsk,cmprsk,ggsurvfit,ggfortify,ggh4x,corrplot,quantreg,reshape2)

#Custom function for table 1 output
my.render.cont <- function(x) {
    with(stats.default(x), 
         c("",
           
          "Mean (SD)" = sprintf("%s (%s)",
                                round_pad(MEAN, 2),
                                round_pad(SD, 2)),
         
          "Median (Q1, Q3)" = sprintf("%s (%s, %s)",
                                       round_pad(MEDIAN, 2), 
                                       round_pad(Q1, 2), 
                                       round_pad(Q3, 2)))
    )
}

```

# Section {.tabset}
## Bookeeping tables during data processing

```{r}

#Read in the main data
df = read_excel(unzip("Vitc_Data.zip","EVI2_data_VAI.xlsx"))

# These patients had to be manually changed per Stine, email sent May 2024 and another Feb 2025
df[df$ID == 64,]$Diagnosis_simple_baseline = "MDS"
df[df$ID == 72,]$Diagnosis_simple_baseline = "NA"
df[df$ID == 96,]$Diagnosis_simple_baseline = "NA"

#Convert dummy coding to named treatment arms
df$Treatment_arm = ifelse(df$Treatment_arm == 0, "Vitamin C","Placebo")

#Make an age in years variable
df$Age = (df$Age_inclusion_days/365.250 )

#Dummy code diagnosis into CCUS or non CCUS
df$CCUS = ifelse(df$Diagnosis_simple_baseline =="CCUS","CCUS","NonCCUS")
df$CCUS = as.numeric(factor(df$CCUS,levels=c("NonCCUS","CCUS")))

#This is to make the IDs match across data.frames, all we're doing is removing hyphens in character IDs and removing leading 0s.
main.before.ID = df$ID
df$ID = sapply(strsplit(df$ID,"-"), function(x){ifelse(length(x) >1, paste0(x[[1]],as.numeric(x[[2]])),x)})

knitr::kable(data.frame(main.before.ID,df$ID),caption = "ID before and after main file",col.names = c("Before","After"))

#New classification per Stine
df$Diagnosis_simple_combo = case_when(df$Diagnosis_simple_baseline == "CCUS" ~ "CCUS",
                                      df$Diagnosis_simple_baseline == "MDS" ~ "MDS",
                                      df$Diagnosis_simple_baseline == "CMML" | df$Diagnosis_simple_baseline == "MDS/MPN"~ "CMML_MDS_MPN",
                                      TRUE~NA)


VAFS = read_excel("EVI2_curated_variants_07052025_studyIDs.xlsx")

#These check out, USC-60 never received treatment and 14 and 96 do not have any VAFS
knitr::kable(df$ID[which(! df$ID %in%  VAFS$ID)],caption = "IDs in main file but not in VAFS")
vdates = read_excel("EVI2_sample dates_DNAseq.xlsx",skip=1)


#Some data cleaning to give IDs consistent formatting across data frames and create a time variable from the ID column
VAFS.before.ID = VAFS$Original_ID

## Make factor so we can grab baseline as 1 and EOT as 2
VAFS$Time = as.numeric(factor(VAFS$Visit,levels=c("Baseline","EOT")))

VAFS$ID = sapply(strsplit(VAFS$Original_ID,"-"), function(x){ifelse(length(x) >1, paste0(x[[1]],as.numeric(x[[2]])),x)})

knitr::kable(data.frame(VAFS.before.ID,VAFS$ID),caption = "ID before and after VAFS file",col.names = c("Before","After"))

knitr::kable(VAFS$ID[which(! VAFS$ID %in%  df$ID)],caption = "IDs in VAFS file but not in main")

vdates$Dato = as.Date(vdates$Dato)
vdates$ID = zoo::na.locf(vdates$ID)
vdates$ID = sapply(strsplit(vdates$ID,"-"), function(x){ifelse(length(x) >1, paste0(x[[1]],as.numeric(x[[2]])),x)})

vdates$`Study visit`[vdates$`Study visit` == "baseline"] = "Baseline"
colnames(vdates)[2] = "Visit"
vdates=dcast(vdates, ID ~ Visit)

vafs.merged = full_join(subset(VAFS,select=c(ID,Treatment_Group,Visit,Hugo_Symbol,cDNA_Change,AF)),vdates,by ='ID' )
vafs.merged$Time = (vafs.merged$EOT-vafs.merged$Baseline)/365.25
vafs.merged = subset(vafs.merged, select = - c(Baseline,EOT))

vafs.base = vafs.merged[vafs.merged$Visit=="Baseline",]
colnames(vafs.base) = c("ID","Treatment","visit_1","Hugo_Symbol","cDNA_Change","AF_1","Time")

vafs.eot =  vafs.merged[vafs.merged$Visit=="EOT",]
colnames(vafs.eot) = c("ID","Treatment","visit_FU","Hugo_Symbol","cDNA_Change","AF_FU","Time")

vafs.wide = full_join(vafs.base,vafs.eot)
vafs.wide = vafs.wide[!is.na(vafs.wide$ID),]
# These ids do not have a follow-up time and should be excluded
IDs.ex = unique(vafs.wide$ID[which(is.na(vafs.wide$Time) )])
# 3  6 17 22 36 46 53 54 check with Stine that these should be removed

knitr::kable(vafs.wide[vafs.wide$ID %in% IDs.ex,],caption = "patients being removed from VAF analysis due to no follow-up visit")

```

## Exposure-adjusted incidence rates of adverse and serious adverse events

Code from Stine

```{r}
eviy = read_excel("EVI2_safety_treatment exposure dates.xlsx")

# Calculate exposure time in days and months
eviy$exposure_days <- as.numeric(eviy$EOT - eviy$Baseline + 1)
eviy$exposure_months <- eviy$exposure_days / 30.44

# Total patient-months in each group
total_months_placebo <- sum(eviy$exposure_months[eviy$Treatment_arm == 1], na.rm = TRUE)
total_months_vitC <- sum(eviy$exposure_months[eviy$Treatment_arm == 0], na.rm = TRUE)

n_SAEs_placebo <- 66
n_SAEs_vitC <- 39
n_AEs_placebo <- 82
n_AEs_vitC <- 54

# SAE incidence rate
sae_rate_placebo <- (n_SAEs_placebo / total_months_placebo) * 100
sae_rate_vitC <- (n_SAEs_vitC / total_months_vitC) * 100

# AE incidence rate (any grade)
ae_rate_placebo <- (n_AEs_placebo / total_months_placebo) * 100
ae_rate_vitC <- (n_AEs_vitC / total_months_vitC) * 100

cat("SAE incidence per 100 patient-months:\n")
cat("Placebo:", round(sae_rate_placebo, 1), "\n")
cat("Vitamin C:", round(sae_rate_vitC, 1), "\n")

cat("\nAE incidence per 100 patient-months:\n")
cat("Placebo:", round(ae_rate_placebo, 1), "\n")
cat("Vitamin C:", round(ae_rate_vitC, 1), "\n")


```

## Median cumulative Clonal Growth Rates {.tabset}
### Bookkeeping
Growth rate, ð›¼ = log (ð‘‰ð´ð¹ ð‘“ð‘œð‘™ð‘™ð‘œð‘¤-ð‘¢ð‘/ð‘‰ð´ð¹ ð‘ð‘Žð‘ ð‘’ð‘™ð‘–ð‘›ð‘’ )/ð‘¡ð‘–ð‘šð‘’
Patients 14 and 96 have no VAF info because they did not have nay VAFS, they are not included in this current version. The below table is all variants, patients can have multiple VAFs.


```{r}
# Two patients have all their follow-up time points, but no VAFs Patients 14 and 96

vafs.wide = vafs.wide[! vafs.wide$ID %in% IDs.ex,]
vafs.wide$Emergent = rep(0)

vafs.wide$Emergent[is.na(vafs.wide$AF_1 ) ] = 1

vafs.wide$AF_1[is.na(vafs.wide$AF_1)] = 0.01
vafs.wide$AF_FU[is.na(vafs.wide$AF_FU)] = 0.01

vafs.wide$GR = log(as.numeric(vafs.wide$AF_FU)/as.numeric(vafs.wide$AF_1))/ as.numeric(vafs.wide$Time)

agg.med = aggregate(GR~ID+Treatment,vafs.wide,median)

# Summary table
table1::table1(~ GR | Treatment,data=vafs.wide)

```

### Median GR {.tabset}
#### All patients

Primary outcome

```{r, fig.width=6,fig.height=6}
# Summary table
table1::table1(~ GR | Treatment,data=agg.med,render.continuous=my.render.cont)

fit.median = rq(GR~Treatment,data=agg.med)

# coin::median_test(GR~as.factor(Treatment),data=agg.med)

knitr::kable(confint(emmeans(fit.median,pairwise~Treatment)),caption = "emmeans and contrast")

#Store p-value for plot annotation
p =  round(joint_tests(fit.median)[5],2)
if(p>0.99){p = 0.99}
if(p <0.0001){p=0.0001}
#Output joint_tests table
knitr::kable(joint_tests(fit.median),caption="p-value on difference")
#This grabs the means as estimated by rlm for plotting
# agg.mu = summary(emmeans(fit.median,pairwise~Treatment)$emmeans)
# colnames(agg.mu)[2] = "GR"


CI = signif(round(confint(emmeans(fit.median,pairwise~Treatment))$contrasts[,5:6],4),2)

label_med_vaf = paste0("Median \U0394 ",signif(round(summary(fit.median)$coeff[2,1] ,4),2),", ",
                   "p =  ",p,"\n",
                   "95% CI ",-1*CI[2],", ",-1*CI[1])

clonal_trajectory_figure.primary_withCI = ggplot(agg.med,aes(x = Treatment,y = GR,color=Treatment)) + 
  geom_quasirandom() + 
  geom_boxplot(fill=NA,outliers = F) + 
  theme_prism(base_size = 12) + 
  scale_color_manual(values=c("black",pal_nejm()(4)[c(3)])) +
  ggtitle("Median Clonal Growth Rates") + 
  ylab(expression(bold("Median"~log(frac("VAF at end of treatment","VAF at baseline"))*"/year"))) +
  coord_cartesian(ylim=c(-1.5,1.9))+xlab("Treatment") +
  # geom_point(data=agg.mu,aes(x = Treatment,y = GR,color=Treatment),size=4,shape=2)+
  theme(legend.position = "none") +
  annotate(geom="line",x= c(1,2),y =c(1.65,1.65) ) +
  annotate(geom="text",x = 1.5, y = 1.9, label = label_med_vaf,fontface="italic")

clonal_trajectory_figure.primary_withCI

ggsave("clonal_trajectory_figure_primary_with_CI.pdf",width=6,height=6,device=cairo_pdf)

```


### Subgroups {.tabset}
#### Bookkeeping

```{r}

knitr::kable(table(df$Diagnosis_simple_combo))

#Merge the median GRs with clinical information
vafs.subgroup = merge(agg.med,subset(df,select=c(ID,Treatment_arm,VitC_PB1,Diagnosis_simple_combo)))

#Make the median vitamin C variable
vafs.subgroup$med_vitc = ifelse(vafs.subgroup$VitC_PB1 >= median(df$VitC_PB1),"above","below")

vafs.subgroup$vitc_ccus = vafs.subgroup$Diagnosis_simple_combo

#CMML is part of MDS/MPN
vafs.subgroup$vitc_ccus[vafs.subgroup$vitc_ccus == "CMML_MDS_MPN"] <- "MDS/MPN"
knitr::kable(table(vafs.subgroup$vitc_ccus),caption = "Counts per diagnosis main file")

#If a patient has any of these mutations at baseline they are positive for mutation
mut_ids = unique(VAFS$ID[VAFS$Hugo_Symbol %in% c("TET2","IDH1","IDH2") & VAFS$Visit=="Baseline"])
#Create a temp file for accounting purposes
acct.tmp = subset(VAFS[VAFS$ID %in% mut_ids & VAFS$Time == 1,],select=c(ID,Time,Hugo_Symbol))
acct.tmp=acct.tmp[order(acct.tmp$ID),]
knitr::kable(acct.tmp,caption = "Patients with baseline TET2, IDH1, or IDH2 mutations")
acct.tmp=NULL

#Dummy variable for TET2 IDH1/2 or no
vafs.subgroup$TET2 = ifelse(vafs.subgroup$ID %in% mut_ids,"TET2Mut", "TET2neg")


#Function to run the subgroup analyses
vafs.subgroup.fxn = function(subgroup){
  #Create a temporary subgroup data frame
  tmp.subgp = na.omit(subset(vafs.subgroup,select=c("GR","Treatment_arm",subgroup,"ID")))

  #Generic naming convention for the subgroup variable
  colnames(tmp.subgp)[3] = "grp"

  #Conditionals to factor and relabel for manuscript
  if(subgroup == "TET2"){
    tmp.subgp$grp =factor(tmp.subgp$grp,levels=c("TET2Mut", "TET2neg"),labels=c("TET2 or IDH1/2 mutation","No TET2 or IDH1/2 mutation"))
  }

  if(subgroup == "med_vitc"){
    tmp.subgp$grp = factor(tmp.subgp$grp,levels=c("below","above"),labels=c("Below median","Above median"))
  }

  if(subgroup == "vitc_ccus"){

    tmp.subgp$grp = factor(tmp.subgp$grp,
                      levels = c("CCUS","MDS","MDS/MPN"),
                      labels= c("CCUS","MDS","MDS/MPN" ))
    tmp.subgp=tmp.subgp[!is.na(tmp.subgp$grp),]

  }

  # Summary table
  table1::table1(~ GR | grp ,data=tmp.subgp)

  # robust linear model for continuity with main analysis
  fit = rq(GR~Treatment_arm*grp,data=tmp.subgp)

  sig.df = data.frame(confint(emmeans(fit,pairwise~Treatment_arm|grp)$contrasts))

  #We're flipping the reference order here so that the plot annotations ar emore natural to read
  sig.df$CI = paste0("Median \U0394 ",-1*signif(sig.df$estimate,2),"\n95% CI ",
                     -1*signif(sig.df$upper.CL,2),", ",
                     -1*signif(sig.df$lower.CL,2))

  #This sets the x-values for the significance bars
  sig.df.line = rbind(sig.df,sig.df)
  if(subgroup == "vitc_ccus"){
    sig.df.line$x = c(1,1,1,2,2,2)
  } else {
    sig.df.line$x = c(1,1,2,2)
  }

  #set the initial height of the lines and text annotation
  sig.df$y = max(tmp.subgp$GR,na.rm=T)
  sig.df.line$y = max(tmp.subgp$GR,na.rm=T)

  agg.mu = data.frame(summary(emmeans(fit,pairwise~Treatment_arm|grp)$emmeans))
  colnames(agg.mu)[3] = "GR"

  # Box plot or raw cumualtive GRs adding means as triangles and p-value with 2 sig-figs
  print(ggplot(tmp.subgp,aes(x = Treatment_arm,y = GR,color=Treatment_arm)) + geom_quasirandom() +
    geom_boxplot(fill=NA,outliers = F) + theme_prism() + scale_color_manual(values=c("black",pal_nejm()(4)[c(3)])) +ggtitle("Median clonal growth rates") + ylab(expression("Median"~log(frac("VAF at end of treatment","VAF at baseline"))*"/year")) + coord_cartesian(ylim=c(-1.5,1.9))+xlab("Treatment") + geom_point(data=agg.mu,aes(x = Treatment_arm,y = GR,color=Treatment_arm),size=4,shape=2)+theme(legend.position = "none",strip.background = element_blank()) +
    facet_wrap(~grp) + geom_line(data=sig.df.line,aes(x=x,y =1.75 ),color="black") + geom_text(data=sig.df,aes(x = 1.5,y=1.9,label=CI),color="black"))

}

```


#### TET2 mutation status

```{r,fig.height=6,fig.width=10}
vafs.subgroup.fxn("TET2")
```

#### Median baseline vitamin C mutation status

```{r,fig.height=6,fig.width=10}
vafs.subgroup.fxn("med_vitc")
```

#### Baseline diagnosis

```{r,fig.height=6,fig.width=10}
vafs.subgroup.fxn("vitc_ccus")
```

### Individual mutations {.tabset}

```{r}

#Counts of each mutation type
knitr::kable(table(vafs.wide$Hugo_Symbol)[order(table(vafs.wide$Hugo_Symbol))],caption="Counts of each mutation")

# Function to plot GR of specific mutations with p-values
plot.spec.mut = function(muts) {
  vafs.mut = merge(vafs.wide[vafs.wide$Hugo_Symbol %in% muts,],subset(df,select=c(ID,Treatment_arm)))
  
  indv.genes = NULL
  for(i in muts){
    agg.vafs.mut = aggregate(GR~ID+Treatment, vafs.mut[vafs.mut$Hugo_Symbol == i,],function(x) median(x,na.rm=T))
    fit = rq(GR~Treatment,data=agg.vafs.mut)
    indv.genes = rbind(indv.genes,data.frame(mut = i,joint_tests(fit)))
  }
  indv.genes$FDR = p.adjust(indv.genes$p,method="BH")
  
  for(i in muts){
    agg.vafs.mut = aggregate(GR~ID+Treatment, vafs.mut[vafs.mut$Hugo_Symbol == i,],function(x) median(x,na.rm=T))  
    y = max(agg.vafs.mut$GR,na.rm=T)
    FDR = signif(indv.genes$FDR[indv.genes$mut == i],2)
    if(FDR > 0.99){FDR_label = "FDR p > 0.99"
    } else {
        FDR_label = paste0("FDR p = ",FDR)
    } 
    print(ggplot(agg.vafs.mut,aes(x = Treatment,y = GR,color=Treatment)) + geom_quasirandom() + 
      geom_boxplot(fill=NA,outliers = F) + theme_prism() + scale_color_manual(values=c("black",pal_nejm()(4)[c(3)])) + ggtitle(i)+xlab("Treatment") + annotate(geom="line",x= c(1,2),y =c(y*1.05,y*1.05) ) + annotate(geom="text",x = 1.5, y = y*1.13, label = FDR_label,fontface="italic")+ylab("Growth rate")+theme(legend.position = "none")+xlab("Treatment"))
  }
}

plot.spec.mut(c("TET2","DNMT3A","SRSF2","ZRSR2","ASXL1" ))

```


## Plasma Vitamin C {.tabset}
### Manuscript Figure

**Patient plasma vitamin C level.** A) Composition of all patients, placebo controls, and vitamin C treated individuals with: vitamin C deficiency, inadequate vitamin C, or adequate vitamin C at inclusion; distributions are relatively similar between the two treatment arms. B) Plasma vitamin C level at each timepoint by treatment arm; p-values were estimated using robust linear mixed-effects model with a random intercept for each individual. C) LOESS smoothed vitamin C level over time with standard error ribbons. Individuals with *>200 \mumol/L* vitamin C level were censored in C) to better characterize dispersion among trial participants, all individual measures are shown in panel B).

```{r, fig.height = 14, fig.width=9}

## Specific subset
df.sp =  subset(df,select = c(ID,Treatment_arm,VitC_PB1,VitC_PB2,VitC_PB3,VitC_PB4,VitC_PB5,Age,Sex,Hb1,CCUS))
df.sp =  reshape2::melt(df.sp,id.vars=c("ID","Treatment_arm","Age","Sex","Hb1","CCUS"))
df.sp$variable = factor(df.sp$variable, levels = c("VitC_PB1","VitC_PB2","VitC_PB3","VitC_PB4","VitC_PB5"))

#Turning these variables into continuous times, grouped by ~months not exact
df.sp$variable = case_when(df.sp$variable == "VitC_PB1" ~ 0,
                               df.sp$variable == "VitC_PB2" ~ 365.25/4,
                               df.sp$variable == "VitC_PB3" ~ 2*365.25/4,
                               df.sp$variable == "VitC_PB4" ~ 3*365.25/4,
                               df.sp$variable == "VitC_PB5" ~ 365.25)

#This is to plot the 'Deficient' and 'inadequate' labels
tmp1 = data.frame(variable =12.25/12*365.25,value2 = 11.5,Treatment_arm = "Vitamin C" )
tmp2 = data.frame(variable =12.25/12*365.25,value2 = 36.5,Treatment_arm = "Vitamin C" )
tmp3 = data.frame(variable =12.25/12*365.25,value2 = 61.5,Treatment_arm = "Vitamin C" )

#This to plot vertical arrows for the labels
def = data.frame(variable = c(12.1*365.25/12,12.1*365.25/12),value2=c(3,20),Treatment_arm="Vitamin C")
inad = data.frame(variable = c(12.1*365.25/12,12.1*365.25/12),value2=c(26,47),Treatment_arm="Vitamin C")
ad = data.frame(variable = c(12.1*365.25/12,12.1*365.25/12),value2=c(53,200),Treatment_arm="Vitamin C")

#This is for plotting purposes only and is noted in the figure legend
df.sp$value2 = df.sp$value
df.sp$value2[df.sp$value>200] = 200
df.sp$value2[df.sp$value2==200 & df.sp$variable == 182.6250] = NA

# Straightforward ggplot code with 
p2.smoothed.vitc = 
  ggplot(df.sp[!is.na(df.sp$value),],aes(x = 12*as.numeric(variable)/365.25,y=value2,color=Treatment_arm,fill=Treatment_arm))  +    geom_hline(aes(yintercept =50),color = colorRampPalette(c("steelblue3","black"))(5)[1],linewidth=1,linetype="dashed") +
 geom_hline(aes(yintercept =23),color=colorRampPalette(c("red3","black"))(5)[1],linewidth=1,linetype="dotted") +
theme_prism() + geom_line(aes(group =  ID),alpha=0.15 ) + stat_smooth(se=T,linewidth=1.5,span=1) + scale_color_manual(values =c("black",pal_nejm()(4)[3])) + scale_fill_manual(values =c("black",pal_nejm()(4)[3])) +scale_y_continuous(expand=c(0,0))+ scale_x_continuous(breaks=c(0,3,6,9,12),expand=c(0,0)) +xlab("Months from randomisation")+ylab("Vitamin C concentration (\U03BCmol/L)") +
  geom_text(data=tmp1,color =colorRampPalette(c("red3","black"))(5)[1],label="Deficient",hjust=0) +
  geom_text(data=tmp2,color =colorRampPalette(c("steelblue3","black"))(5)[1],label="Inadequate",hjust=0) +
  geom_text(data=tmp3,color =colorRampPalette(c("palegreen3","black"))(5)[1],label="Adequate",hjust=0) +
  # geom_line(data=def,color="red3",linewidth=1,linetype="solid",alpha=0.5) +
  # geom_line(data=inad,color="steelblue3",linewidth=1,linetype="solid",alpha=0.5) +
  # geom_line(data=ad,color="palegreen3",linewidth=1,linetype="solid",alpha=0.5) +
  guides(color=guide_legend(title=""),fill=guide_legend(title=""))+coord_cartesian(clip="off")+facet_wrap(~Treatment_arm)+ theme(panel.spacing = unit(2, "lines"))+
  theme(legend.position = "top", plot.margin = unit(c(1,2,1,1), "cm"))


#Years into months
df.sp$Time = factor(paste0(12*df.sp$variable/365.25,""),levels=seq(0,12,3))

##robust lmer to get treatment difference per timepoint
fit = rlmer(log2(value)~Time*Treatment_arm+(1|ID),data=df.sp)
res = data.frame(summary(emmeans(fit,pairwise~Treatment_arm|Time)$contrasts))
res = merge(res,data.frame(summary(confint(emmeans(fit,pairwise~Treatment_arm|Time)$contrasts))))

##2 sig figs on p-values out to 0.0001
res$Label = ifelse(res$p.value <0.0001,"p < 0.0001",paste0("p =  ",signif(round(res$p.value,4),2)))

res$Label_alt = paste0("Mean \U0394 ",signif(round(-1*res$estimate,4),2),", ",
                    res$Label,"\n",
                   "95% CI ",signif(round(-1*res$asymp.UCL,4),2),", ",signif(round(-1*res$asymp.LCL,4),2))

##Similar to above but boxplots annotated with p-vals which are Tukey adjusted
p2.boxplot.vitc= ggplot()  +   geom_hline(aes(yintercept =23),color="red3",linewidth=1,linetype="dotted") +
    geom_hline(aes(yintercept =50),color = "steelblue3",linewidth=1,linetype="dashed") +
theme_prism()+ geom_boxplot(data=df.sp,aes(x = Time,y=value,color=Treatment_arm),outlier.size=-1) + geom_quasirandom(data=df.sp,aes(x = Time,y=value,color=Treatment_arm),alpha=0.25,dodge.width = .75)  + scale_color_manual(values =c("black",pal_nejm()(4)[3]))  +xlab("Months from randomisation")+ylab("Vitamin C concentration (\U03BCmol/L)")+
  annotate(geom="text",x = 5.9,y = 11.5,color = "red3",label="Deficient",hjust=0) +
  annotate(geom="text",x = 5.9,y = 36.5,color = "steelblue3",label="Inadequate",hjust=0) +
  annotate(geom="text",x = 5.9,y = 61.5,color = "palegreen3",label="Adequate",hjust=0) +
  guides(color=guide_legend(title=""),fill=guide_legend(title=""))+coord_cartesian(clip="off")+scale_y_continuous(breaks=seq(0,300,50),expand=c(0,1)) + geom_text(data=res,aes(label=Label,x=Time,y=260),size=2.75,color="black",fontface="italic")+geom_segment(aes(x = 0.75,xend=1.25,y=240,yend=240))+
  geom_segment(aes(x = 1.75,xend=2.25,y=240,yend=240))+
  geom_segment(aes(x = 2.75,xend=3.25,y=240,yend=240))+
  geom_segment(aes(x = 3.75,xend=4.25,y=240,yend=240))+
  geom_segment(aes(x = 4.75,xend=5.25,y=240,yend=240))+
  theme(legend.position = "top", plot.margin = unit(c(1,2,1,1), "cm"))
 


#  geom_line(data=NULL,aes(x = c(5.875,5.875),y=c(3,20)),color="red3",linewidth=1,linetype="solid",alpha=0.5) +
 #  geom_line(data=NULL,aes(x = c(5.875,5.875),y=c(26,47)),color="steelblue3",linewidth=1,linetype="solid",alpha=0.5) +
 # geom_line(data=NULL,aes(x = c(5.875,5.875),y=c(53,300)),color="palegreen3",linewidth=1,linetype="solid",alpha=0.5)
 #  
df.sp$Adequacy = case_when( df.sp$value < 23 ~ "Deficient",
                            df.sp$value < 50 ~ "Inadequate",
                            TRUE ~ "Sufficient")

#Proportions data frame
pt = data.frame(prop.table(table(df.sp$Adequacy,df.sp$Time),2))
pt.c = data.frame(prop.table(table(df.sp$Adequacy[df.sp$Treatment_arm=="Vitamin C"],df.sp$Time[df.sp$Treatment_arm=="Vitamin C"]),2))
pt.p =  data.frame(prop.table(table(df.sp$Adequacy[df.sp$Treatment_arm=="Placebo"],df.sp$Time[df.sp$Treatment_arm=="Placebo"]),2))

pt$Group =  "All patients (n = 109)"
pt.c$Group =  "Vitamin C (n = 55)"
pt.p$Group =  "Placebo (n = 54)"

pt = rbind(pt,pt.c,pt.p)
pt$Var1 = as.character(pt$Var1)
pt$Var1[pt$Var1 == "Deficient"] = "Deficient (< 23 \U03BCmol/L)"
pt$Var1[pt$Var1 == "Inadequate"] = "Inadequate (23-49 \U03BCmol/L)"
pt$Var1[pt$Var1 == "Sufficient"] = "Adequate (\u2265 50 \U03BCmol/L)"

hsize=1.5
pt$hsize=hsize
pt$Freq = round(pt$Freq,2)
pt$Var1 = factor(pt$Var1,levels=c("Adequate (â‰¥ 50 Î¼mol/L)",
                                  "Inadequate (23-49 Î¼mol/L)",
                                  "Deficient (< 23 Î¼mol/L)" ))

p2.pi = ggplot(pt[pt$Var2 == 0,],aes(x =hsize,y = Freq,fill=Var1 )) +
  geom_col() +facet_wrap(~Group)+
  coord_polar(theta = "y",clip="off") +
  xlim(c(0.2, hsize + 0.5))+
  geom_text(aes(label = scales::percent(Freq)),
          position = position_stack(vjust = 0.5))+
  theme_prism(base_family = "") + 
  theme(legend.position = "bottom",
    panel.background = element_rect(fill = "white"),
                        panel.grid = element_blank(),
                        axis.title = element_blank(),
                        axis.ticks = element_blank(),
                        axis.text = element_blank(),
                        axis.line = element_blank(),
        legend.key.spacing.x = grid::unit(1,"cm")) +
  scale_fill_manual(values=c("Adequate (â‰¥ 50 Î¼mol/L)" = "palegreen3","Inadequate (23-49 Î¼mol/L)" = "steelblue3","Deficient (< 23 Î¼mol/L)" = "red3"),breaks=c("Deficient (< 23 Î¼mol/L)"  , 
                             "Inadequate (23-49 Î¼mol/L)",
                              "Adequate (â‰¥ 50 Î¼mol/L)"))+ggtitle("Vitamin C concentration in peripheral blood plasma at inclusion")

cowplot::plot_grid(p2.pi,p2.boxplot.vitc,p2.smoothed.vitc,ncol=1,labels=c("a","b","c"),label_size = 15,scale=0.9)

ggsave("vitaminc_concentration_figure.pdf",width=10,height=14,device=cairo_pdf)

#### This section is to create individual pdfs for each panel
p2.pi
ggsave("vitaminc_concentration_figure.a.pdf",width=9,height=14/3,device=cairo_pdf)

p2.boxplot.vitc
ggsave("vitaminc_concentration_figure.b.pdf",width=9,height=14/3,device=cairo_pdf)

p2.smoothed.vitc
ggsave("vitaminc_concentration_figure.c.pdf",width=9,height=14/3,device=cairo_pdf)

```

### Supplemental figure, Vitamin C change over time

**Supplemental figure xx. Change in vitamin C over time.** 

```{r, fig.height=5, fig.width = 5.5 }

#Give variable names labels for formatting
label(df$VitC_PB1) <- 'Baseline Vitamin C'
label(df$VitC_PB5) <- 'EOT Vitamin C'

table1::table1(~ VitC_PB1+VitC_PB5 |Treatment_arm,data=df,
               render.continuous=my.render.cont)
               
#robust regression on mean change over time
fit = rlm(VitC_PB5 - VitC_PB1 ~ Treatment_arm,data=df)

knitr::kable(test(emmeans(fit,"Treatment_arm",type="response")),caption="change in vitamin C")

p1.delta.c = ggplot(df,aes(x =Treatment_arm,y= VitC_PB5 - VitC_PB1, color=Treatment_arm)) + geom_quasirandom() +
  geom_boxplot(fill=NA,outliers=F)+scale_color_manual(values =c("black",pal_nejm()(4)[3]))+theme_prism()+theme(legend.position = "none") + xlab("") + ylab("Change in plasma vitamin C \n(End of treatment - Baseline)") + annotate(geom = "line",x = c(1,2), y = c(135,135)) + annotate(geom="text" , x=1.5,y = 145,label="p < 0.0001 ",fontface="italic")

fit = rlm(log2(VitC_PB5/VitC_PB1) ~ Treatment_arm,data=df)
# knitr::kable(test(emmeans(fit,"Treatment_arm")),caption="change in vitamin C")
knitr::kable(summary(confint(emmeans(fit,pairwise~Treatment_arm,type="response"))),caption="Fold change in vitamin C")

p2.delta.c = ggplot(df,aes(x =Treatment_arm,y= log2(VitC_PB5/VitC_PB1), color=Treatment_arm)) + geom_quasirandom() +
  geom_boxplot(fill=NA,outliers=F)+scale_color_manual(values =c("black",pal_nejm()(4)[3]))+theme_prism()+theme(legend.position = "none") + xlab("") + ylab("log2(FC) plasma vitamin C \n(End of treatment - Baseline)") + annotate(geom = "line",x = c(1,2), y = c(6.5,6.5)) + annotate(geom="text" , x=1.5,y = 7.02,label="p < 0.0001 ",fontface="italic")

##Keeping the FC plot for manuscript
p2.delta.c

```


### Supplemental figure, plasma versus bone marrow vitamin C

```{r, fig.width = 10,fig.height=5}

##Make a bone marrow data frame for easier processing, code is basically identical to plasma vitamin c over time
df.bm =  subset(df,select = c(ID,Treatment_arm,VitC_BM1,VitC_BM2,VitC_BM5,Age,Sex,Hb1,CCUS))
df.bm =  reshape2::melt(df.bm,id.vars=c("ID","Treatment_arm","Age","Sex","Hb1","CCUS"))
df.bm$variable = factor(df.bm$variable, levels = c("VitC_BM1","VitC_BM2","VitC_BM5"))

df.bm$variable = case_when(df.bm$variable == "VitC_BM1" ~ 0,
                               df.bm$variable == "VitC_BM2" ~ 365.25/4,
                               df.bm$variable == "VitC_BM5" ~ 365.25)

tmp1 = data.frame(variable =12.1/12*365.25,value2 = 11.5,Treatment_arm = "Vitamin C" )
tmp2 = data.frame(variable =12.1/12*365.25,value2 = 36.5,Treatment_arm = "Vitamin C" )

#This is for plotting purposes only and is noted in the figure legend
df.bm$value2 = df.bm$value
df.bm$value2[df.bm$value>200] = 200
df.bm$value2[df.bm$value2==200 & df.bm$variable == 182.6250] = NA
df.sp$Tissue = "Peripheral Blood"
df.bm$Tissue = "Bone Marrow"

df.vitc = rbind(df.sp[,colnames(df.sp) %in% colnames(df.bm)],df.bm)
df.vitc$ID = paste0(df.vitc$ID,df.vitc$Tissue)

ggplot(df.vitc[!is.na(df.vitc$value),],aes(x = 12*as.numeric(variable)/365.25,y=value2,color=Tissue,fill=Tissue)) + theme_prism() + geom_line(aes(group =  ID),alpha=0.15 ) + stat_smooth(se=T,linewidth=1.5,span=1) + scale_color_manual(values =c("#648ad3","#880808")) + scale_fill_manual(values =c("#648ad3","#880808")) +scale_y_continuous(expand=c(0,0))+ scale_x_continuous(breaks=c(0,3,6,9,12),expand=c(0,0)) +xlab("Months from randomisation")+ylab("Vitamin C concentration (\U03BCmol/L)")+
  guides(color=guide_legend(title=""),fill=guide_legend(title=""))+coord_cartesian(clip="off")+facet_wrap(~Treatment_arm)+ theme(panel.spacing = unit(2, "lines"))

```



### Correlate Vitamin C in plasma and bone marrow
Spearman 

```{r, fig.height = 5,fig.width=15}
#To correlate bone marrow with plasma we need a wide data.frame, we're creating this merging the two data.frames
colnames(df.sp)[colnames(df.sp) == "value"] = "blood"
colnames(df.bm)[colnames(df.bm) == "value"] = "bone"
df.sp = subset(df.sp,select= - c(Tissue))
df.bm = subset(df.bm,select= - c(Tissue))
df.vitc.corr = merge(df.sp,df.bm,by=c("ID","Treatment_arm","variable"))
df.vitc.corr$variable = factor(df.vitc.corr$variable,levels=c(0,91.3125,365.25),labels=c("Baseline","3 months","12 months"))

ggplot(df.vitc.corr,aes(x = blood,y=bone )) + geom_point() +
  facet_wrap(~variable) +theme_prism() +stat_smooth(method="lm",color="black",fill="black") +
  stat_cor(method = "spearman",p.accuracy = 0.0001, r.accuracy = 0.01,
             aes(label = paste(..r.label.., ..p.label.., sep = "~`,`~"))) +
  ylab("Bone marrow vitamin C (Î¼mol/L)") + 
  xlab("Peripheral blood vitamin C (Î¼mol/L)") +
   theme(panel.spacing = unit(2, "lines"))

```

## Survival {.tabset}
Includes all samples, unadjusted. We also performed a landmark analysis and analyzed this same data with peto-peto. Both were still significant, therefore an FDR correction would not change our findings here.


### Primary outcome KM (Figure 3)

**Unadjusted Kaplan-Meier curves for overall survival.** Estimated hazard ratio for vitamin C treated patients versus controls based on an unadjusted Cox proportional hazards regression 0.35, 95% CI 0.17, 0.71.

```{r,fig.width=8, fig.height=9}
#Years of FU and Months of FU
df$Years_FU_OS = (df$Days_FU_OS/365.25)
df$Months_FU_OS = (df$Days_FU_OS/365.25 * 12)

#Label these variables for formatting
label(df$Months_FU_OS) = "Follow up (Months)"
label(df$FU_Death) = "Died"

df$FU_event = factor(df$FU_Death,levels=c(0,1),labels=c("Censored","Died"))
label(df$FU_event) = "Event"

table1::table1(~ Months_FU_OS +FU_event |Treatment_arm,data=df,
               render.continuous=c(.="Mean (sd)", 
                                           .="Median [Q1, Q3]"))

fit_os = survfit(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df)

os.p =  ggsurvplot(fit_os,
           legend.labs =gsub("Treatment_arm=","",names(fit_os$strata)),
           palette=c("black",pal_nejm()(4)[c(3)]),
           cumevents=F,
           risk.table="nrisk_cumcensor",
           # cumcensor=T,
           break.time.by=365.25,
           xscale="d_m",
           xlab="Months from randomisation",
           pval.method = F,
           pval=F,legend.title="")

#Also fit Cox model so we can annotate with HR and 95% CI
fit_os.cox = coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df)

p.y = min(os.p$data.survplot[os.p$data.survplot$Treatment_arm != "Placebo", ]$surv)
p.c = min(os.p$data.survplot[os.p$data.survplot$Treatment_arm == "Placebo", ]$surv)
p.os = os.p$plot+theme(legend.position = "none") +
  annotate(geom="text",x = 6*365.25,y = p.y,label="Vitamin C", color=pal_nejm()(4)[c(3)],hjust=0) + 
  annotate(geom="text",x = 6*365.25,y = p.c,label="Placebo", color="black",hjust=0) + 
  scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+
  coord_cartesian(clip="off")+theme_prism()+theme(legend.position = "none")+
  geom_hline(yintercept = 0.5,linetype="dotted",linewidth = 1.25) +theme(legend.position = "none",plot.margin = unit(c(0, 1, 0, 0), "inches")) + annotate(geom="text", x = 365,y = 0.25, label=paste0("Log-rank\np =  0.0025\nHR ",round(exp(fit_os.cox$coefficients),2),"; ", "95% CI ",round(exp(confint(fit_os.cox)[1]),2),", ", round(exp(confint(fit_os.cox)[2]),2) ),fontface="italic",hjust=0,size=4.5)+ggtitle("Overall survival")

t.os = os.p$table + theme_prism()+ theme(axis.line = element_blank())+scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+coord_cartesian(clip="off")+theme(text = element_text(face="bold",family=""),axis.text.y = element_text(hjust=1,colour = c(pal_nejm()(4)[c(3)],"black"),margin= margin(0, 15, 0, 7) ),axis.ticks.y = element_blank(),plot.margin = unit(c(0, 0, 0, 0), "inches"))+xlab("")  +ylab("Number at risk\n(Cumulative censoring)") + theme(axis.text.x = element_blank(),axis.ticks.x = element_blank(),plot.title = element_blank())

#Piece the KM and risk table together
p.os/t.os/plot_spacer() + plot_layout(heights = c(1,0.25,0.02))

ggsave("survival_figure.a.pdf",width=8,height=9.5)


knitr::kable(summary(fit_os.cox)$conf.int,caption="Hazard ratio and CI")

p.fig.3a = p.os/t.os +plot_layout(heights = c(1,0.25))

```

### Cox model (No interaction)

**Supplemntal figure XX. Forest plot for adjusted Cox model.** Cox proportional hazards regression to estimated efficacy of vitamin C on overall survival was adjusted for patient age at inclusion, gender, hemoglobin levels at inclusion, and whether or not they had been diagnosed with CCUS. 

```{r, fig.width=11, figheight=5}

df$Treatment = as.numeric(factor(df$Treatment_arm,levels=c("Placebo","Vitamin C"),labels=c("Placebo","VitC")))

df$VitaminC = as.numeric(factor(df$Treatment_arm,levels=c("Placebo","Vitamin C"),labels=c("Placebo","VitC")))-1

#Create a Male variable for forest plots
df$Male = df$Sex

## Better labels for the forest plot
labelled::var_label(df) <- list(
    VitaminC = "Vitamin C group", 
    Age = "Age (years)",
    Hb1 = "Haemoglobin at baseline (g/dL)",
    CCUS = "CCUS diagnosis (yes/no)"
)

#We'll call this the simple model as it does not have any exploratory interactions
fit.cox.simple =  coxph(Surv(Days_FU_OS,FU_Death)~VitaminC+Age +Male + Hb1 + CCUS,data=df)

panels <- list(
  list(width = 0.03),
  list(width = 0.1, display = ~variable, fontface = "bold", heading = "Variable"),
  list(width = 0.1, display = ~level),
  list(width = 0.03, item = "vline", hjust = 0.5),
  list(
    width = 0.55, item = "forest", hjust = 0.5, heading = "Hazard ratio", linetype = "dashed",
    line_x = 0
  ),
  list(width = 0.05, item = "vline", hjust = 0.5),
  list(width = 0.12, display = ~ ifelse(reference, "Reference", sprintf(
    "%0.2f (%0.2f, %0.2f)",
    trans(estimate), trans(conf.low), trans(conf.high)
  )), display_na = NA, heading = "HR (95% CI)"),
    list(width = 0.03, item = "vline", hjust = 0.5),
  list(
    width = 0.05,
    display = ~ ifelse(reference, "", paste0(ifelse(p.value < 0.0001,"P ","p =  "), signif(round(p.value,4),2))),
    display_na = NA, hjust = 0.5,fontface="italic"
  ),
  list(width = 0.03)
)

forest_model(fit.cox.simple,panels) 

```

### Cox model (Age interaction)

**Supplemental figure XX. Forest plot for adjusted Cox model in age interaction.** Cox proportional hazards regression to estimated efficacy of vitamin C as a function of age. Age was mean centered and converted to decades to improve hazard ratio scales. Overall vitamin C was still effective, with evidence that it's efficacy decreases with age.

```{r, fig.width=11, figheight=5}
#Create a centered age for better interpretation when we include an interaction
df$age_c = (df$Age-mean(df$Age,na.rm=T))

#age * treatmentr variable
df$VitaminC_Age_Interaction = as.numeric(as.factor(df$Treatment_arm))*df$age_c
df$age_tx = df$age_c * df$VitaminC
labelled::var_label(df) <- list(
    VitaminC = "Vitamin C group", 
    age_c = "Age (Years)",
    age_tx = "Age x Vitamin C group",
    Hb1 = "Haemoglobin at baseline (g/dL)",
    CCUS = "CCUS diagnosis (yes/no)"
)

fit.cox.age_int =  coxph(Surv(Days_FU_OS,FU_Death)~VitaminC+age_c + age_tx+Male + Hb1 + CCUS,data=df)

forest_model(fit.cox.age_int,panels)

```


### Cox model (TET interaction)

```{r, fig.width=11, figheight=5}

df$TET2 = ifelse(gsub("-","",df$ID) %in% mut_ids,"TET2Mut", "TET2neg")

knitr::kable(table(df$TET2,df$Treatment_arm))

#Binarized mutationstatus
df$TET2.bi = ifelse(df$ID %in% mut_ids,1,0)
df$tet_tx = I(df$TET2 == "TET2Mut") * df$VitaminC

labelled::var_label(df) <- list(
    VitaminC = "Vitamin C group", 
    age_c = "Age (Years)",
    tet_tx = "TET2 or IDH1/2 x Vitamin C group",
    Hb1 = "Haemoglobin at baseline (g/dL)",
    CCUS = "CCUS diagnosis (yes/no)",
    TET2.bi = "TET2 or IDH1/2 Mutation (yes/no)"
)

fit.cox.age_int =  coxph(Surv(Days_FU_OS,FU_Death)~VitaminC + age_c + TET2.bi + tet_tx + Male + Hb1 + CCUS,data=df)

forest_model(fit.cox.age_int,panels)

```


### Cox model (Hb1 interaction)

```{r, fig.width=11, figheight=5}
#center Hb1 then create interaction var
df$Hb1_c = df$Hb1 - mean(df$Hb1,na.rm=T)
df$vitc_hb1 = df$Hb1_c * df$VitaminC

labelled::var_label(df) <- list(
    VitaminC = "Vitamin C group", 
    age_c = "Age (Years)",
    vitc_hb1 = "Haemoglobin at baseline (g/dL) x Vitamin C group",
    Hb1_c = "Haemoglobin at baseline (g/dL)",
    CCUS = "CCUS diagnosis (yes/no)"
)

fit.cox.hb1_int =  coxph(Surv(Days_FU_OS,FU_Death)~VitaminC+age_c + vitc_hb1+Male + Hb1_c + CCUS,data=df)

forest_model(fit.cox.hb1_int,panels)


```



### Cox model (Sex interaction)


```{r, fig.width=11, figheight=5}

#sex and treatment interaction
df$sex_vitc = df$Male * df$VitaminC
labelled::var_label(df) <- list(
    VitaminC = "Vitamin C group", 
    age_c = "Age (Years)",
    sex_vitc = "Male x Vitamin C group",
    Hb1_c = "Haemoglobin at baseline (g/dL)",
    CCUS = "CCUS diagnosis (yes/no)"
)

fit.cox.sex_int =  coxph(Surv(Days_FU_OS,FU_Death)~VitaminC+age_c + sex_vitc+Male + Hb1_c + CCUS,data=df)

forest_model(fit.cox.sex_int,panels)

```



### Cox model (CCUS interaction)

Forest plot cannot be made due to diverging SE in treatment

```{r, fig.width=11, figheight=5}

df$vitc_ccus = (df$CCUS-1) * df$VitaminC

labelled::var_label(df) <- list(
    VitaminC = "Vitamin C group", 
    age_c = "Age (Years)",
    vitc_ccus = "CCUS x Vitamin C group",
    Hb1_c = "Haemoglobin at baseline (g/dL)",
    CCUS = "CCUS diagnosis (yes/no)"
)

fit.cox.ccus_int =  coxph(Surv(Days_FU_OS,FU_Death)~VitaminC+age_c + vitc_ccus+Male + Hb1_c + CCUS,data=df)

fit.cox.ccus_int

## Forst plot could not be made due to divergence
# forest_model(fit.cox.ccus_int,panels)

```


### Cox model (baseline vitamin C interaction)

```{r, fig.width=11, figheight=5}

df$VitC_PB1_c = df$VitC_PB1 - mean(df$VitC_PB1,na.rm=T)
df$vitc_PB1_int = df$VitC_PB1_c * df$VitaminC
labelled::var_label(df) <- list(
    VitaminC = "Vitamin C group", 
    age_c = "Age (Years)",
    VitC_PB1 = "Baseline vitamin C (Î¼mol/L)",
    vitc_PB1_int = "Baseline vitamin C (Î¼mol/L) x Vitamin C group",
    Hb1_c = "Haemoglobin at baseline (g/dL)",
    CCUS = "CCUS diagnosis (yes/no)"
)

fit.cox.vitc_pb1_int =  coxph(Surv(Days_FU_OS,FU_Death)~VitaminC+age_c + VitC_PB1+Male + vitc_PB1_int + Hb1_c + CCUS,data=df)

forest_model(fit.cox.vitc_pb1_int,panels)

```


### 2-year landmark 

**Maybe Figure?** A 2 year landmark analysis was performed to determine if there was still evidence of vitamin C efficacy after removing individuals who had events prior to 2 years. This landmark analysis ensures the effects of vitamin C had sufficient time to reach efficacy.


```{r,fig.width=8, fig.height=9}

knitr::kable(subset(df[df$Days_FU_OS<2*365.25,],select=c(ID, Days_FU_OS,FU_Death,Treatment_arm)),caption="Removed from landmark")

fit_os.L = survfit(Surv(Days_FU_OS-2*365.25,FU_Death)~Treatment_arm,data=df[df$Days_FU_OS>=2*365.25,])

os.p =  ggsurvplot(fit_os.L,
           legend.labs =gsub("Treatment_arm=","",names(fit_os$strata)),
           palette=c("black",pal_nejm()(4)[c(3)]),
           cumevents=F,
           risk.table="nrisk_cumcensor",
           # cumcensor=T,
           break.time.by=365.25,
           xscale="d_m",
           xlab="Months post 2 year landmark",
           pval.method = F,
           pval=F,legend.title="")


p.y = min(os.p$data.survplot[os.p$data.survplot$Treatment_arm != "Placebo", ]$surv)
p.c = min(os.p$data.survplot[os.p$data.survplot$Treatment_arm == "Placebo", ]$surv)
p.os = os.p$plot+theme(legend.position = "none") +
  annotate(geom="text",x = 4*365.25,y = p.y,label="Vitamin C", color=pal_nejm()(4)[c(3)],hjust=0) + 
  annotate(geom="text",x = 4*365.25,y = p.c,label="Placebo", color="black",hjust=0) + 
  scale_x_continuous(limits=c(0,365.25*4), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+
  coord_cartesian(clip="off")+theme_prism()+theme(legend.position = "none")+
  geom_hline(yintercept = 0.5,linetype="dotted",linewidth = 1.25) +theme(legend.position = "none",plot.margin = unit(c(0, 1, 0, 0), "inches")) + annotate(geom="text", x = 365,y = 0.25, label="Log-rank\np =  0.028",fontface="italic",hjust=0,size=4.5)

t.os = os.p$table + theme_prism()+ theme(axis.line = element_blank())+scale_x_continuous(limits=c(0,365.25*4), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+coord_cartesian(clip="off")+theme(text = element_text(face="bold",family=""),axis.text.y = element_text(hjust=1,colour = c(pal_nejm()(4)[c(3)],"black"),margin= margin(0, 15, 0, 7) ),axis.ticks.y = element_blank(),plot.margin = unit(c(0, 0, 0, 0), "inches"))+xlab("") + ggtitle("") +ylab("Number at risk\n(Cumulative censoring)") + theme(axis.text.x = element_blank(),axis.ticks.x = element_blank(),plot.title = element_blank())

p.os/t.os +plot_layout(heights = c(1,0.25))

```



### Per patient protocol
likelihood ratio test p-value is nearly identical to log-rank

```{r,fig.width=8, fig.height=9}
#Read in file with per patient IDs
df.pp =  read_excel("Vitc_Data/EVI2_per protocol population.xlsx")

df.pp$ID = sapply(strsplit(df.pp$ID,"-"), function(x){ifelse(length(x) >1, paste0(x[[1]],as.numeric(x[[2]])),x)})

knitr::kable(df$ID[which(! df$ID %in% df.pp$ID)],caption = "IDs removed from main")
#dummy variable is they're 'per-patient' or not
df$pp =  ifelse(df$ID %in% df.pp$ID,1,0)

fit_os = survfit(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$pp == 1 &df$ID != "USC60",])

os.p =  ggsurvplot(fit_os,
           legend.labs =gsub("Treatment_arm=","",names(fit_os$strata)),
           palette=c("black",pal_nejm()(4)[c(3)]),
           cumevents=F,
           risk.table="nrisk_cumcensor",
           # cumcensor=T,
           break.time.by=365.25,
           xscale="d_m",
           xlab="Months from randomisation",
           pval.method = F,
           pval=F,legend.title="")

p.y = min(os.p$data.survplot[os.p$data.survplot$Treatment_arm != "Placebo", ]$surv)
p.c = min(os.p$data.survplot[os.p$data.survplot$Treatment_arm == "Placebo", ]$surv)
p.os = os.p$plot+theme(legend.position = "none") +
  annotate(geom="text",x = 6*365.25,y = p.y,label="Vitamin C", color=pal_nejm()(4)[c(3)],hjust=0) +
  annotate(geom="text",x = 6*365.25,y = p.c,label="Placebo", color="black",hjust=0) +
  scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+
  coord_cartesian(clip="off")+theme_prism()+theme(legend.position = "none")+
  geom_hline(yintercept = 0.5,linetype="dotted",linewidth = 1.25) +theme(legend.position = "none",plot.margin = unit(c(0, 1, 0, 0), "inches")) 

t.os = os.p$table + theme_prism()+ theme(axis.line = element_blank())+scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+coord_cartesian(clip="off")+theme(text = element_text(face="bold",family=""),axis.text.y = element_text(hjust=1,colour = c(pal_nejm()(4)[c(3)],"black"),margin= margin(0, 15, 0, 7) ),axis.ticks.y = element_blank(),plot.margin = unit(c(0, 0, 0, 0), "inches"))+xlab("") + ggtitle("") +ylab("Number at risk\n(Cumulative censoring)") + theme(axis.text.x = element_blank(),axis.ticks.x = element_blank(),plot.title = element_blank())


HR.no = summary(coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$pp == 1 &df$ID != "USC60",]))$conf.int

HR.no.anno = paste0("HR ",round(HR.no[1],2), "\n95% CI ",round(HR.no[3],2),", ",round(HR.no[4],2))

p.os = p.os + annotate(geom="text", x = 365.25,y = 0.25, label=HR.no.anno,hjust=0,size=4.5)


p.os/t.os +plot_layout(heights = c(1,0.25))


fit1 = coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$pp == 1,])

fit2 = coxph(Surv(Days_FU_OS,FU_Death)~1,data=df[df$pp == 1,])
knitr::kable(lmtest::lrtest(fit1,fit2),caption = "LRT on treatment")

```



### Subgroups (suppl) {.tabset}
All subgroup analyses include only the hazard ratio with 95% CI as these are exploratory. 


#### TET1 or IDH1/2 at baseline

**Supplemental Figure XX. Overall survival by TET2 or IDH1/2 mutation status.** Hazard ratio and confidence intervals were estimated using Cox proportional hazards models on data stratified by TET2 and IDH1/2 mutation status. Survival curves and estimates are consistent with the primary outcome.

```{r, fig.width=14, fig.height=9}

fit1 = coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm*TET2,data=df)
fit2 = coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm+TET2,data=df)

knitr::kable(lmtest::lrtest(fit1,fit2),caption = "LRT for interaction")

fit_os.no.tet = survfit(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$TET2 == "TET2neg",])

p1 = ggsurvplot(fit_os.no.tet,
           legend.labs =gsub("TET2=","", gsub("Treatment_arm=","",names(fit_os.no.tet$strata))),
           palette=c("black",pal_nejm()(4)[c(3)]),
           cumevents=F,
           risk.table="nrisk_cumcensor",
           break.time.by=365.25,
           xscale="d_m",
           xlab="Months from randomisation",
           pval.method = F,
           pval=F,legend.title="TET2 Negative")

fit_os.tet = survfit(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$TET2 == "TET2Mut",])

p2 = ggsurvplot(fit_os.tet,
           legend.labs =gsub("TET2=","", gsub("Treatment_arm=","",names(fit_os.tet$strata))),
palette=c("black",pal_nejm()(4)[c(3)]),
cumevents=F,
           risk.table="nrisk_cumcensor",
           break.time.by=365.25,
           xscale="d_m",
           xlab="Months from randomisation",
           pval.method = F,
           pval=F,legend.title="TET2 Positive")

p.y1 = min(p1$data.survplot[p1$data.survplot$Treatment_arm != "Placebo", ]$surv)
p.c1 = min(p1$data.survplot[p1$data.survplot$Treatment_arm == "Placebo", ]$surv)

p.y2 = min(p2$data.survplot[p2$data.survplot$Treatment_arm != "Placebo", ]$surv)
p.c2 = min(p2$data.survplot[p2$data.survplot$Treatment_arm == "Placebo", ]$surv)



p1$plot = p1$plot + ggtitle(expression(bold(bolditalic("TET2") ~ or ~ bolditalic("IDH1/2") ~ "wild-type"))) + theme(legend.position = "none") +theme(legend.position = "none") +
  annotate(geom="text",x = 6*365.25,y = p.y1,label="Vitamin C", color=pal_nejm()(4)[c(3)],hjust=0) + 
  annotate(geom="text",x = 6*365.25,y = p.c1,label="Placebo", color="black",hjust=0) + 
  scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+
  coord_cartesian(clip="off")+theme_prism()+theme(legend.position = "none")+
  geom_hline(yintercept = 0.5,linetype="dotted",linewidth = 1.25) +theme(legend.position = "none",plot.margin = unit(c(0, 1, 0, 0), "inches"))  

HR.no = summary(coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$TET2 == "TET2neg",]))$conf.int

HR.no.anno = paste0("HR ",round(HR.no[1],2), "\n95% CI ",round(HR.no[3],2),", ",round(HR.no[4],2))

p1$plot = p1$plot + annotate(geom="text", x = 365,y = 0.25, label=HR.no.anno,hjust=0,size=4.5)

p2$plot = p2$plot + ggtitle(expression(bold(bolditalic("TET2") ~ or ~ bolditalic("IDH1/2") ~ "mutant"))) + theme(legend.position = "none") + theme(legend.position = "none") +theme(legend.position = "none") +
  annotate(geom="text",x = 6*365.25,y = p.y2,label="Vitamin C", color=pal_nejm()(4)[c(3)],hjust=0) + 
  annotate(geom="text",x = 6*365.25,y = p.c2,label="Placebo", color="black",hjust=0) + 
  scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+
  coord_cartesian(clip="off")+theme_prism()+theme(legend.position = "none")+
  geom_hline(yintercept = 0.5,linetype="dotted",linewidth = 1.25) +theme(legend.position = "none",plot.margin = unit(c(0, 1, 0, 0), "inches")) 


HR.tet = summary(coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$TET2 == "TET2Mut",]))$conf.int

HR.tet.anno = paste0("HR ",round(HR.tet[1],2), "\n95% CI ",round(HR.tet[3],2),", ",round(HR.tet[4],2))

p2$plot = p2$plot + annotate(geom="text", x = 365,y = 0.25, label=HR.tet.anno,hjust=0,size=4.5)


p1$table = p1$table +theme_prism()+ theme(axis.line = element_blank())+scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+coord_cartesian(clip="off")+theme(text = element_text(face="bold",family=""),axis.text.y = element_text(hjust=1,colour = c(pal_nejm()(4)[c(3)],"black"),margin= margin(0, 15, 0, 7) ),axis.ticks.y = element_blank(),plot.margin = unit(c(0, 0, 0, 0), "inches"))+xlab("") + ggtitle("") +ylab("Number at risk\n(Cumulative censoring)") + theme(axis.text.x = element_blank(),axis.ticks.x = element_blank(),plot.title = element_blank())

p2$table = p2$table +theme_prism()+ theme(axis.line = element_blank())+scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+coord_cartesian(clip="off")+theme(text = element_text(face="bold",family=""),axis.text.y = element_text(hjust=1,colour = c(pal_nejm()(4)[c(3)],"black"),margin= margin(0, 15, 0, 7) ),axis.ticks.y = element_blank(),plot.margin = unit(c(0, 0, 0, 0), "inches"))+xlab("") + ggtitle("") +ylab("Number at risk\n(Cumulative censoring)") + theme(axis.text.x = element_blank(),axis.ticks.x = element_blank(),plot.title = element_blank())

#We flipped the order
p2$plot  + p1$plot + p2$table +p1$table +plot_spacer()+plot_spacer() +plot_layout(ncol=2,heights = c(1,0.25,0.02))

ggsave("survival_figure_b.pdf",width=18,height=9.5)

p.fig.3b = p2$plot  + p1$plot + p2$table +p1$table +plot_layout(ncol=2,heights = c(1,0.25))
```

### Manuscript Figure

```{r, fig.width=20, fig.height=9}

cowplot::plot_grid(p.fig.3a,p.fig.3b,ncol=2,scale=0.9,labels=c('a','b'),label_size=15,rel_widths = c(0.5,1))

ggsave("survival_figure.pdf",width=26,height=9.5)


```

#### LOF TET2

AKA TET2 hypermethylation vs WT signature

```{r, fig.width=14, fig.height=9}
#Read in the TET2 signatures based on methylation
lof.tet2 = read.csv("250306_SU2C_Vitamin_C_TET2_meth_signatures_categories.csv")

lof.tet2$Patient_ID = gsub("AAL0","AAL",lof.tet2$Patient_ID)
df$lof.tet2 = rep(NA)
df$lof.tet2[gsub("-","",df$ID) %in% unique(lof.tet2$Patient_ID[lof.tet2$Timepoint == "Baseline" & lof.tet2$TET2_Meth_Signature == "High"])] = "TET2 hypermethylation signature"

df$lof.tet2[gsub("-","",df$ID) %in% unique(lof.tet2$Patient_ID[lof.tet2$Timepoint == "Baseline" & lof.tet2$TET2_Meth_Signature == "Low"])] = "Wildtype"

fit1 = coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm*lof.tet2,data=df)
fit2 = coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm+lof.tet2,data=df)

knitr::kable(lmtest::lrtest(fit1,fit2),caption = "LRT for interaction")

fit_os.no.tet = survfit(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$lof.tet2 == "Wildtype",])

p1 = ggsurvplot(fit_os.no.tet,
           legend.labs =gsub("TET2=","", gsub("Treatment_arm=","",names(fit_os.no.tet$strata))),
           palette=c("black",pal_nejm()(4)[c(3)]),
           cumevents=F,
           risk.table="nrisk_cumcensor",
           break.time.by=365.25,
           xscale="d_m",
           xlab="Months from randomisation",
           pval.method = F,
           pval=F,legend.title="Low TET2")

fit_os.tet = survfit(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$lof.tet2 == "TET2 hypermethylation signature",])

p2 = ggsurvplot(fit_os.tet,
           legend.labs =gsub("TET2=","", gsub("Treatment_arm=","",names(fit_os.tet$strata))),
palette=c("black",pal_nejm()(4)[c(3)]),
cumevents=F,
           risk.table="nrisk_cumcensor",
           break.time.by=365.25,
           xscale="d_m",
           xlab="Months from randomisation",
           pval.method = F,
           pval=F,legend.title="High TET2")

p.y1 = min(p1$data.survplot[p1$data.survplot$Treatment_arm != "Placebo", ]$surv)
p.c1 = min(p1$data.survplot[p1$data.survplot$Treatment_arm == "Placebo", ]$surv)

p.y2 = min(p2$data.survplot[p2$data.survplot$Treatment_arm != "Placebo", ]$surv)
p.c2 = min(p2$data.survplot[p2$data.survplot$Treatment_arm == "Placebo", ]$surv)



p1$plot = p1$plot + ggtitle("TET2 wildtype methylation signature") + theme(legend.position = "none") +theme(legend.position = "none") +
  annotate(geom="text",x = 6*365.25,y = p.y1,label="Vitamin C", color=pal_nejm()(4)[c(3)],hjust=0) + 
  annotate(geom="text",x = 6*365.25,y = p.c1,label="Placebo", color="black",hjust=0) + 
  scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+
  coord_cartesian(clip="off")+theme_prism()+theme(legend.position = "none")+
  geom_hline(yintercept = 0.5,linetype="dotted",linewidth = 1.25) +theme(legend.position = "none",plot.margin = unit(c(0, 1, 0, 0), "inches"))  

HR.no = summary(coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$lof.tet2 == "Wildtype",]))$conf.int

HR.no.anno = paste0("HR ",round(HR.no[1],2), "\n95% CI ",round(HR.no[3],2),", ",round(HR.no[4],2))

p1$plot = p1$plot + annotate(geom="text", x = 365,y = 0.25, label=HR.no.anno,hjust=0,size=4.5)

p2$plot = p2$plot + ggtitle(expression(bold("TET2"^bolditalic(mut) ~"hypermethylation signature"))) + theme(legend.position = "none") + theme(legend.position = "none") +theme(legend.position = "none") +
  annotate(geom="text",x = 6*365.25,y = p.y2,label="Vitamin C", color=pal_nejm()(4)[c(3)],hjust=0) + 
  annotate(geom="text",x = 6*365.25,y = p.c2,label="Placebo", color="black",hjust=0) + 
  scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+
  coord_cartesian(clip="off")+theme_prism()+theme(legend.position = "none")+
  geom_hline(yintercept = 0.5,linetype="dotted",linewidth = 1.25) +theme(legend.position = "none",plot.margin = unit(c(0, 1, 0, 0), "inches") ) 

HR.tet = summary(coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$lof.tet2 == "TET2 hypermethylation signature",]))$conf.int

HR.tet.anno = paste0("HR ",round(HR.tet[1],2), "\n95% CI ",round(HR.tet[3],2),", ",round(HR.tet[4],2))

p2$plot = p2$plot + annotate(geom="text", x = 365,y = 0.25, label=HR.tet.anno,hjust=0,size=4.5)


p1$table = p1$table +theme_prism()+ theme(axis.line = element_blank())+scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+coord_cartesian(clip="off")+theme(text = element_text(face="bold",family=""),axis.text.y = element_text(hjust=1,colour = c(pal_nejm()(4)[c(3)],"black"),margin= margin(0, 15, 0, 7) ),axis.ticks.y = element_blank(),plot.margin = unit(c(0, 0, 0, 0), "inches"))+xlab("") + ggtitle("") +ylab("Number at risk\n(Cumulative censoring)") + theme(axis.text.x = element_blank(),axis.ticks.x = element_blank(),plot.title = element_blank())

p2$table = p2$table +theme_prism()+ theme(axis.line = element_blank())+scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+coord_cartesian(clip="off")+theme(text = element_text(face="bold",family=""),axis.text.y = element_text(hjust=1,colour = c(pal_nejm()(4)[c(3)],"black"),margin= margin(0, 15, 0, 7) ),axis.ticks.y = element_blank(),plot.margin = unit(c(0, 0, 0, 0), "inches"))+xlab("") + ggtitle("") +ylab("Number at risk\n(Cumulative censoring)") + theme(axis.text.x = element_blank(),axis.ticks.x = element_blank(),plot.title = element_blank())

p1$plot  + p2$plot + p1$table +p2$table +plot_layout(ncol=2,heights = c(1,0.25))


```

#### Above/Below median Vitamin C

**Supplemental figure XX. Overall survival by vitamin C above or below median vitamin C at inclusion.** Hazard ratio and confidence intervals were estimated using Cox proportional hazards models on data stratified by patients with plasma vitamin C level below the median measure at inclusion, and greater than or equal to the median. Results are consistent with the primary outcome.  

```{r, fig.width=14, fig.height=9}

df$med_vitc = ifelse(df$VitC_PB1 >= median(df$VitC_PB1,na.rm=T),"above","below")

fit1 = coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm*med_vitc,data=df)
fit2 = coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm+med_vitc,data=df)
knitr::kable(lmtest::lrtest(fit1,fit2),caption = "LRT for interaction")

fit_os.vitc.be = survfit(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$med_vitc == "below",])

p1 = ggsurvplot(fit_os.vitc.be,
           legend.labs =gsub("TET2=","", gsub("Treatment_arm=","",names(fit_os.vitc.be$strata))),
           palette=c("black",pal_nejm()(4)[c(3)]),
           cumevents=F,
           risk.table="nrisk_cumcensor",
           break.time.by=365.25,
           xscale="d_m",
           xlab="Months from randomisation",
           pval.method = F,
           pval=F,legend.title="TET2 Negative")



fit_os.vitc.a = survfit(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$med_vitc == "above",])

p2 = ggsurvplot(fit_os.vitc.a,
           legend.labs =gsub("TET2=","", gsub("Treatment_arm=","",names(fit_os.vitc.a$strata))),
palette=c("black",pal_nejm()(4)[c(3)]),
cumevents=F,
           risk.table="nrisk_cumcensor",
           break.time.by=365.25,
           xscale="d_m",
           xlab="Months from randomisation",
           pval.method = F,
           pval=F,legend.title="TET2 Positive")

p.y1 = min(p1$data.survplot[p1$data.survplot$Treatment_arm != "Placebo", ]$surv)
p.c1 = min(p1$data.survplot[p1$data.survplot$Treatment_arm == "Placebo", ]$surv)

p.y2 = min(p2$data.survplot[p2$data.survplot$Treatment_arm != "Placebo", ]$surv)
p.c2 = min(p2$data.survplot[p2$data.survplot$Treatment_arm == "Placebo", ]$surv)

p1$plot = p1$plot + ggtitle("Below the median vitamin C level at inclusion") + theme(legend.position = "none") +theme(legend.position = "none") +
  annotate(geom="text",x = 6*365.25,y = p.y1,label="Vitamin C", color=pal_nejm()(4)[c(3)],hjust=0) + 
  annotate(geom="text",x = 6*365.25,y = p.c1,label="Placebo", color="black",hjust=0) + 
  scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+
  coord_cartesian(clip="off")+theme_prism()+theme(legend.position = "none")+
  geom_hline(yintercept = 0.5,linetype="dotted",linewidth=1.25) +theme(legend.position = "none",plot.margin = unit(c(0, 1, 0, 0), "inches")) 


HR.a = summary(coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$med_vitc == "below",]))$conf.int
HR.a.anno = paste0("HR ",round(HR.a[1],2), "\n95% CI ",round(HR.a[3],2),", ",round(HR.a[4],2))
p1$plot = p1$plot + annotate(geom="text", x = 365,y = 0.25, label=HR.a.anno,hjust=0,size=4.5)


p2$plot = p2$plot + ggtitle("At or above the median vitamin C level at inclusion") + theme(legend.position = "none") + theme(legend.position = "none") +theme(legend.position = "none") +
  annotate(geom="text",x = 6*365.25,y = p.y2,label="Vitamin C", color=pal_nejm()(4)[c(3)],hjust=0) + 
  annotate(geom="text",x = 6*365.25,y = p.c2,label="Placebo", color="black",hjust=0) + 
  scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+
  coord_cartesian(clip="off")+theme_prism()+theme(legend.position = "none")+
  geom_hline(yintercept = 0.5,linetype="dotted",linewidth=1.25) +theme(legend.position = "none",plot.margin = unit(c(0, 1, 0, 0), "inches")) 

HR.a = summary(coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$med_vitc == "above",]))$conf.int
HR.a.anno = paste0("HR ",round(HR.a[1],2), "\n95% CI ",round(HR.a[3],2),", ",round(HR.a[4],2))
p2$plot = p2$plot + annotate(geom="text", x = 365,y = 0.25, label=HR.a.anno,hjust=0,size=4.5)

p1$table = p1$table +theme_prism()+ theme(axis.line = element_blank())+scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+coord_cartesian(clip="off")+theme(text = element_text(face="bold",family=""),axis.text.y = element_text(hjust=1,colour = c(pal_nejm()(4)[c(3)],"black"),margin= margin(0, 15, 0, 7) ),axis.ticks.y = element_blank(),plot.margin = unit(c(0, 0, 0, 0), "inches"))+xlab("") + ggtitle("") +ylab("Number at risk\n(Cumulative censoring)") + theme(axis.text.x = element_blank(),axis.ticks.x = element_blank(),plot.title = element_blank())


p2$table = p2$table +theme_prism()+ theme(axis.line = element_blank())+scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+coord_cartesian(clip="off")+theme(text = element_text(face="bold",family=""),axis.text.y = element_text(hjust=1,colour = c(pal_nejm()(4)[c(3)],"black"),margin= margin(0, 15, 0, 7) ),axis.ticks.y = element_blank(),plot.margin = unit(c(0, 0, 0, 0), "inches"))+xlab("") + ggtitle("") +ylab("Number at risk\n(Cumulative censoring)") + theme(axis.text.x = element_blank(),axis.ticks.x = element_blank(),plot.title = element_blank())

p1$plot  + p2$plot +p1$table +p2$table +plot_layout(ncol=2,heights = c(1,0.25))


```

#### Diagnostic subgroup

**Supplemental figure XX. Overall survival by diagnostic subgroup at inclusion.** Hazard ratio and confidence intervals were estimated using Cox proportional hazards models on data stratified by diagnosis at inclusion. CCUS appears to be a subgroup that can benefit particularly well from vitamin C supplementation. 

```{r, fig.width=18,fig.height=7.75}

fit1 = coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm*Diagnosis_simple_baseline,data=df)
fit2 = coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm+Diagnosis_simple_baseline,data=df)
knitr::kable(lmtest::lrtest(fit1,fit2),caption = "LRT for interaction")

fit_os.diags = survfit(Surv(Days_FU_OS,FU_Death)~Treatment_arm+Diagnosis_simple_combo,data=df)

p.a = ggsurvplot(fit_os.diags,
           legend.labs =gsub("Diagnosis_simple_combo=","",gsub("Treatment_arm=","",names(fit_os.diags$strata))),
           palette=c("black",pal_nejm()(7)[c(5,2,3,1,7)]),
           cumevents=F,
           risk.table=F,
           break.time.by=365.25,
           xscale="d_m",
           xlab="Months from randomisation",
           pval=T,pval.method = T,
           title= "All patients",legend.title="")

df$Sex.c = ifelse(df$Sex==1,"Male","Female")

fit_os.CCUS = survfit(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$Diagnosis_simple_combo=="CCUS",])

p1 = ggsurvplot(fit_os.CCUS,
           legend.labs =gsub("Treatment_arm=","",names(fit_os.CCUS$strata)),
           palette=c("black",pal_nejm()(7)[c(3)]),
           cumevents=F,
           risk.table="nrisk_cumcensor",
           break.time.by=365.25,
           xscale="d_m",
           xlab="Months from randomisation",
           pval=F,pval.method = F,legend.title="CCUS")



fit_os.mds = survfit(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$Diagnosis_simple_combo=="MDS",])

p2 = ggsurvplot(fit_os.mds,
           legend.labs =gsub("Treatment_arm=","",names(fit_os.mds$strata)),
           palette=c("black",pal_nejm()(7)[c(3)]),
           cumevents=F,
           risk.table="nrisk_cumcensor",
           break.time.by=365.25,
           xscale="d_m",
           xlab="Months from randomisation",
           pval=F,pval.method = F,legend.title="MDS")



fit_os.CMMLplus = survfit(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$Diagnosis_simple_combo=="CMML_MDS_MPN",])

p3 = ggsurvplot(fit_os.CMMLplus,
           legend.labs =gsub("Treatment_arm=","",names(fit_os.CMMLplus$strata)),
           palette=c("black",pal_nejm()(7)[c(3)]),
           cumevents=F,
           risk.table="nrisk_cumcensor",
           break.time.by=365.25,
           xscale="d_m",
           xlab="Months from randomisation",
           pval=F,pval.method = F,legend.title="CMMLplus")

p.c.ccus = min(p1$data.survplot$surv[p1$data.survplot$Treatment_arm == "Vitamin C"])
p.p.ccus = min(p1$data.survplot$surv[p1$data.survplot$Treatment_arm != "Vitamin C"])

p1$plot = p1$plot +theme(legend.position = "none") +
  annotate(geom="text",x = 6*365.25,y = p.c.ccus,label="Vitamin C", color=pal_nejm()(4)[c(3)],hjust=0) + 
  annotate(geom="text",x = 6*365.25,y = p.p.ccus,label="Placebo", color="black",hjust=0) + 
  scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+
  coord_cartesian(clip="off")+theme_prism()+theme(legend.position = "none")+
  geom_hline(yintercept = 0.5,linetype="dotted",linewidth=1.25) +theme(legend.position = "none",plot.margin = unit(c(0, 1, 0, 0), "inches")) + ggtitle("CCUS")


HR.ccus = summary(coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$Diagnosis_simple_combo=="CCUS",]))$conf.int
HR.ccus.anno = paste0("HR ",ifelse(round(HR.ccus[1],2)<0.01,"< 0.01",round(HR.ccus[1],2)), "\n95% CI ",round(HR.ccus[3],2),", ",round(HR.ccus[4],2))
p1$plot = p1$plot + annotate(geom="text", x = 365,y = 0.25, label=HR.ccus.anno,hjust=0,size=4.5)


p.c.ccus = min(p2$data.survplot$surv[p2$data.survplot$Treatment_arm == "Vitamin C"])
p.p.ccus = min(p2$data.survplot$surv[p2$data.survplot$Treatment_arm != "Vitamin C"])

p2$plot = p2$plot +theme(legend.position = "none") +
  annotate(geom="text",x = 6*365.25,y = p.c.ccus,label="Vitamin C", color=pal_nejm()(4)[c(3)],hjust=0) + 
  annotate(geom="text",x = 6*365.25,y = p.p.ccus,label="Placebo", color="black",hjust=0) + 
  scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+
  coord_cartesian(clip="off")+theme_prism()+theme(legend.position = "none")+
  geom_hline(yintercept = 0.5,linetype="dotted",linewidth=1.25) +theme(legend.position = "none",plot.margin = unit(c(0, 1, 0, 0), "inches"))+ ggtitle("MDS") 


HR.MDS = summary(coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$Diagnosis_simple_combo=="MDS",]))$conf.int
HR.MDS.anno = paste0("HR ",round(HR.MDS[1],2), "\n95% CI ",round(HR.MDS[3],2),", ",round(HR.MDS[4],2))
p3$plot = p3$plot + annotate(geom="text", x = 365,y = 0.25, label=HR.MDS.anno,hjust=0,size=4.5)


p.c.ccus = min(p3$data.survplot$surv[p3$data.survplot$Treatment_arm == "Vitamin C"])
p.p.ccus = min(p3$data.survplot$surv[p3$data.survplot$Treatment_arm != "Vitamin C"])

p3$plot = p3$plot +theme(legend.position = "none") +
  annotate(geom="text",x = 6*365.25,y = p.c.ccus,label="Vitamin C", color=pal_nejm()(4)[c(3)],hjust=0) + 
  annotate(geom="text",x = 5*365.25,y = p.p.ccus,label="Placebo", color="black",hjust=0) + 
  scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+
  coord_cartesian(clip="off")+theme_prism()+theme(legend.position = "none")+
  geom_hline(yintercept = 0.5,linetype="dotted",linewidth=1.25) +theme(legend.position = "none",plot.margin = unit(c(0, 1, 0, 0), "inches"))+ ggtitle("MDS/MPN") 


HR.mds.mpn = summary(coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$Diagnosis_simple_combo=="CMML_MDS_MPN",]))$conf.int

HR.mds.mpn.anno = paste0("HR ",round(HR.mds.mpn[1],2), "\n95% CI ",round(HR.mds.mpn[3],2),", ",round(HR.mds.mpn[4],2))
p2$plot = p2$plot + annotate(geom="text", x = 365,y = 0.25, label=HR.mds.mpn.anno,hjust=0,size=4.5)


p1$table = p1$table +theme_prism()+ theme(axis.line = element_blank())+scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+coord_cartesian(clip="off")+theme(text = element_text(face="bold",family=""),axis.text.y = element_text(hjust=1,colour = c(pal_nejm()(4)[c(3)],"black"),margin= margin(0, 15, 0, 7) ),axis.ticks.y = element_blank(),plot.margin = unit(c(0, 0, 0, 0), "inches"))+xlab("") + ggtitle("") +ylab("Number at risk\n(Cumulative censoring)") + theme(axis.text.x = element_blank(),axis.ticks.x = element_blank(),plot.title = element_blank())

p2$table = p2$table +theme_prism()+ theme(axis.line = element_blank())+scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+coord_cartesian(clip="off")+theme(text = element_text(face="bold",family=""),axis.text.y = element_text(hjust=1,colour = c(pal_nejm()(4)[c(3)],"black"),margin= margin(0, 15, 0, 7) ),axis.ticks.y = element_blank(),plot.margin = unit(c(0, 0, 0, 0), "inches"))+xlab("") + ggtitle("") +ylab("Number at risk\n(Cumulative censoring)") + theme(axis.text.x = element_blank(),axis.ticks.x = element_blank(),plot.title = element_blank())

p3$table = p3$table +theme_prism()+ theme(axis.line = element_blank())+scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+coord_cartesian(clip="off")+theme(text = element_text(face="bold",family=""),axis.text.y = element_text(hjust=1,colour = c(pal_nejm()(4)[c(3)],"black"),margin= margin(0, 15, 0, 7) ),axis.ticks.y = element_blank(),plot.margin = unit(c(0, 0, 0, 0), "inches"))+xlab("") + ggtitle("") +ylab("Number at risk\n(Cumulative censoring)") + theme(axis.text.x = element_blank(),axis.ticks.x = element_blank(),plot.title = element_blank())

p1$plot + p2$plot + p3$plot +
  p1$table + p2$table + p3$table + plot_layout(ncol=3,heights=c(1,0.4))

```


## Competing risk regression (progression)

For this analysis death is a competing risk of high-risk progression. We are a bit limited in sample size and events; there are only 14 progression events. Power to detect any differences in progression is quite low and we will have difficulty adjusting for any covariates here. We see corroboration with our overall survival findings that death (the dotted lines) are significantly different, however we do not have enough evidence to determine if progression also differs.


```{r,fig.width=16, fig.height=9}
#A lot of custom code to get this figure to match the other

#Start with competing risk event, 1 for progression, 2 for death, 0 for censor
df$crevent = ifelse(df$HR_FU_Progression == 1,1,ifelse(df$FU_Death==1,2,0))
df$crevent=factor(df$crevent,levels=c(0,1,2),labels=c("Censor","Higher-risk progression","Death"))

#Create months variable
df$Months = 12*df$HR_Days_FU_progression/365.25
#getting rid of space to prevent any issues in functions
df$tx = ifelse(df$Treatment_arm == "Vitamin C", "Vit_c","Placebo")

CI.tx = tidycmprsk::cuminc(Surv( Months ,crevent)~Treatment_arm,data=df)
CI.tx$cmprsk = timepoints(CI.tx$cmprsk,times=seq(0,60,12))

#Extract death and progression points
dth = tidycmprsk::crr(Surv( Months ,crevent)~VitaminC,data=df,failcode = "Death")
hrp = tidycmprsk::crr(Surv( Months ,crevent)~VitaminC,data=df,failcode = "Higher-risk progression")

p.ci = ggcuminc(CI.tx,outcome=c("Higher-risk progression","Death"),linewidth=1)+
  theme_prism()+scale_color_manual(values=c("black",pal_nejm()(4)[3])) +
  scale_y_continuous(limits=c(0,0.4)) +
  ##placebo annotations for labels on the RH of the plot
  annotate(geom="text",x = 65,y = max(CI.tx$tidy$estimate[CI.tx$tidy$strata == "Placebo" & CI.tx$tidy$outcome == "Death"],na.rm=T),label="Death, Placebo",hjust=0,size=4.5) +
  annotate(geom="text",size=4.5,x = 65,y = max(CI.tx$tidy$estimate[CI.tx$tidy$strata == "Placebo" & CI.tx$tidy$outcome == "Higher-risk progression"],na.rm=T)+0.005,label="Higher-risk progression, Placebo",hjust=0) +
  #The vitamin c labels at right
  annotate(geom="text",size=4.5,x = 65,y = max(CI.tx$tidy$estimate[CI.tx$tidy$strata == "Vitamin C" & CI.tx$tidy$outcome == "Death"],na.rm=T)-0.005,label="Death, Vitamin C",hjust=0,color=pal_nejm()(4)[3])+
  annotate(geom="text",size=4.5,x = 65,y = max(CI.tx$tidy$estimate[CI.tx$tidy$strata == "Vitamin C" & CI.tx$tidy$outcome == "Higher-risk progression"],na.rm=T),label="Higher-risk progression, Vitamin C",hjust=0,color=pal_nejm()(4)[3]) + 
  ggtitle("Competing risks analysis") +
  xlab("Months from randomisation") + 
  theme(legend.position = "none",plot.margin = unit(c(0, 2.75, 0, 0), "inches")) +
  scale_linetype_manual(values=c("dashed","solid","dashed","solid")) +
  #Add p-values, point estimates and 95% CIs
  annotate(geom="text", x = 4,y = 0.375, label=paste0("Higher-risk progression\np =  0.22\nHR ",round(exp(hrp$coefs),2),"; 95% CI ", round(exp(tidy(hrp,conf.int=T)$conf.low),2),", ", round(exp(tidy(hrp,conf.int=T)$conf.high),2) ),fontface="italic",hjust=0,size=4.5)+
  annotate(geom="text", x = 4,y = 0.29, label=paste0("Death\np =  ",tidy(dth,conf.int=T)$p.value,"\nHR ",round(exp(dth$coefs),2),"; 95% CI ", round(exp(tidy(dth,conf.int=T)$conf.low),2),", ", round(exp(tidy(dth,conf.int=T)$conf.high),2) ),fontface="italic",hjust=0,size=4.5) +
  scale_x_continuous(limits=c(0,65),breaks = seq(0,60,12),labels = seq(0,60,12),expand=c(0.0,0))+
  ylab("Cumulative incidence") +
  coord_cartesian(clip="off")

#Custom theme for only this risk table
my_theme = theme_prism()+ theme(axis.line = element_blank(),text = element_text(face="bold",family=""),axis.text.y = element_text(hjust=1,colour = c(pal_nejm()(4)[c(3)],"black"),margin= margin(0, 22, 0,16 ) ),axis.ticks.y = element_blank(),plot.margin = unit(c(0, 0, 0, 0), "inches"),axis.ticks.x = element_blank(),axis.title.x = element_blank(),plot.title=element_blank(),axis.text.x = element_blank())

p.ci = p.ci + add_risktable(times = seq(0,60,12),
                     risktable_stats = "{n.risk} ({cum.censor})",
                     risktable_group = "risktable_stats",
                     theme=my_theme,
                     size = 5) +
 theme(legend.position = "none",plot.margin = unit(c(0, 2.75, 0, 2), "inches"))

p.grobs = ggsurvfit_build(p.ci ,combine_plots = F)
## To get the formatting to match the KM plots, we need to manually modify the grobs, this doesn't impact the data, just alignments of labels
p.grobs[[2]]$layout$clip = rep("off")
p.grobs[[2]]$grobs[[13]][6]$children[[1]]$label = "Number at risk\n(Cumulative censoring)"

tbl.r = ggpubr::as_ggplot(p.grobs[[2]])

ggpubr::as_ggplot(p.grobs[[1]]) +tbl.r+plot_spacer()+plot_layout(ncol=1,heights=c(1,0.25,0.02))&coord_cartesian(clip="off")

ggsave("competing_risks.pdf",width=14,height=10)

```



## Global Methylation {.tabset}

**Figure XX. Global methylation changes over time by treatment arm** A) 5-mC/dG and B) 5-hmC/5-mC level for each patient at inclusion and end-of-therapy (EOT) and the change in these measures between inclusion and EOT. Data were analyzed using robust linear mixed-effects models with a random intercept for each patient and an interaction between treatment and time point to assess if measures changed differently over time. Models also included a covariate for batch.


```{r, fig.width=8, fig.height=5}
#Grab the plasma vitamin C data.frame and rename the columns for convenience
colnames(df.sp)[7:8] = c("Timepoint","Vitamin_C")

#Mass spec global methylation batch 1
df.hm1 = as.data.frame( read_excel(unzip("Vitc_Data.zip","EVI2_data_consolidated030524.xlsx"),sheet=2))
df.hm1 = df.hm1[,c(1,7,8,9:11)]
colnames(df.hm1) = c("ID", "mC","hmC","hmC/mC","mC/dG","hmC/dG")

#Mass spec global methylation batch 2
df.hm2 = as.data.frame( read_excel("Vitc_Data/MC00600_QuantCombined 112524 ST edits.xlsx",sheet=6,skip=1))
df.hm2 = df.hm2[,c(1,5,7,9:11)]
colnames(df.hm2) = c("ID", "hmC","mC","hmC/dG","mC/dG","hmC/mC")

#dummy variable for batch adjustment
df.hm1$Batch = 1
df.hm2$Batch = 2

## rbind the batches into one file
df.hm = rbind(df.hm1,df.hm2)

##These were mistakenly changed to 5 as EOT, they are not and we should model these as their respective times
df.hm$ID[match(c("1-5","12-5","33-5","25-5","44-5","78-5"),df.hm$ID )] = 
 c("1-2","12-2","33-2","25-2","44-2","78-4")

#These steps align IDs with main file, drop hyphen and leading 0s
df.hm$Time = as.numeric(gsub(".*-","",df.hm$ID))
df.hm$Time = factor(df.hm$Time,levels=seq(1:5),labels=c(0,3,6,9,12))
df.hm$ID = gsub("AAL0","AAL",gsub("-.*","",df.hm$ID))
df.sp$ID = gsub("AAL0","AAL",gsub("-","",df.sp$ID))

#Merge global methylation with plasma vitamin C info
df.merge = left_join(df.hm,df.sp,by= c("ID","Time"))
colnames(df.merge) = make.names(colnames(df.merge))
df.merge= df.merge[!is.na(df.merge$Time),]
df.merge$Timepoint  = ifelse(df.merge$Time == 0, "Baseline","EOT")

knitr::kable(unique(lof.tet2$Patient_ID[which(! lof.tet2$Patient_ID %in% df.merge$ID)]), caption = "Methylation IDs not in Mass Spec data")

```


```{r, fig.width=8, fig.height=5}
#This is for sever deficiency
svr = df$ID[df$VitC_PB1 <=11.4]
df$vitc_group =  case_when(df$VitC_PB1 <23 ~ "Deficient", 
                            df$VitC_PB1 <50 ~ "Inadequate",
                            TRUE ~ "Adequate")

###function for all subgroups 
subgrp.glbl = function(outcome,subgroup){
  #Create and end of therapy data frame and baseline, then merge them together
  EOT = subset(df.merge[df.merge$Timepoint == "EOT",], select=c(outcome,"Treatment_arm","ID","Sex","Vitamin_C"))
  Base = subset(df.merge[df.merge$Timepoint == "Baseline",], select=c(outcome,"Treatment_arm","ID","Sex","Vitamin_C"))
  colnames(EOT)[1] = "EOT"
  colnames(Base)[1] = "Base"
  
  colnames(EOT)[5] = "EOT.Vitc"
  colnames(Base)[5] = "Base.Vitc"
  
  wide = left_join(EOT,Base)
  
  # This section is formatting labels
  if(subgroup == "TET2"){
    wide$grp = ifelse( wide$ID %in% mut_ids,"TET2 or IDH1/2 mutation","No TET2 or IDH1/2 mutation")
    cols = c("black","#6F99ADFF")
    
  } 
  
  if(subgroup == "lof.TET2"){
      wide = wide[wide$ID %in% lof.tet2$Patient_ID,]
      wide$grp = ifelse( wide$ID %in% lof.tet2$Patient_ID[lof.tet2$TET2_Meth_Signature == "Low" | is.na(lof.tet2$TET2_Meth_Signature) ] ,"No or Low", ifelse(wide$ID %in% lof.tet2$Patient_ID[lof.tet2$TET2_Meth_Signature == "Intermediate"],"Intermediate","High")) 
      wide=wide[wide$grp!="Intermediate",]
      wide$grp = factor(wide$grp,levels=c("No or Low","High"),
                        labels = c("TET2 WT signature","TET2 hypermethylation signature"))
    cols = c("black","#6F99ADFF")
  } 
  

  if(subgroup == "vitc_med"){
    wide$grp = ifelse(wide$ID %in% df$ID[df$med_vitc == "above"],"Above median","Below median")  
    wide$grp = factor(wide$grp,levels=c("Below median","Above median"))
    cols = c("#FFDC91FF","#E18727FF")
  }
  
  if(subgroup == "vitc_severe"){
    wide$grp = ifelse(wide$ID %in% svr ,"Severe (< 11.4 Î¼mol/L)","All others")  
    wide$grp = factor(wide$grp,levels=c("Severe (< 11.4 Î¼mol/L)","All others"))
    cols = c("grey","#E18727FF")
  }
  
  if(subgroup == "vitc_ccus"){
        wide$grp = ifelse(wide$ID %in% df$ID[df$Diagnosis_simple_combo == "CCUS"],"CCUS",
                                         ifelse(wide$ID %in% df$ID[df$Diagnosis_simple_combo == "MDS"],"MDS",
                                         ifelse(wide$ID %in% df$ID[df$Diagnosis_simple_combo == "CMML_MDS_MPN"],"MDS/MPN",NA)))
    wide=wide[!is.na(wide$grp),]    
    wide$grp = factor(wide$grp,
                      levels= c("CCUS","MDS","MDS/MPN" ))
    
        cols = values=c("CCUS" =  viridis::viridis(4)[1] ,
                        "MDS" = viridis::viridis(4)[2],
                        "MDS/MPN" = viridis::viridis(4)[3])
    
  }
  
  # autoplot(lm(as.numeric( EOT.mC.dG) - as.numeric(Base.mC.dG ) ~ Treatment_arm*TET2,  data=mc.dg.wide),1:6)
  
  if(subgroup=="All"){
    wide$grp="All"
  }
  #Print frequnecy table for accounting purposes
  print(table(wide$Treatment_arm,wide$grp))

  yl = case_when(outcome == "mC.dG" ~ "5-mC/dG",
                   outcome == "hmC.dG" ~ "5-hmC/dG",
                   outcome == "hmC.mC"~"5-hmC/5-mC")
    
  if(subgroup == "All"){
        p.c.b = ggplot(wide,aes(x = Base.Vitc,y = as.numeric(Base),color=Treatment_arm,fill = Treatment_arm )) + 
      theme_minimal()+
      geom_point(alpha=0.5) +
      stat_smooth(method="lm") + 
      stat_cor(method = "spearman",p.accuracy = 0.0001, r.accuracy = 0.001,
               aes(label = paste(..rr.label.., ..p.label.., sep = "~`,`~"))) +
      theme_prism() + scale_color_manual(values=c("black","#E18727FF"))+theme(legend.position="none")+xlab("")+scale_fill_manual(values=c("black","#E18727FF"))+
        ylab(paste0("Baseline ",yl)) +xlab("Baseline Vitamin C")
        print(p.c.b)
        
  } else {
      p.c.b = ggplot(wide,aes(x = Base.Vitc,y = as.numeric(Base),color=Treatment_arm,fill = Treatment_arm )) + 
      theme_minimal()+facet_wrap(~grp)+
      geom_point(alpha=0.5) +
      stat_smooth(method="lm") + 
      stat_cor(method = "spearman", r.accuracy = 0.001,
               aes(label = paste(..rr.label.., sep = "~`,`~"))) +
      theme_prism() + scale_color_manual(values=c("black","#E18727FF"))+theme(legend.position="none")+xlab("")+scale_fill_manual(values=c("black","#E18727FF"))+
        ylab(paste0("Baseline ",yl)) +xlab("Baseline Vitamin C")
        print(p.c.b)
        
  }    
}

## for manuscript figures
delta.only = function(outcome,subgroup){
  EOT = subset(df.merge[df.merge$Timepoint == "EOT",], select=c(outcome,"Treatment_arm","ID","Sex","Vitamin_C"))
  Base = subset(df.merge[df.merge$Timepoint == "Baseline",], select=c(outcome,"Treatment_arm","ID","Sex","Vitamin_C"))
  colnames(EOT)[1] = "EOT"
  colnames(Base)[1] = "Base"
  
  colnames(EOT)[5] = "EOT.Vitc"
  colnames(Base)[5] = "Base.Vitc"
  
  wide = left_join(EOT,Base)
  
  if(subgroup == "TET2"){
    wide$grp = ifelse( wide$ID %in% mut_ids,"TET2 or IDH1/2 mutant","TET2 or IDH1/2 wild-type")
    cols = c("black","#6F99ADFF")
    
  } 
  
  if(subgroup == "lof.TET2"){
      wide = wide[wide$ID %in% lof.tet2$Patient_ID,]
      wide$grp = ifelse( wide$ID %in% lof.tet2$Patient_ID[lof.tet2$TET2_Meth_Signature == "Low" | is.na(lof.tet2$TET2_Meth_Signature) ] ,"No or Low", ifelse(wide$ID %in% lof.tet2$Patient_ID[lof.tet2$TET2_Meth_Signature == "Intermediate"],"Intermediate","High")) 
      wide=wide[wide$grp!="Intermediate",]
      wide$grp = factor(wide$grp,levels=c("No or Low","High"),
                        labels = c("TET2 WT signature","TET2 hypermethylation signature"))
    cols = c("black","#6F99ADFF")
  } 
  
  if(subgroup == "vitc_med"){
    wide$grp = ifelse(wide$ID %in% df$ID[df$med_vitc == "above"],"Above median","Below median")  
    wide$grp = factor(wide$grp,levels=c("Below median","Above median"))
    cols = c("#FFDC91FF","#E18727FF")
  }
  
  if(subgroup == "vitc_severe"){
    wide$grp = ifelse(wide$ID %in% svr ,"Severe (< 11.4 Î¼mol/L)","All others")  
    wide$grp = factor(wide$grp,levels=c("Severe (< 11.4 Î¼mol/L)","All others"))
    cols = c("grey","#E18727FF")
  }
  
  if(subgroup == "vitc_ccus"){
        wide$grp = ifelse(wide$ID %in% df$ID[df$Diagnosis_simple_combo == "CCUS"],"CCUS",
                                         ifelse(wide$ID %in% df$ID[df$Diagnosis_simple_combo == "MDS"],"MDS",
                                         ifelse(wide$ID %in% df$ID[df$Diagnosis_simple_combo == "CMML_MDS_MPN"],"MDS/MPN",NA)))
    wide=wide[!is.na(wide$grp),]    
    wide$grp = factor(wide$grp,
                      levels= c("CCUS","MDS","MDS/MPN" ))
    
        cols = values=c("CCUS" =  viridis::viridis(4)[1] ,
                        "MDS" = viridis::viridis(4)[2],
                        "MDS/MPN" = viridis::viridis(4)[3])
    
  }
  

  fit = rlm(as.numeric( EOT) - as.numeric(Base) ~ Treatment_arm*grp ,  data=wide)
  
  # print(summary(fit))
  res = data.frame(summary(confint(emmeans(fit,pairwise~Treatment_arm|grp)$contrasts)))
  yl = case_when(outcome == "mC.dG" ~ "5-mC/dG",
                 outcome == "hmC.dG" ~ "5-hmC/dG",
                 outcome == "hmC.mC"~"5-hmC/5-mC")
  
  ymin = signif(0.975*min(as.numeric(c(wide$Base,wide$EOT)),na.rm=T),2)
  ymax = signif(1.025*max(as.numeric(c(wide$Base,wide$EOT)),na.rm=T),2)
  
  ybar = signif(1.1*quantile(as.numeric(c(wide$Base,wide$EOT)),na.rm=T,p=0.975),2)
  ybar = signif(quantile(as.numeric(c(wide$Base,wide$EOT)),na.rm=T,p=0.975),2) + 0.1*(ymin-ymax)

  fit.a = rlm(rank(as.numeric(Base)) ~ grp+Treatment_arm ,  data=wide[wide$grp != "Intermediate",])
  p.1 = summary(emmeans(fit.a,pairwise~grp)$contrasts)
  # label1 = ifelse(p.1$p.value<0.0001, "p < 0.0001", paste0("p = ",signif(p.1$p.value,2)))
   ci.1 = confint(emmeans(fit.a,pairwise~grp)$contrasts)
    label1 = paste0("Mean \U0394 ",-1*signif(round(ci.1$estimate,4),2),"\n95% CI ",
                    -1*signif(round(ci.1$asymp.UCL,4),2),", ",
                    -1*signif(round(ci.1$asymp.LCL,4),2))
    
  fit.b = rlm(rank(as.numeric(EOT)) ~ grp+Treatment_arm ,  data=wide[wide$grp != "Intermediate",])
  p.2 = summary(emmeans(fit.b,pairwise~grp)$contrasts)
  # label2 = ifelse(p.2$p.value<0.0001, "p < 0.0001", paste0("p = ",signif(p.2$p.value,2)))
  ci.2 = confint(emmeans(fit.b,pairwise~grp)$contrasts)
  label2 = paste0("Mean \U0394 ",-1*signif(round(ci.2$estimate,4),2),"\n95% CI ",
                    -1*signif(round(ci.2$asymp.UCL,4),2),", ",
                    -1*signif(round(ci.2$asymp.LCL,4),2))
    
  p = ggplot(wide,aes(x = Treatment_arm,y =as.numeric(EOT) - as.numeric(Base),color=Treatment_arm )) + 
    facet_wrap(~grp,ncol=3) +theme_minimal()+
    geom_quasirandom() +
    geom_boxplot(fill=NA,outlier.size = -1) +
    theme_prism(axis_text_angle = 45) + scale_color_manual(values=c("black","#E18727FF"))+theme(legend.position="none")+xlab("")+
    ylab(paste0("Delta ",yl))
  
  
  res.p = res
  res.p$Treatment_arm = "Placebo"
  
  res.c = res
  res.c$Treatment_arm = "Vitamin C"
  res.sig = rbind(res.p,res.c)  
  value = max(as.numeric(wide$EOT) - as.numeric(wide$Base),na.rm=T)
  res.sig$x = as.numeric(as.factor(res.sig$Treatment_arm))
  # res$pv = paste0("p = ",signif(res$p.value,2))
  res$CI = paste0("Mean \U0394 ",-1*signif(round(res$estimate,4),2),"\n95% CI ",
                    -1*signif(round(res$asymp.UCL,4),2),", ",
                    -1*signif(round(res$asymp.LCL,4),2))
  res$x = 1.5
  p = p + geom_line(data=res.sig,aes(x = x, y = value*1.1),color="black") +
    geom_text(data=res,aes(x = x, y= value*1.3 ,label = CI),fontface="italic",color="black")+geom_hline(yintercept=0,linetype="dotted",linewidth=0.75)
  
  return(p )
      
}

#This is for the FDR, to recreate these p-values the function can be rerun and the p-values printed. 
saved_pvals = data.frame(outcome = rep(c("mC.dG","hmC.mC"),each=5),
                         subgroup = rep(c("TET2","vitc_med",rep("vitc_ccus",3) ),2),
                         p = c(0.1991,
                               0.1066,
                               0.0078, 0.6186, 0.0797,
                               0.00000008861223,
                               0.5790,
                               0.7838, 0.1405, 0.0837
                               ))

saved_pvals$FDR = p.adjust(saved_pvals$p,method="BH")

#manuscript figures
base.only = function(outcome,subgroup){
  EOT = subset(df.merge[df.merge$Timepoint == "EOT",], select=c(outcome,"Treatment_arm","ID","Sex","Vitamin_C"))
  Base = subset(df.merge[df.merge$Timepoint == "Baseline",], select=c(outcome,"Treatment_arm","ID","Sex","Vitamin_C"))
  colnames(EOT)[1] = "EOT"
  colnames(Base)[1] = "Base"
  
  colnames(EOT)[5] = "EOT.Vitc"
  colnames(Base)[5] = "Base.Vitc"
  
  wide = left_join(EOT,Base)
  
  if(subgroup == "TET2"){
    wide$grp = ifelse( wide$ID %in% mut_ids,"TET2 or IDH1/2 mutant","TET2 or IDH1/2 wild-type")
    cols = c("black","#6F99ADFF")
    
  } 

  if(subgroup == "vitc_med"){
    wide$grp = ifelse(wide$ID %in% df$ID[df$med_vitc == "above"],"Above median","Below median")  
    wide$grp = factor(wide$grp,levels=c("Below median","Above median"))
    cols = c("#FFDC91FF","#E18727FF")
  }
  

  if(subgroup == "vitc_ccus"){
        wide$grp = ifelse(wide$ID %in% df$ID[df$Diagnosis_simple_combo == "CCUS"],"CCUS",
                                         ifelse(wide$ID %in% df$ID[df$Diagnosis_simple_combo == "MDS"],"MDS",
                                         ifelse(wide$ID %in% df$ID[df$Diagnosis_simple_combo == "CMML_MDS_MPN"],"MDS/MPN",NA)))
    wide=wide[!is.na(wide$grp),]    
    wide$grp = factor(wide$grp,
                      levels= c("CCUS","MDS","MDS/MPN" ))
    
        cols = values=c("CCUS" =  viridis::viridis(4)[1] ,
                        "MDS" = viridis::viridis(4)[2],
                        "MDS/MPN" = viridis::viridis(4)[3])
    
  }
  
  

  fit = rlm(as.numeric( EOT) - as.numeric(Base) ~ Treatment_arm*grp ,  data=wide)
  res = data.frame(summary(emmeans(fit,pairwise~Treatment_arm|grp)$contrasts))
  yl = case_when(outcome == "mC.dG" ~ "5-mC/dG",
                 outcome == "hmC.dG" ~ "5-hmC/dG",
                 outcome == "hmC.mC"~"5-hmC/5-mC")
  
  ymin = signif(0.975*min(as.numeric(c(wide$Base,wide$EOT)),na.rm=T),2)
  ymax = signif(1.025*max(as.numeric(c(wide$Base,wide$EOT)),na.rm=T),2)
  
  ybar = max(as.numeric(wide$Base),na.rm=T)
  
  fit.a = rlm(as.numeric(Base) ~ grp+Treatment_arm ,  data=wide[wide$grp != "Intermediate",])
  knitr::kable(joint_tests(fit.a),caption="joint tests p should match table above")
  #p-vals are getting FDR adjusted across all subgroups
  p.1 = summary(emmeans(fit.a,pairwise~grp)$contrasts,adjust="none")
  knitr::kable(p.1,caption="emmeans and contrasts")
  p.1 = merge(p.1,summary(confint(emmeans(fit.a,pairwise~grp)$contrasts,adjust="none")))
  label1 = ifelse(saved_pvals[saved_pvals$outcome == outcome & saved_pvals$subgroup==subgroup,]$FDR <0.0001, "FDR p < 0.0001", paste0("FDR p = ",signif(saved_pvals[saved_pvals$outcome == outcome & saved_pvals$subgroup==subgroup,]$FDR,2)))
  
    label1 = paste0("Mean \U0394 ",-1*signif(round(p.1$estimate,4),2),
                    ", ",label1,
                    "\n95% CI ",
                    -1*signif(round(p.1$asymp.UCL,4),2),", ",
                    -1*signif(round(p.1$asymp.LCL,4),2))
  
  p.b.a = ggplot(wide[wide$grp != "Intermediate",],aes(x = grp,y = as.numeric(Base),color=grp )) + 
    geom_quasirandom() +
    geom_boxplot(fill=NA,outlier.size = -1) +
    theme_prism(axis_text_angle = 45) + scale_color_manual(values=cols)+theme(legend.position="none")+xlab("")+
      ylab(paste0("Baseline ",yl)) 
  
  if(subgroup != "vitc_ccus"){
   p.b.a = p.b.a + annotate(geom = "text",x=1.5,y = ybar*1.14,label = label1,fontface="italic",size=5) + theme(axis.text.x = element_text(angle=40,hjust=1))+
    annotate(geom = "line", y= ybar*1.05, x=c(1,2)) +
     scale_y_continuous(limits=c(ymin,ymax*1.3))

  } else{
  p.b.a = p.b.a + 
      annotate(geom = "text",x=1.5,y = ybar*1.025,label = label1[1],fontface="italic",size=5) + theme(axis.text.x = element_text(angle=40,hjust=1))+
    annotate(geom = "line", y= ybar*0.95, x=c(1,1.9)) +
    
      annotate(geom = "text",x=2.5,y = ybar*1.025,label = label1[3],fontface="italic",size=5) + theme(axis.text.x = element_text(angle=40,hjust=1))+
    annotate(geom = "line", y= ybar*0.95, x=c(2.1,3)) +
    
      annotate(geom = "text",x=2,y = ybar*1.28,label = label1[2],fontface="italic",size=5) + theme(axis.text.x = element_text(angle=40,hjust=1))+
    annotate(geom = "line", y= ybar*1.2, x=c(1,3))
  }  

  return(p.b.a )
      
}

```

### Delta 5-mC and 5-hmC/5-mC {.tabset}
#### TET2

```{r,fig.width=14,fig.height=5.5}

cowplot::plot_grid(delta.only("mC.dG","TET2"), delta.only("hmC.mC","TET2"),ncol=2,
                   labels = c("a","b"), label_size = 15)


```


#### baseline vitamin C

```{r,fig.width=14,fig.height=5.5}

cowplot::plot_grid(delta.only("mC.dG","vitc_med"), delta.only("hmC.mC","vitc_med"),ncol=2,
                   labels = c("a","b"))


```

#### Diagnosis

```{r,fig.width=14,fig.height=5.5}

cowplot::plot_grid(delta.only("mC.dG","vitc_ccus"), delta.only("hmC.mC","vitc_ccus"),ncol=2,
                   labels = c("a","b"))


```


### Baseline 5-mC and 5-hmC/5-mC {.tabset}

The raw p-values should be printing in each section

```{r}

knitr::kable(saved_pvals,caption="Saved p-values and FDR adjustments")


```

#### TET2

```{r,fig.width=14,fig.height=6.5}

cowplot::plot_grid(base.only("mC.dG","TET2"), base.only("hmC.mC","TET2"),ncol=2,
                   labels = c("a","b"))


```


#### baseline vitamin C

```{r,fig.width=14,fig.height=6.5}

cowplot::plot_grid(base.only("mC.dG","vitc_med"), base.only("hmC.mC","vitc_med"),ncol=2,
                   labels = c("a","b"))


```

#### Diagnosis

```{r,fig.width=20,fig.height=6.5}

cowplot::plot_grid(base.only("mC.dG","vitc_ccus"), base.only("hmC.mC","vitc_ccus"),ncol=2,
                   labels = c("a","b"))

```


### 5-mC/dG {.tabset}

#### All corr

```{r, fig.width=7,fig.height=7}

subgrp.glbl("mC.dG","All")

```

#### TET2 

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("mC.dG","TET2")

```

#### LOF TET2 

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("mC.dG","lof.TET2")

```

#### Median Vitamin C

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("mC.dG","vitc_med")

```


#### Vitamin C Severe

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("mC.dG","vitc_severe")

```

#### Diagnosis (CCUS)

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("mC.dG","vitc_ccus")

```




### 5-hmC/dG {.tabset}

#### All corr

```{r, fig.width=7,fig.height=7}

subgrp.glbl("hmC.dG","All")

```

#### TET2 

```{r, fig.width=12.3, fig.height=7}
subgrp.glbl("hmC.dG","TET2")

```

#### LOF TET2 

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("hmC.dG","lof.TET2")

```

#### Median Vitamin C

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("hmC.dG","vitc_med")

```


#### Vitamin C Severe

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("hmC.dG","vitc_severe")

```


#### Diagnosis (CCUS)

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("hmC.dG","vitc_ccus")

```


### 5-hmC/5-mC {.tabset}

#### All corr

```{r, fig.width=7,fig.height=7}

subgrp.glbl("hmC.mC","All")

```

#### TET2 

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("hmC.mC","TET2")

```

#### LOF TET2 

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("hmC.mC","lof.TET2")

```

#### Median Vitamin C

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("hmC.mC","vitc_med")

```


#### Vitamin C Severe

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("hmC.mC","vitc_severe")

```


#### Diagnosis (CCUS)

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("hmC.mC","vitc_ccus")

```


### Manuscript Figure

```{r, fig.width=11, fig.height=5}

df.merge$TET2 = ifelse(df.merge$ID %in% mut_ids,1,0)

#### This section for mC/dG #####
##Split and merge grabbing mC.dG
EOT = subset(df.merge[df.merge$Timepoint == "EOT",], select=c(mC.dG,Treatment_arm,ID))
Base = subset(df.merge[df.merge$Timepoint == "Baseline",], select=c(mC.dG,Treatment_arm,ID))
colnames(EOT)[1] = "EOT.mC.dG"
colnames(Base)[1] = "Base.mC.dG"

mc.dg.wide = left_join(EOT,Base)

fit_5mc = rlmer(as.numeric(mC.dG)  ~ Treatment_arm*Timepoint + Batch+ (1|ID),data=df.merge)

##Custom function for rlmer CIs
confint.rlmerMod <- function(object,parm,level=0.95) {
     beta <- fixef(object)
     if (missing(parm)) parm <- names(beta)
     se <- sqrt(diag(vcov(object)))
     z <- qnorm((1+level)/2)
     ctab <- cbind(beta-z*se,beta+z*se)
     colnames(ctab) <- stats:::format_perc(c((1-level)/2,(1+level)/2),
                                           digits=3)
     return(ctab[parm,])
}


knitr::kable(cbind(summary(fit_5mc)$coeff,confint(fit_5mc)),caption = "rlmer estimates 5mC/dG")

knitr::kable(summary(emmeans(fit_5mc,pairwise~Timepoint|Treatment_arm)),caption = "Time comparisons by treatment 5mC/dG")

knitr::kable(joint_tests(fit_5mc),caption = "anova on 5mC/dG model")
df.merge$mC.dG = as.numeric(df.merge$mC.dG)
df.merge$hmC.mC = as.numeric(df.merge$hmC.mC)

table1(~mC.dG|Timepoint+Treatment_arm,data=df.merge,render.continuous=my.render.cont)

p =  round(joint_tests(fit_5mc)$p.value[4],2)
df.merge$Timepoint = factor(df.merge$Timepoint,levels=c("Baseline","EOT"))
p1.tp =  ggplot(df.merge,aes(x = Time,y=as.numeric(mC.dG),color=Treatment_arm)) + geom_point(alpha=0.25) +
  geom_line(aes(group=ID),alpha=0.25) + facet_wrap(~Treatment_arm) +
  theme_prism(axis_text_angle = 45) + scale_color_manual(values=c("black","#E18727FF"))+theme(legend.position="none")+xlab("")+ylab('5-mC/dG')

#One version of the plot
p1.mc = ggplot(df.merge,aes(x = Timepoint,y=as.numeric(mC.dG),color=Treatment_arm)) + geom_point(alpha=0.25) +
  geom_line(aes(group=ID),alpha=0.25) + facet_wrap(~Treatment_arm) +
  theme_prism(axis_text_angle = 45) + scale_color_manual(values=c("black","#E18727FF"))+theme(legend.position="none")+xlab("")+ylab('5-mC/dG')


CI = round(confint(fit_5mc)[5,],4)

label_5mc = paste0("Mean \U0394 ",round(summary(fit_5mc)$coefficient[5,1] ,4),", ",
                   "p =  ",p,"\n",
                   "95% CI ",CI[1],", ",CI[2])

p2.mc = ggplot(mc.dg.wide,aes(x = Treatment_arm, y = as.numeric(EOT.mC.dG) - as.numeric(Base.mC.dG ),color=Treatment_arm)) +
  geom_quasirandom() + geom_boxplot(fill=NA,outliers=F) +
  stat_smooth(method="rlm") +
  xlab("") +
  ylab("Delta 5-mC/dG\n(Post - Pre)")+
  theme_prism(axis_text_angle = 45)+theme(legend.title = element_blank())+stat_cor()+
  scale_color_manual(values=c("black","#E18727FF")) +
  annotate(geom = "line",x = c(1,2), y= c(0.016,0.016) ,color="black") +
  annotate(geom="text",x = 1.5,y = 0.019,label=label_5mc,fontface="italic")+theme(legend.position = "none")+geom_hline(yintercept=0,linetype="dotted")

#Keeping p2.mc for manuscript
pmc = (p2.mc)
#### End mC/dG #####

#### This section for hmC/mC #####
fit_5hmc.5mc = rlmer(as.numeric(hmC.mC)  ~ Treatment_arm*Timepoint + Batch + (1|ID),data=df.merge )

knitr::kable(cbind(summary(fit_5hmc.5mc)$coeff,confint(fit_5hmc.5mc)),caption = "rlmer estimates 5hmC/5mC")
knitr::kable(summary(emmeans(fit_5hmc.5mc,pairwise~Timepoint|Treatment_arm)),caption = "Time comparisons by treatment 5hmC/5mC")

knitr::kable(joint_tests(fit_5hmc.5mc),caption = "anova on 5hmC/5mC model")
table1(~hmC.mC|Timepoint+Treatment_arm,data=df.merge,render.continuous=my.render.cont)

p =  round(joint_tests(fit_5hmc.5mc)$p.value[4],2)

p1.tp.hmC.mC = ggplot(df.merge,aes(x = Time,y=as.numeric(hmC.mC),color=Treatment_arm)) + geom_point(alpha=0.25) +
  geom_line(aes(group=ID),alpha=0.25) + facet_wrap(~Treatment_arm) +
  theme_prism(axis_text_angle = 45) + scale_color_manual(values=c("black","#E18727FF"))+theme(legend.position="none")+xlab("")+ylab('5-mC/dG')


p1.hmC.mC = ggplot(df.merge,aes(x = Timepoint,y=as.numeric(hmC.mC),color=Treatment_arm)) + geom_point(alpha=0.25) +
  geom_line(aes(group=ID),alpha=0.25) + facet_wrap(~Treatment_arm) +
  theme_prism(axis_text_angle = 45) + scale_color_manual(values=c("black","#E18727FF"))+theme(legend.position="none")+xlab("")+ylab('5-hmC/5-mC')

#grabbing hmC/mC this time
EOT = subset(df.merge[df.merge$Timepoint == "EOT",], select=c(hmC.mC,Treatment_arm,ID))
Base = subset(df.merge[df.merge$Timepoint == "Baseline",], select=c(hmC.mC,Treatment_arm,ID))
colnames(EOT)[1] = "EOT.hmC.mC"
colnames(Base)[1] = "Base.hmC.mC"

hmC.mC.wide = left_join(EOT,Base)

CI = round(confint(fit_5hmc.5mc)[5,],4)

label_5hmc.5mc = paste0("Mean \U0394 ",round(summary(fit_5hmc.5mc)$coefficient[5,1] ,4),", ",
                        "p =  ",p,"\n",
                        "95% CI ",CI[1],", ",CI[2])

p2.hmC.mC = ggplot(hmC.mC.wide,aes(x = Treatment_arm, y = as.numeric(EOT.hmC.mC) - as.numeric(Base.hmC.mC ),color=Treatment_arm)) +
  geom_quasirandom() + geom_boxplot(fill=NA,outliers=F) +
  stat_smooth(method="rlm") +
  xlab("") +
  ylab("Delta 5-hmC/5-mC\n(Post - Pre)")+
  theme_prism(axis_text_angle = 45)+theme(legend.title = element_blank())+stat_cor()+
  scale_color_manual(values=c("black","#E18727FF")) +
  annotate(geom = "line",x = c(1,2), y= c(0.0067,0.0067) ,color="black") +
  annotate(geom="text",x = 1.5,y = 0.0067*1.1,label=label_5hmc.5mc,fontface="italic")+theme(legend.position = "none")+geom_hline(yintercept=0,linetype="dotted")

phmC.mC = (p2.hmC.mC)
### End hmC/mC ###

## Stitch together for manuscript
cowplot::plot_grid(pmc ,phmC.mC,
                     labels=c("a","b"),ncol=2,label_size = 15)

ggsave("global_methylation_figure.pdf",width=12,height=6,device=cairo_pdf)


## Stitch together for manuscript
cowplot::plot_grid(pmc,
                     labels=c("a"),ncol=1,label_size = 15)

ggsave("global_methylation_figure_a.pdf",width=6,height=6,device=cairo_pdf)

cowplot::plot_grid(phmC.mC,
                     labels=c("b"),ncol=1,label_size = 15)

ggsave("global_methylation_figure_b.pdf",width=6,height=6,device=cairo_pdf)


```


## Cytokines {.tabset}
### Primary Analysis, all patients {.tabset}

#### Suppl all cytokines

**Supplemental Figure XX. log2 fold-change in cytokine measures for all individuals.** Data were analyzed using robust linear mixed-effects models with a random intercept for each patient and an interaction between treatment and timepoint to assess if measures changed differently over time. 


```{r, fig.height=11,fig.width=11}

#Read in cytokine data
cyto = read.table(unzip("Vitc_Data.zip","20240417_EVI_MSD.tsv"),sep="\t",header=T)

#Same idea as before, align ID variables
cyto$ID = gsub("-.*","",cyto$Source_Tube_ID)
cyto$ID[which(grepl("USC",fixed=T,cyto$Source_Tube_ID))] = substr(cyto$Source_Tube_ID[which(grepl("USC",fixed=T,cyto$Source_Tube_ID))],1,6)
cyto$ID[which(grepl("AAL",fixed=T,cyto$Source_Tube_ID))] = substr(cyto$Source_Tube_ID[which(grepl("AAL",fixed=T,cyto$Source_Tube_ID))],1,6)

cyto$ID = sapply(strsplit(cyto$ID,"-"), function(x){ifelse(length(x) >1, paste0(x[[1]],as.numeric(x[[2]])),x)})

# unique(cyto$ID)

# Create time variable from tube ID
cyto$Time = as.numeric(substr(gsub(" .*","",gsub(".*-","",cyto$Source_Tube_ID)),1,1))
cyto$Timepoint = ifelse(cyto$Time ==1,"Baseline","EOT")
cyto = cyto[order(cyto$Time,cyto$ID),]

#Merge with main file
cyto.m = merge(cyto,subset(df,select=c(ID, Treatment_arm,Age_inclusion_years, Sex,CCUS,Hb1)))

#Average the tech reps
agg = aggregate(calc_conc_imputed~Timepoint +ID+assay+Treatment_arm,cyto.m,function(x) {mean(x,na.rm=T)})

#Subset
agg.sub = subset(agg,select=c(ID,calc_conc_imputed,assay,Treatment_arm,Timepoint))

#Make wide data frame
cyto.wide = reshape2::dcast(agg.sub, ID + Timepoint +Treatment_arm  ~ assay,value.var="calc_conc_imputed")


#These will hold the various results
grps=NULL
diff=NULL
time=NULL

#For each marker analyze and save results
for(i in unique(cyto.m$assay)){
  #Do markers change differently over time by treatment, this includes the tech reps
  fit = rlmer(log2(calc_conc_imputed) ~ Timepoint*Treatment_arm  +(1|ID),data=cyto.m[cyto.m$assay==i,])
  
  ## Do treatments differ at timepoints
  grps = rbind(grps,data.frame(assay = i,summary(emmeans(fit,pairwise~Treatment_arm|Timepoint)$contrasts)))
  
  ## Did either treatment change over time
  time = rbind(time,data.frame(assay = i,summary(emmeans(fit,pairwise~Timepoint|Treatment_arm)$contrasts)))

  ## This is the actual p-value for"changed differently over time
  diff = rbind(diff,data.frame(assay = i,p =  emmeans::joint_tests(fit)[3,6],est = -1*summary(fit)$coeff[4,1],se = summary(fit)$coeff[4,2],lower.CL = -1*confint(fit)[4,2],upper.CL=-1*confint(fit)[4,1]))

}

diff##FDR adjustments
grps$FDR = p.adjust(grps$p.value,method="BH")
time$FDR = p.adjust(time$p.value,method="BH")

#Get fold change and flip the reference 
time$FoldChange = 1/(2^time$estimate)

knitr::kable(time,caption = "Significant changes over time by treatment")

diff$FDR = p.adjust(diff$p,method="BH")

knitr::kable(diff[diff$FDR < 0.05,],caption= "Cytokines that changed differently over time for Placebo versus Vitamin C")

#A data frame of only markers that changed over time
sig.t = time[time$FDR<0.05,]

#We are likely going to use actual p-values, but if we ever use stars, we'll have them
sig.t$Label = case_when(sig.t$FDR<0.0001 ~"***",
                        sig.t$FDR<0.01 ~"**",
                        TRUE~"*")

## This is just to get a y-value for plot annotations
sig.t$y = rep(0)
for(i in 1:nrow(sig.t)){
  sig.t$y[i] = log2(max(agg$calc_conc_imputed[agg$assay ==sig.t$assay[i]],na.rm=T))
}

## factor our time variable for plotting
agg$Timepoint = factor(agg$Timepoint,levels=c("Baseline","EOT"))
##pre and post data.frame to make a wide data frame
agg.pr = agg[agg$Timepoint=="Baseline",]
agg.post = subset(agg[agg$Timepoint=="EOT",],select=c(calc_conc_imputed, assay,ID))
agg.w = left_join(agg.pr,agg.post,by=c("ID","assay"))

#delta on the log scales
agg.w$delta = log2(agg.w$calc_conc_imputed.y) - log2(agg.w$calc_conc_imputed.x)
diff$assay = gsub("_"," ",diff$assay)
sig.d=diff

sig.d$Label = ifelse(sig.d$FDR<0.0001 ,"FDR p < 0.0001",paste0("FDR p =  ",signif(round(sig.d$FDR,4),2)))

sig.d$Label_alt = paste0("Mean \U0394 ",signif(round(-1*sig.d$est,4),2),", ",
                    sig.d$Label,"\n",
                   "95% CI ",-1*signif(round(sig.d$upper.CL,4),2),", ",signif(round(-1*sig.d$lower.CL,4),2))

#Get y for p-value annotation
# sig.d$y = rep(0)
# for(i in 1:nrow(sig.d)){
#   sig.d$y[i] = max(agg.w$delta[agg.w$assay ==sig.d$assay[i]],na.rm=T)*1.275
# }

#Set common y for p-value annotation
sig.d$y = 5

#remove under scores and make spaces
agg.w$assay = gsub("_"," ",agg.w$assay)
time$assay = gsub("_"," ",time$assay)

```


```{r, fig.height=27,fig.width=17}

#Marker levels for factoring
lvls=c("IL-1Î±", "IL-1Î²", "IL-6", "TNF-Î±", "IL-12p70", "IL-17",
 "IL-4", "IL-10", "TGF-Î²1",
 "GM-CSF", "G-CSF", "M-CSF",
 "IL-8 HA", "IP-10", "RANTES", "SDF-1Î±",
 "IL-7", "IFN-Î³", "VEGF cyto")

#Updated labels for plotting
lbls = c("IL-1Î±", "IL-1Î²", "IL-6", "TNF-Î±", "IL-12p70", "IL-17A",
 "IL-4", "IL-10", "TGF-Î²1",
 "GM-CSF", "G-CSF", "M-CSF",
 "IL-8", "CXCL10 (IP-10)", "RANTES (CCL5)", "SDF-1Î±",
 "IL-7", "IFN-Î³", "VEGF")

#Factor for order and relabeling
sig.d$assay = factor(sig.d$assay,levels=lvls,labels=lbls)
time$assay = factor(time$assay,levels=lvls,labels=lbls)
agg.w$assay = factor(agg.w$assay,levels=lvls,labels=lbls)

#Grps we want to plot by
grps = list(pro.inflm = c("IL-1Î±", "IL-1Î²", "IL-6", "TNF-Î±", "IL-12p70", "IL-17A"),
ant.inflm = c("IL-4", "IL-10", "TGF-Î²1"),
m.grow = c("GM-CSF", "G-CSF", "M-CSF"),
chemok = c("IL-8", "CXCL10 (IP-10)", "RANTES (CCL5)", "SDF-1Î±"),
reg.cyt = c("IL-7", "IFN-Î³", "VEGF"))

grp.nms = c("Pro-inflammatory cytokines",
          "Anti-inflammatory cytokines",
          "Myeloid growth factors",
          "Chemokines",
          "Immune regulatory cytokines")

#Plot each group of markers
p=list()
for(i in 1:length(grps)){
  p[[i]] = ggplot(agg.w[agg.w$assay %in% grps[[i]],],aes(x = Treatment_arm, y = delta,color= Treatment_arm)) +
    facet_wrap2(vars(assay),scales="free",ncol=6, trim_blank = FALSE)+geom_hline(yintercept = 0,linewidth=1,color="grey",linetype="dashed") +
    theme_prism() + geom_quasirandom()+ geom_boxplot(fill=NA,outlier.size=-1)+ stat_smooth(method="lm",linewidth=1.25)+scale_color_manual(values=c("black","#E18727FF"))+theme(strip.background = element_blank())+ylab(expression(~ log2(frac("Cytokine concentration post","Cytokine concentration pre")))) +geom_text(data=sig.d[sig.d$assay %in% grps[[i]],],aes(x=1.5,y=y,label=Label_alt),color="black",size=4,fontface="italic")+geom_segment(data=sig.d[sig.d$assay %in% grps[[i]],],aes(x = 1,xend=2,y =0.8*y,yend=0.8*y ),color="black")+theme(legend.title = element_blank(),legend.position = "none")+xlab("")+
    ggtitle(grp.nms[i])
}

cowplot::plot_grid(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],ncol=1)

```



```{r, fig.height=11,fig.width=16}

ggplot(cyto.m,aes(x = Timepoint, y = log2(calc_conc_imputed), color=Treatment_arm)) + facet_wrap(~assay,scales="free") +
  geom_point(size=1.5)+
  geom_line(aes(group=ID),alpha=0.3,size=1) +
  theme_prism()+scale_color_manual(values=c("black","#E18727FF"))+theme(strip.background = element_blank())

```


#### Cytokine figure

**Supplemental Figure XX. log2 fold-change in cytokine measures that changed significantly.** Data were analyzed using robust linear mixed-effects models with a random intercept for each patient and an interaction between treatment and timepoint to assess if measures changed differently over time. Larger points indicate the mean log2(fold-change) with 95% CI error bars. Scales were limited to +/-1.5 log2(fold-change) to better characterize dispersion, all data points are shown in supplemental figure XX. P-values were false discovery rate adjusted using Benjamini-Hochberg. 


```{r,fig.height=6,fig.width=18}
#Grab only those that were significant
sig.d = sig.d[sig.d$FDR<0.05,]

#Factor
diff$assay = factor(gsub("_"," ",diff$assay),levels=lvls,labels=lbls)

#Create duplicates of these so can reference them again later
time.h = time
agg.w.h = agg.w

time = time[time$assay %in% diff$assay[diff$FDR<0.05 ],]
agg.w = agg.w[agg.w$assay %in% diff$assay[diff$FDR<0.05 ],]

#This is just for the annotation
sig.d$Treatment_arm = "Placebo"

##Individual deltas
agg.w$abs_delta =  agg.w$calc_conc_imputed.y - agg.w$calc_conc_imputed.x

time$estimate = time$estimate*-1

time$assay = factor(time$assay,levels=c("IL-6","IL-10","CXCL10 (IP-10)","M-CSF","G-CSF","RANTES (CCL5)"))
ggplot(time,aes(x = Treatment_arm, y = estimate,fill=Treatment_arm,color=Treatment_arm)) + geom_point(size=2.5)+ facet_wrap(~assay,scales="free_x",ncol=6)+geom_hline(yintercept = 0,linewidth=1,color="grey",linetype="dashed",linewidth=1.25) +
  theme_prism() + scale_fill_manual(values=c("black","#E18727FF"))+
  scale_color_manual(values=c("black","#E18727FF")) +theme(strip.background = element_blank(),plot.caption =  element_text(hjust = 0, margin = margin(0, 0, 0, 0, "pt")))+ylab(expression(log[2]~"[FC]"))+geom_text(data=sig.d,aes(x=1.5,y=1.5,label=Label_alt,group=1),fill="black",size=4,fontface="italic")+geom_segment(data=sig.d,aes(x = 1,xend=2,y =0.9*1.5,yend=0.9*1.5 ,group=1),color="black")+theme(legend.title = element_blank(),legend.position = "none")+xlab("") + geom_errorbar(aes(ymin = estimate - 1.96*SE,ymax= estimate + 1.96*SE),width=0.25,linewidth=0.75) +geom_quasirandom(data=agg.w,aes(x = Treatment_arm, y = delta,color=Treatment_arm),alpha=0.25)+
  coord_cartesian(ylim = c(-1.5,1.5))+scale_y_continuous(breaks=seq(-1.5,1.5,0.5))

ggsave("cytokine_figure.pdf",width=18,height=6, device=cairo_pdf)

```


```{r}

agg.df = aggregate(abs_delta ~ assay+Treatment_arm,agg.w,median)
colnames(agg.df)[3] = "Median change"
agg.df$IQR  = aggregate(abs_delta ~ assay+Treatment_arm,agg.w,IQR)$abs_delta
agg.df$Q1 = aggregate(abs_delta ~ assay+Treatment_arm,agg.w,function(x){quantile(x,p=0.25)})$abs_delta
agg.df$Q3 = aggregate(abs_delta ~ assay+Treatment_arm,agg.w,function(x){quantile(x,p=0.75)})$abs_delta
time$FoldChange = 2^(time$estimate)
time$FC.CI = paste0(round(2^(time$estimate - 1.96*time$SE),2),", ",round(2^(time$estimate + 1.96*time$SE),2))

cyto.res = merge(subset(time,  select=c(assay,Treatment_arm,FoldChange,FC.CI)),agg.df,by=c("assay","Treatment_arm"))
cyto.res = cyto.res[order(cyto.res$assay,cyto.res$Treatment_arm),]
cyto.res$assay = as.character(cyto.res$assay)
cyto.res$assay[duplicated( cyto.res$assay)] = ""
knitr::kable(cyto.res)

```



```{r}

#Create blank results data frames to collate the subgroup results into one long tabl;e
cyto.mm = merge(cyto.m,df,by=c("ID"))
cyto.sub=NULL

cyto.mm$Timepoint = factor(cyto.mm$Timepoint,levels=c("Baseline","EOT"))
cyto.time = NULL

for(i in unique(cyto.m$assay)){
  fit = rlmer(log2(calc_conc_imputed) ~ Timepoint*Treatment_arm.x +(1|ID),data=cyto.mm[cyto.mm$assay==i & cyto.mm$TET2 == "TET2neg",])
  cyto.sub = rbind(cyto.sub,data.frame(Subgroup =  "TET2 neg",assay = i,p =  emmeans::joint_tests(fit)[3,6],est=-1*summary(fit)$coeff[4,1],se = summary(fit)$coeff[4,2]))

  cyto.time = rbind(cyto.time,data.frame(assay = i,Subgroup =  "TET2 neg",summary(emmeans(fit,pairwise~Timepoint|Treatment_arm.x)$contrasts)))


}

for(i in unique(cyto.m$assay)){
  fit = rlmer(log2(calc_conc_imputed) ~ Timepoint*Treatment_arm.x +(1|ID),data=cyto.mm[cyto.mm$assay==i & cyto.mm$TET2 == "TET2Mut",])
  cyto.sub = rbind(cyto.sub,data.frame(Subgroup =  "TET2Mut",assay = i,p =  emmeans::joint_tests(fit)[3,6],est=-1*summary(fit)$coeff[4,1],se = summary(fit)$coeff[4,2]))

  cyto.time = rbind(cyto.time,data.frame(assay = i,Subgroup =  "TET2Mut",summary(emmeans(fit,pairwise~Timepoint|Treatment_arm.x)$contrasts)))

}

for(i in unique(cyto.m$assay)){
  fit = rlmer(log2(calc_conc_imputed) ~ Timepoint*Treatment_arm.x +(1|ID),data=cyto.mm[cyto.mm$assay==i & cyto.mm$med_vitc == "above",])
  cyto.sub = rbind(cyto.sub,data.frame(Subgroup =  "Above Med Vit C",assay = i,p =  emmeans::joint_tests(fit)[3,6],est=-1*summary(fit)$coeff[4,1],se = summary(fit)$coeff[4,2]))
  cyto.time = rbind(cyto.time,data.frame(assay = i,Subgroup =  "Above Med Vit C",summary(emmeans(fit,pairwise~Timepoint|Treatment_arm.x)$contrasts)))

}

for(i in unique(cyto.m$assay)){
  fit = rlmer(log2(calc_conc_imputed) ~ Timepoint*Treatment_arm.x +(1|ID),data=cyto.mm[cyto.mm$assay==i & cyto.mm$med_vitc == "below",])
  cyto.sub = rbind(cyto.sub,data.frame(Subgroup =  "below Med Vit C",assay = i,p =  emmeans::joint_tests(fit)[3,6],est=-1*summary(fit)$coeff[4,1],se = summary(fit)$coeff[4,2]))
  cyto.time = rbind(cyto.time,data.frame(assay = i,Subgroup =  "below Med Vit C",summary(emmeans(fit,pairwise~Timepoint|Treatment_arm.x)$contrasts)))

}

for(i in unique(cyto.m$assay)){
  fit = rlmer(log2(calc_conc_imputed) ~ Timepoint*Treatment_arm.x +(1|ID),data=cyto.mm[cyto.mm$assay==i & cyto.mm$Diagnosis_simple_combo == "CCUS",])
  cyto.sub = rbind(cyto.sub,data.frame(Subgroup =  "CCUS",assay = i,p =  emmeans::joint_tests(fit)[3,6],est=-1*summary(fit)$coeff[4,1],se = summary(fit)$coeff[4,2]))
  cyto.time = rbind(cyto.time,data.frame(assay = i,Subgroup =  "CCUS",summary(emmeans(fit,pairwise~Timepoint|Treatment_arm.x)$contrasts)))
}

for(i in unique(cyto.m$assay)){
  fit = rlmer(log2(calc_conc_imputed) ~ Timepoint*Treatment_arm.x +(1|ID),data=cyto.mm[cyto.mm$assay==i & cyto.mm$Diagnosis_simple_combo == "MDS",])
  cyto.sub = rbind(cyto.sub,data.frame(Subgroup =  "MDS",assay = i,p =  emmeans::joint_tests(fit)[3,6],est=-1*summary(fit)$coeff[4,1],se = summary(fit)$coeff[4,2]))
  cyto.time = rbind(cyto.time,data.frame(assay = i,Subgroup =  "MDS",summary(emmeans(fit,pairwise~Timepoint|Treatment_arm.x)$contrasts)))
}

for(i in unique(cyto.m$assay)){
  fit = rlmer(log2(calc_conc_imputed) ~ Timepoint*Treatment_arm.x +(1|ID),data=cyto.mm[cyto.mm$assay==i & cyto.mm$Diagnosis_simple_combo == "CMML_MDS_MPN",])
  cyto.sub = rbind(cyto.sub,data.frame(Subgroup =  "CMML_MDS_MPN",assay = i,p =  emmeans::joint_tests(fit)[3,6],est=-1*summary(fit)$coeff[4,1],se = summary(fit)$coeff[4,2]))
  cyto.time = rbind(cyto.time,data.frame(assay = i,Subgroup =  "CMML_MDS_MPN",summary(emmeans(fit,pairwise~Timepoint|Treatment_arm.x)$contrasts)))
}

#We aren't reporting these, but we'll calculate them in case any future readers want them
cyto.sub$FDR = p.adjust(cyto.sub$p,method="BH")

```


#### Volcano plots of each subroup by assay

```{r, fig.height=11,fig.width=11}

#Rename for consistency with main marker analysis
cyto.sub$assay[cyto.sub$assay == "IL-8_HA"] = "IL-8 HA"
cyto.sub$assay[cyto.sub$assay == "VEGF_cyto"] = "VEGF cyto"
cyto.sub$assay = factor(cyto.sub$assay,levels=lvls,labels=lbls)

cyto.time$assay[cyto.time$assay == "IL-8_HA"] = "IL-8 HA"
cyto.time$assay[cyto.time$assay == "VEGF_cyto"] = "VEGF cyto"
cyto.time$assay = factor(cyto.time$assay,levels=lvls,labels=lbls)

#We are removing these results from the main manuscript
cyto.sub = cyto.sub[!cyto.sub$Subgroup %in% c("Deficient","Inadequate","Adequate"),]

diff=subset(diff, select = - c(lower.CL,upper.CL))

#Add variable in original results call 'All' then rbind into the subgroups
diff$Subgroup =  "All"
cyto.sub = rbind(cyto.sub,diff)

#Factor and rename for plotting           
cyto.sub$Subgroup =  factor(cyto.sub$Subgroup,levels=c("All",
                                                      "TET2 neg",
                                                      "TET2Mut",
                                                      "below Med Vit C",
                                                      "Above Med Vit C",
                                                      "CCUS",
                                                      "MDS",
                                                      "CMML_MDS_MPN"),
                                             labels=c("All patients",
                                                    "No TET2 or IDH1/2 mutations",
                                                    "TET2 or IDH1/2 mutation",
                                                    "Below median vitamin C",
                                                    "Above median vitamin C",
                                                    "CCUS",
                                                    "MDS",
                                                    "MDS/MPN")
)

ggplot(cyto.sub, aes(x = est, y = -log10(FDR),color=Subgroup )) + 
        geom_point() + 
        theme_prism(axis_text_angle=45) +
        scale_color_viridis_d(option="turbo")+    
        xlab(expression(log[2]~"[FC]")) +
  facet_wrap(~assay,scales = "free") +
  scale_x_continuous(limits=c(-1,1)) + theme(legend.position = "bottom")+ geom_hline(aes(yintercept = 1.30,linetype = "FDR < 0.05"),color="red",alpha=0.25) + 
  scale_linetype_manual(values=c(2)) +
  scale_y_continuous(limits=c(0,5))

```


#### Subgroup plots {.tabset}

```{r}

cyto.time$Subgroup =  factor(cyto.time$Subgroup,levels=rev(c("All",
                                                      "TET2 neg",
                                                      "TET2Mut",
                                                      "below Med Vit C",
                                                      "Above Med Vit C",
                                                      "CCUS",
                                                      "MDS",
                                                      "CMML_MDS_MPN")),
                                             labels=rev(c("All patients",
                                                    "No TET2 or IDH1/2 mutations",
                                                    "TET2 or IDH1/2 mutation",
                                                    "Below median vitamin C",
                                                    "Above median vitamin C",
                                                    "CCUS",
                                                    "MDS",
                                                    "MDS/MPN"))
)

plot.cyto.sub = function(subgroup){
  
  assays = unique(cyto.sub$assay[cyto.sub$Subgroup %in% subgroup & cyto.sub$FDR<0.05])
  time = cyto.time[cyto.time$assay %in% assays & cyto.time$Subgroup %in% subgroup,]
  agg.w = agg.w.h[agg.w.h$assay %in% assays,]
  agg.w$assay = factor(agg.w$assay,levels = unique(agg.w$assay))
  sig.d = cyto.sub[cyto.sub$Subgroup %in% subgroup & cyto.sub$assay %in% assays,]
  sig.d$Treatment_arm = rep("Placebo")
  sig.d$Label = ifelse(sig.d$FDR<0.0001 ,"FDR p < 0.0001",paste0("FDR p =  ",signif(round(sig.d$FDR,4),2)))
  sig.d = sig.d[sig.d$FDR < 0.1,]
  agg.w$abs_delta =  agg.w$calc_conc_imputed.y - agg.w$calc_conc_imputed.x
  
  time$estimate = time$est*-1
  
  colnames(time)[which(colnames(time) == "Treatment_arm.x")] = "Treatment_arm"
  
  if("No TET2 or IDH1/2 mutations" %in% subgroup ){
    agg.w = left_join(agg.w,subset(df,select=c(ID,TET2)),by="ID")
  
    agg.w$Subgroup = factor(agg.w$TET2,levels=c("TET2neg","TET2Mut"),
                                               labels=c("No TET2 or IDH1/2 mutations",
                                                      "TET2 or IDH1/2 mutation"))
  }
  
  if("CCUS" %in% subgroup ){
    agg.w = left_join(agg.w,subset(df,select=c(ID,Diagnosis_simple_combo)),by="ID")
    agg.w = agg.w[!is.na(agg.w$Diagnosis_simple_combo),]
    agg.w$Subgroup = factor(agg.w$Diagnosis_simple_combo,levels=c("CCUS",
                                                      "MDS",
                                                      "CMML_MDS_MPN"),
                                               labels=c("CCUS",
                                                    "MDS",
                                                    "MDS/MPN"))
  }
  
  if("Below median vitamin C" %in% subgroup ){
    agg.w = left_join(agg.w,subset(df,select=c(ID,med_vitc)),by="ID")
    agg.w$Subgroup = factor(agg.w$med_vitc,levels=c("below", "above"),
                                               labels=c("Below median vitamin C",
                                                    "Above median vitamin C"))
  }
  
  print(table(agg.w$Treatment_arm,agg.w$assay,agg.w$Subgroup))
  
 
  p = ggplot(time,aes(x = Treatment_arm, y = estimate,fill=Treatment_arm,color=Treatment_arm)) + geom_point(size=2.5)+
    facet_nested_wrap(vars(Subgroup,assay),scales="free_x",ncol=length(assays),trim_blank = FALSE)+
      geom_hline(yintercept = 0,linewidth=1,color="grey",linetype="dashed",linewidth=1.25) +
    theme_prism() + scale_fill_manual(values=c("black","#E18727FF"))+
    scale_color_manual(values=c("black","#E18727FF")) +theme(strip.background = element_blank(),plot.caption =  element_text(hjust = 0, margin = margin(0, 0, 0, 0, "pt")))+ylab(expression(log[2]~"[FC]")) +
    theme(legend.title = element_blank(),legend.position = "none")+
    xlab("") +
    geom_errorbar(aes(ymin = estimate - 1.96*SE,ymax= estimate + 1.96*SE),width=0.25,linewidth=0.75) +
    geom_quasirandom(data=agg.w,aes(x = Treatment_arm, y = delta,color=Treatment_arm),alpha=0.25) + coord_cartesian(ylim=c(-1.5,1.5))

  return(p)

}


plot.cyto.sub = function(subgroup){
  
  assays = unique(cyto.sub$assay[cyto.sub$Subgroup %in% subgroup & cyto.sub$FDR<0.05])
  time = cyto.time[cyto.time$assay %in% assays & cyto.time$Subgroup %in% subgroup,]
  agg.w = agg.w.h[agg.w.h$assay %in% assays,]
  agg.w$assay = factor(agg.w$assay,levels = unique(agg.w$assay))
  sig.d = cyto.sub[cyto.sub$Subgroup %in% subgroup & cyto.sub$assay %in% assays,]
  sig.d$Treatment_arm = rep("Placebo")
  sig.d$Label = ifelse(sig.d$FDR<0.0001 ,"FDR p < 0.0001",paste0("FDR p =  ",signif(round(sig.d$FDR,4),2)))
  sig.d = sig.d[sig.d$FDR < 0.1,]
  agg.w$abs_delta =  agg.w$calc_conc_imputed.y - agg.w$calc_conc_imputed.x
  
  time$estimate = time$est*-1
  
  colnames(time)[which(colnames(time) == "Treatment_arm.x")] = "Treatment_arm"
  
  if("No TET2 or IDH1/2 mutations" %in% subgroup ){
    agg.w = left_join(agg.w,subset(df,select=c(ID,TET2)),by="ID")
  
    agg.w$Subgroup = factor(agg.w$TET2,levels=c("TET2neg","TET2Mut"),
                                               labels=c("No TET2 or IDH1/2 mutations",
                                                      "TET2 or IDH1/2 mutation"))
  }
  
  if("CCUS" %in% subgroup ){
    agg.w = left_join(agg.w,subset(df,select=c(ID,Diagnosis_simple_combo)),by="ID")
    agg.w = agg.w[!is.na(agg.w$Diagnosis_simple_combo),]
    agg.w$Subgroup = factor(agg.w$Diagnosis_simple_combo,levels=c("CCUS",
                                                      "MDS",
                                                      "CMML_MDS_MPN"),
                                               labels=c("CCUS",
                                                    "MDS",
                                                    "MDS/MPN"))
  }
  
  if("Below median vitamin C" %in% subgroup ){
    agg.w = left_join(agg.w,subset(df,select=c(ID,med_vitc)),by="ID")
    agg.w$Subgroup = factor(agg.w$med_vitc,levels=c("below", "above"),
                                               labels=c("Below median vitamin C",
                                                    "Above median vitamin C"))
  }
  
  print(knitr::kable(table(agg.w$Treatment_arm,agg.w$assay,agg.w$Subgroup)))
  
 
  p = ggplot(time,aes(x = Treatment_arm, y = estimate,fill=Treatment_arm,color=Treatment_arm)) + geom_point(size=2.5)+
    facet_nested_wrap(vars(Subgroup,assay),scales="free_x",ncol=length(assays),trim_blank = FALSE)+
      geom_hline(yintercept = 0,linewidth=1,color="grey",linetype="dashed",linewidth=1.25) +
    theme_prism() + scale_fill_manual(values=c("black","#E18727FF"))+
    scale_color_manual(values=c("black","#E18727FF")) +theme(strip.background = element_blank(),plot.caption =  element_text(hjust = 0, margin = margin(0, 0, 0, 0, "pt")))+ylab(expression(log[2]~"[FC]")) +
    # geom_text(data=sig.d,aes(x=1.5,y=1.5,label=Label,group=1),fill="black",size=4,fontface="italic")+geom_segment(data=sig.d,aes(x = 1,xend=2,y =0.925*1.5,yend=0.925*1.5 ,group=1),color="black")+
    theme(legend.title = element_blank(),legend.position = "none")+
    xlab("") +
    geom_errorbar(aes(ymin = estimate - 1.96*SE,ymax= estimate + 1.96*SE),width=0.25,linewidth=0.75) +
    geom_quasirandom(data=agg.w,aes(x = Treatment_arm, y = delta,color=Treatment_arm),alpha=0.25) + coord_cartesian(ylim=c(-1.5,1.5))

  return(p)
  
  
}

```


##### All subgroups

95% CIs for subgroup comparisons

```{r, fig.height=18,fig.width=18}

p=list()
for(i in 1:length(grps)){
  
p[[i]] = ggplot(cyto.sub[cyto.sub$assay  %in% grps[[i]],],aes(x = -1*est , y = Subgroup)) +
  geom_vline(xintercept=0,color="red",linetype="dotted",size=0.75) +
  geom_point(size=2) + facet_wrap2(vars(assay),scales="free_x",ncol=6, trim_blank = FALSE,remove_labels = "all")+
  theme_minimal(15) + geom_errorbar(aes(xmin = -1*(est+1.96*se),xmax=-1*(est-1.96*se)),width=0.15)+
  xlab("Delta-delta on log2 transformed cytokines")+ theme(legend.title = element_blank(),legend.position = "none")+ylab("")+
    ggtitle(grp.nms[i]) 
  
    
  
}

 
cowplot::plot_grid(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],ncol=1)

```



##### TET2

Showing only those with FDR < 0.1

```{r, fig.width = 16,fig.height=8}

plot.cyto.sub(c("No TET2 or IDH1/2 mutations","TET2 or IDH1/2 mutation"))
```


##### Median Vitamin C

```{r, fig.width = 16,fig.height=8}

plot.cyto.sub(c("Below median vitamin C","Above median vitamin C"))

```

##### Diagnosis

```{r, fig.width = 16,fig.height=12}

plot.cyto.sub(c("CCUS","MDS","MDS/MPN"))

```

#### Correlogram

```{r}

M = cor(cyto.wide[,4:ncol(cyto.wide)],use='pairwise.complete.obs')
corrplot(M, method = 'square', order = 'FPC', type = 'upper', diag = FALSE)
```

