---
title: "EVI-2 Manuscript"
output: html_document
date: "2025-03-28"
---

```{r setup, include=FALSE}
#Prevent messages, warnings, and code from showing up in the report to keep things tidy. 
#All code chunks have been run independently to verify model fits
knitr::opts_chunk$set(echo = FALSE,warning = FALSE,message = FALSE)

#Turn off scientific notation
options(scipen=999)

#Helper function for taking the last n characters of a string
substrRight <- function(x, n){
  substr(x, nchar(x)-n+1, nchar(x))
}

library(ggmosaic)
library(ggh4x)
library(ggsurvfit)
library(gginnards)
library(readxl)
library(robustlmm)
library(emmeans)
library(survival)
library(ggbeeswarm)
library(tidycmprsk)
library(survminer)
library(ggprism)
library(forestmodel)
library(survival)
library(ggbeeswarm)
library(table1)
library(MASS)
library(car)
library(patchwork)
library(ranger)
library(factoextra)
library(ggplot2)
library(dplyr)
library(ggfortify)
library(ggsci)
library(timereg)
library(grDevices)
library(cmprsk)
library(readr)

```

# Section {.tabset}

```{r}
#Read in the main data
df = read_excel(unzip("/varidata/research/projects/jones/computerome/Vitc_Data.zip","EVI2_data_VAI.xlsx"))

# These patients had to be manually changed per Stine, email sent 2024-05-28 and another Feb 2025
df[df$ID == 64,]$Diagnosis_simple_baseline = "MDS"
df[df$ID == 72,]$Diagnosis_simple_baseline = "NA"
df[df$ID == 96,]$Diagnosis_simple_baseline = "NA"

#Convert dummy coding to named treatment arms
df$Treatment_arm = ifelse(df$Treatment_arm == 0, "Vitamin C","Placebo")

#Make an age in years variable
df$Age = (df$Age_inclusion_days/365.250 )

#Dummy code diagnosis into CCUS or non CCUS
df$CCUS = ifelse(df$Diagnosis_simple_baseline =="CCUS","CCUS","NonCCUS")
df$CCUS = as.numeric(factor(df$CCUS,levels=c("NonCCUS","CCUS")))

#This is to make the IDs match across data.frames
df$ID = sapply(strsplit(df$ID,"-"), function(x){ifelse(length(x) >1, paste0(x[[1]],as.numeric(x[[2]])),x)})

#Read in VAF data
VAFS = read.table("~/bbc-secondary/research/JONP_20240430_vitc_trial_VBCS/cura4_20250202.tsv",sep =  "\t",header=T)

#Some data cleaning to give IDs consistent formatting across data frames and create a time variable from the ID column
VAFS$ID = sapply(strsplit(VAFS$Tumor_Sample_Barcode,"-"), function(x){paste0(x[[1]],"-",x[[2]])})
VAFS$Time = substrRight(VAFS$ID,1)
VAFS$ID = substr(VAFS$ID,1,nchar(VAFS$ID)-2)

```


## Cumulative Clonal Growth Rates {.tabset}
Growth rate, 𝛼 = log (𝑉𝐴𝐹 𝑓𝑜𝑙𝑙𝑜𝑤-𝑢𝑝/𝑉𝐴𝐹 𝑏𝑎𝑠𝑒𝑙𝑖𝑛𝑒 )/𝑡𝑖𝑚𝑒
Patients 14 and 96 have no VAF info because they did not have nay VAFS, therefore their growth rate is 0

```{r}

# data processing
# First thing we need to do is calculate growth rates
# To do this, we need more precise time intervals, so we must merge with study visit dates
vdates = read_excel("~/bbc-secondary/research/JONP_20240430_vitc_trial_VBCS/EVI2_study visit dates_all pts.xlsx")
vdates$...2 = as.numeric(vdates$...2)
vdates = vdates[! is.na(vdates$...2),]
vdates$ID = zoo::na.locf(vdates$EVI2)
vdates$ID = sapply(strsplit(vdates$ID,"-"), function(x){ifelse(length(x) >1, paste0(x[[1]],as.numeric(x[[2]])),x)})
vdates$visit = vdates$...2
VAFS.merge = left_join(VAFS,vdates,by = c("ID","visit"))
#Study dates are merged in, now we can calculate precise growth rates by mutation

#variables to keep: ID, ...4 (time), AF, Hugo symbol, and cDNA_change for merging
VAFS.sub = subset(VAFS.merge, select = c("ID","visit","AF","...4","Hugo_Symbol","cDNA_Change"))

#Split into 2 then merge
VAFS.sub.1 = VAFS.sub[VAFS.sub$visit == 1,]
colnames(VAFS.sub.1) = c("ID","visit_1","AF_1","Time_1","Hugo_Symbol","cDNA_Change")

VAFS.sub.2 = VAFS.sub[VAFS.sub$visit > 1,]
colnames(VAFS.sub.2) = c("ID","visit_FU","AF_FU","Time_FU","Hugo_Symbol","cDNA_Change")

vafs = full_join(VAFS.sub.1,VAFS.sub.2)

```

```{r}

# Patients with only one visit
IDs.ex = unique(names(which(table(vdates$ID) == 1)))

#Patients without a visit after visit 1
IDs.ex.alt = unique(vafs$ID[which(! vafs$ID %in% unique(vdates$ID[vdates$visit>1]))])

knitr::kable(vdates[vdates$ID %in% IDs.ex,],caption = "patients being removed from VAF analysis due to no follow-up visit")

#exclude these patients from the growth curve data
vafs = vafs[! vafs$ID %in% IDs.ex,]

```

```{r}

#Two patients have all follow-up time points, but no VAFs, these should still be included, adding them here. Patients 14 and 96
non.ev = df[df$ID %in% df$ID[which(! df$ID %in% vafs$ID)],]
non.ev = non.ev[! non.ev$ID %in% IDs.ex,]
non.ev$Stability = "Stable"
vafs = plyr::rbind.fill(vafs,subset(non.ev,select = c(ID)))

# Patients in the data.frame without a VAF at a time point are imputed with 0.01, a sufficiently small value
for( i in non.ev$ID){
  vafs[vafs$ID ==i,]$visit_1 = vdates[vdates$ID == i ,]$...2[1]
  vafs[vafs$ID ==i,]$visit_FU = tail(vdates[vdates$ID == i ,]$...2,1)
  vafs[vafs$ID ==i,]$AF_1 = 0.01
  vafs[vafs$ID ==i,]$AF_FU = 0.01
}

#No one has both baseline and FU missing
knitr::kable(vafs[is.na(vafs$AF_1) &is.na(vafs$AF_FU), ],caption = "Missing baseline and FU")

## There are some NA FU_times, we need to make these 0.01 and also fill in the time, loop through each ID to get their FU time
vafs$AF_FU[is.na(vafs$AF_FU)] = "0.01"
for(i in unique(vafs[is.na(vafs$Time_FU),]$ID)){
  tmp =  vdates[vdates$ID == i ,]
  time = tmp$...4[tmp$visit == max(tmp$visit,na.rm=T)]
  vafs[is.na(vafs$Time_FU) & vafs$ID == i,]$Time_FU = time
}

#Create a dummy variable for 'new mutation'
vafs$Emergent = rep(0)
vafs$Emergent[is.na(vafs$AF_1) ] = 1
vafs$AF_1[is.na(vafs$AF_1)] = "0.01"

#Loop through each ID with a missing baseline time, make this time the date of the first time point collected
for(i in unique(vafs[is.na(vafs$Time_1),]$ID)){
  tmp =  vdates[vdates$ID == i ,]
  time = tmp$...4[tmp$visit == 1]
  vafs[is.na(vafs$Time_1) & vafs$ID == i,]$Time_1 = time
}

#Growth rate calc
vafs$GR = log(as.numeric(vafs$AF_FU)/as.numeric(vafs$AF_1))/ ((as.numeric(vafs$Time_FU)-as.numeric(vafs$Time_1))/365.25)

```

### Cumulative {.tabset}
#### All patients

Accounting: 109 patients, 9 with only a follow-up time, 3 without any VAF info, and 1 without either for a total of 11 patients excluded (8 no follow-up only, 2 no VAFS info only, 1 missing both)

```{r}

## Get cumulative growth
agg.d = aggregate(GR~ID,vafs,sum)

#merge in treatment info
vafs.c = merge(agg.d,subset(df,select=c(ID,Treatment_arm)))

# Summary table
table1::table1(~ GR | Treatment_arm,data=vafs.c)

# robust linear model
fit = rlm(GR~Treatment_arm,vafs.c)

knitr::kable(confint(emmeans(fit,pairwise~Treatment_arm)),caption = "emmeans and contrast")

p =  round(joint_tests(fit)[6],2)
knitr::kable(joint_tests(fit),caption="p-value on difference")

agg.mu = summary(emmeans(fit,pairwise~Treatment_arm)$emmeans)
colnames(agg.mu)[2] = "GR"

# Box plot or raw cumualtive GRs adding means as triangles and p-value with 2 sig-figs
clonal_trajectory_figure.a = ggplot(vafs.c,aes(x = Treatment_arm,y = GR,color=Treatment_arm)) + 
  geom_quasirandom() + 
  geom_boxplot(fill=NA,outliers = F) + 
  theme_prism(base_size = 12) + 
  scale_color_manual(values=c("black",pal_nejm()(4)[c(3)])) +
  ggtitle("Cumulative Clonal Growth Rates") + 
  ylab(expression(bold("Cumulative"~log(frac("AF at follow-up","AF at baseline"))*"/year"))) +
  coord_cartesian(ylim=c(-3.25,3.25))+xlab("Treatment") +
  geom_point(data=agg.mu,aes(x = Treatment_arm,y = GR,color=Treatment_arm),size=4,shape=2)+
  theme(legend.position = "none") +
  annotate(geom="line",x= c(1,2),y =c(3.1,3.1) ) +
  annotate(geom="text",x = 1.5, y = 3.3, label = paste0("p = ", p),fontface="italic")

#print the plot for the report
clonal_trajectory_figure.a

#output as pdf for journal
ggsave("~/bbc-secondary/research/JONP_20240430_vitc_trial_VBCS/clonal_trajectory_figure_a.pdf",width=6,height=6)

sum.df = aggregate(GR~Treatment_arm, vafs.c,mean)
sum.df$sd = aggregate(GR~Treatment_arm, vafs.c,sd)$GR

#This is the easiest way to get fold-change estimates with CIs
fit = rlm(log2(GR+0.01)~Treatment_arm,vafs.c)
fc = confint(emmeans(fit,pairwise~Treatment_arm,type="response"))
knitr::kable(fc,caption="FC")

```

#### Subgroups {.tabset}

```{r}

#Merge the summed GRs with clinical information
vafs.subgroup = merge(agg.d,subset(df,select=c(ID,Treatment_arm,VitC_PB1,Diagnosis_simple_baseline)))

#Make the median vitamin C variable
vafs.subgroup$med_vitc = ifelse(vafs.subgroup$VitC_PB1 >= 44.18,"above","below")

df$Diagnosis_simple_combo = case_when(df$Diagnosis_simple_baseline == "CCUS" ~ "CCUS",
                                      df$Diagnosis_simple_baseline == "MDS" ~ "MDS",
                                      df$Diagnosis_simple_baseline == "CMML" | df$Diagnosis_simple_baseline == "MDS/MPN"~ "CMML_MDS_MPN",
                                      TRUE~NA)

vafs.subgroup$vitc_ccus = vafs.subgroup$Diagnosis_simple_baseline
#CMML is part of MDS/MPN
vafs.subgroup$vitc_ccus[vafs.subgroup$vitc_ccus == "CMML"] <- "MDS/MPN"
#If patient has any of these mutations at baseline they are positive for mutation
mut_ids = unique(VAFS$ID[VAFS$Hugo_Symbol %in% c("TET2","IDH1","IDH2") & VAFS$Time==1])
vafs.subgroup$TET2 = ifelse(gsub("-","",vafs.subgroup$ID) %in% mut_ids,"TET2Mut", "TET2neg")

#Function to run the subgroup analyses
vafs.subgroup.fxn = function(subgroup){
  tmp.subgp = na.omit(subset(vafs.subgroup,select=c("GR","Treatment_arm",subgroup,"ID")))
  colnames(tmp.subgp)[3] = "grp"


  if(subgroup == "TET2"){
    tmp.subgp$grp =factor(tmp.subgp$grp,levels=c("TET2Mut", "TET2neg"),labels=c("TET2 or IDH1/2 mutation","No TET2 or IDH1/2 mutation"))
  } 

  if(subgroup == "med_vitc"){
    tmp.subgp$grp = factor(tmp.subgp$grp,levels=c("below","above"),labels=c("Below median","Above median"))
  }
  
  if(subgroup == "vitc_ccus"){
        
    tmp.subgp$grp = factor(tmp.subgp$grp,
                      levels = c("CCUS","MDS","MDS/MPN"),
                      labels= c("CCUS","MDS","MDS/MPN" ))
    tmp.subgp=tmp.subgp[!is.na(tmp.subgp$grp),]    

  }
  
  # Summary table
  table1::table1(~ GR | grp ,data=tmp.subgp)
  
  # robust linear model
  fit = rlm(GR~Treatment_arm*grp,tmp.subgp)
  
  sig.df = data.frame(confint(emmeans(fit,pairwise~Treatment_arm|grp)$contrasts))
  # sig.df$p.value =  ifelse(sig.df$p.value <0.0001,"p < 0.0001",paste0("p =  ",signif(round(sig.df$p.value,4),2)))
  # 
  
  sig.df$CI = paste0("Mean \U0394 ",-1*signif(sig.df$estimate,2),"\n95% CI ",
                     -1*signif(sig.df$asymp.UCL,2),", ",
                     -1*signif(sig.df$asymp.LCL,2))
    
  sig.df.line = rbind(sig.df,sig.df)
  if(subgroup == "vitc_ccus"){
    sig.df.line$x = c(1,1,1,2,2,2)
  } else {
    sig.df.line$x = c(1,1,2,2)
  }
  sig.df$y = max(tmp.subgp$GR,na.rm=T)
  sig.df.line$y = max(tmp.subgp$GR,na.rm=T)

  agg.mu = data.frame(summary(emmeans(fit,pairwise~Treatment_arm|grp)$emmeans))
  colnames(agg.mu)[3] = "GR"
  
  # Box plot or raw cumualtive GRs adding means as triangles and p-value with 2 sig-figs
  print(ggplot(tmp.subgp,aes(x = Treatment_arm,y = GR,color=Treatment_arm)) + geom_quasirandom() + 
    geom_boxplot(fill=NA,outliers = F) + theme_prism() + scale_color_manual(values=c("black",pal_nejm()(4)[c(3)])) +ggtitle("Cumulative clonal growth rates") + ylab(expression("Cumulative"~log(frac("AF at follow-up","AF at baseline"))*"/year")) + coord_cartesian(ylim=c(-3.5,3.5))+xlab("Treatment") + geom_point(data=agg.mu,aes(x = Treatment_arm,y = GR,color=Treatment_arm),size=4,shape=2)+theme(legend.position = "none",strip.background = element_blank()) +
    facet_wrap(~grp) + geom_line(data=sig.df.line,aes(x=x,y =3 ),color="black") + geom_text(data=sig.df,aes(x = 1.5,y=3.35,label=CI),color="black"))

}

```

##### TET2 mutation status

```{r,fig.height=6,fig.width=10}

vafs.subgroup.fxn("TET2")

```

##### Median baseline vitamin C mutation status

```{r,fig.height=6,fig.width=10}

vafs.subgroup.fxn("med_vitc")
```

##### Baseline diagnosis

```{r,fig.height=6,fig.width=10}

vafs.subgroup.fxn("vitc_ccus")
```


### Individual mutations {.tabset}


```{r}

#Counts of each mutation type
table(vafs$Hugo_Symbol)[order(table(vafs$Hugo_Symbol))]


##Manually curating this, code wasn't originally written with FDR in mind, but it should be FDRed p-vals can be obtained by running the function
indv.genes = data.frame(
  mut = c("TET2","DNMT3A","SRSF2","ASXL1","ZRSR2"),
  p = c(0.66,0.26,0.23,0.2,0.29))

indv.genes$FDR = p.adjust(indv.genes$p,method="BH")

# Function to plot GR of specific mutations with p-values
plot.spec.mut = function(mut) {
  vafs.mut = merge(vafs[vafs$Hugo_Symbol == mut,],subset(df,select=c(ID,Treatment_arm)))
  agg.vafs.mut = aggregate(GR~ID+Treatment_arm, vafs.mut,function(x) sum(x,na.rm=T))
  y = max(agg.vafs.mut$GR,na.rm=T)
  fit = rlm(GR~Treatment_arm,agg.vafs.mut)
  p = joint_tests(fit)
  FDR = indv.genes$FDR[indv.genes$mut == mut]
  print(ggplot(agg.vafs.mut,aes(x = Treatment_arm,y = GR,color=Treatment_arm)) + geom_quasirandom() + 
    geom_boxplot(fill=NA,outliers = F) + theme_prism() + scale_color_manual(values=c("black",pal_nejm()(4)[c(3)])) +ggtitle(mut)+xlab("Treatment") + annotate(geom="line",x= c(1,2),y =c(y*1.05,y*1.05) ) + annotate(geom="text",x = 1.5, y = y*1.13, label = paste0("FDR p = ", FDR),fontface="italic")+ylab("Growth rate")+theme(legend.position = "none")+xlab(""))
  
}
```

#### TET2

```{r}
plot.spec.mut("TET2")
```

#### DNMT3A

```{r}
plot.spec.mut("DNMT3A")
```

#### SRSF2

```{r}
plot.spec.mut("SRSF2")
```

#### ASXL1

```{r}
plot.spec.mut("ASXL1")
```

#### ZRSR2

```{r}
plot.spec.mut("ZRSR2")
```


### Emergence of a new mutation

```{r}

vafs$time = (as.numeric(vafs$Time_FU) - as.numeric(vafs$Time_1))/365.25

## Get max emergent (emergent is only 0 or 1, so this let's us know if a patient had any emerging VAFs)
agg.emerg = aggregate(Emergent~ID+time,vafs,max)

#merge in treatment info
vafs.c = merge(agg.emerg,subset(df,select=c(ID,Treatment_arm)))

#logistic regression adjusted for treatment time, most had roughly a year, but we can adjust for exact time
fit = glm(Emergent~Treatment_arm+time ,vafs.c,family="binomial")

knitr::kable(summary(emmeans(fit,pairwise~Treatment_arm)))
knitr::kable(joint_tests(fit))

table1(~Emergent + time|Treatment_arm,data=vafs.c)

```


### Progress, stable, Contracting clones
Each defined as: averaging >10% over 1 year is an expanding clones, <10% contracting clones, within +/-10% stable. 

```{r, fig.height = 8,fig.width=6}

## This accounts for time. Someone with a GR of 1.024, but their final timepoint is at 3 months has a GR of log(1.024)/0.25 = 0.095 ~= log(1.1)
vafs.c$Stability = ifelse(vafs.c$ID %in% unique(vafs[vafs$GR > log(1.1),]$ID), "Expanding clones ",ifelse(vafs.c$ID %in% unique(vafs[vafs$GR < log(0.9),]$ID), "Contracting clones ","Stable clones ") )

vafs.c$Stability = factor(vafs.c$Stability,levels=c("Contracting clones ","Stable clones ","Expanding clones "))

#chi-square test
p = chisq.test(table(vafs.c$Treatment_arm,vafs.c$Stability))$p.value

#Mosaic plot 
clonal_trajectory_figure.b =  ggplot(data = vafs.c) +
  geom_mosaic(aes(x = product(Stability,Treatment_arm), fill=Stability,color=Stability),alpha=1) +   
  theme_prism(base_size = 14) +
  theme(axis.text.x = element_text(colour = c("black",pal_nejm()(4)[c(3)]) ),
        axis.text.y = element_text(colour = c("blue","grey70","red"))) + ylab("") + xlab("Treatment")+
  ggtitle("Clonal trajectories" )+
  annotate(geom="text",x = 0.5,y = 1.1,label =  paste0("p = ", signif(round(p,4),2)),fontface="italic") +
  scale_fill_manual(values= c("blue","grey70","red"))+scale_color_manual(values= c("blue","grey70","red"))+ guides(fill=guide_legend(title=""))+theme(plot.title = element_text(),axis.line = element_blank(),
            axis.ticks = element_line(),axis.ticks.y = element_line(color=c("blue","grey70","red")),
            axis.ticks.x = element_line(color= c("black",pal_nejm()(4)[c(3)])))

  clonal_trajectory_figure.b =  clonal_trajectory_figure.b + geom_text(data = ggplot_build(clonal_trajectory_figure.b )$data[[1]] %>% 
                group_by(x__Treatment_arm) %>%
                mutate(pct = .wt/sum(.wt)*100), 
              aes(x = (xmin+xmax)/2, y = (ymin+ymax)/2, label=paste0(.wt," (",round(pct, 1),"%)")),color="white")+theme(legend.position = "none")

clonal_trajectory_figure.b

#Individual panel for the journal
ggsave("~/bbc-secondary/research/JONP_20240430_vitc_trial_VBCS/clonal_trajectory_figure_b.pdf",width=6,height=6)
```

### Manuscript figure

```{r, fig.height = 8,fig.width=12}

cowplot::plot_grid(clonal_trajectory_figure.a,NULL,clonal_trajectory_figure.b,rel_widths=c(1,0.1,1),
          labels=c("a","","b"),label_size = 15,ncol=3,scale=0.85)

ggsave("~/bbc-secondary/research/JONP_20240430_vitc_trial_VBCS/clonal_trajectory_figure.pdf",width=12,height=6)

```

## Plasma Vitamin C {.tabset}
### Figure 3

**Figure 3: Patient plasma vitamin C level.** A) Composition of all patients, placebo controls, and vitamin C treated individuals with: vitamin C deficiency, inadequate vitamin C, or adequate vitamin C at inclusion; distributions are relatively similar between the two treatment arms. B) Plasma vitamin C level at each timepoint by treatment arm; p-values were estimated using robust linear mixed-effects model with a random intercept for each individual. C) LOESS smoothed vitamin C level over time with standard error ribbons. Individuals with *>200 \mumol/L* vitamin C level were censored in C) to better characterize dispersion among trial participants, all individual measures are shown in panel B).

```{r, fig.height = 14, fig.width=10}

df.sp =  subset(df,select = c(ID,Treatment_arm,VitC_PB1,VitC_PB2,VitC_PB3,VitC_PB4,VitC_PB5,Age,Sex,Hb1,CCUS))
df.sp =  reshape2::melt(df.sp,id.vars=c("ID","Treatment_arm","Age","Sex","Hb1","CCUS"))
df.sp$variable = factor(df.sp$variable, levels = c("VitC_PB1","VitC_PB2","VitC_PB3","VitC_PB4","VitC_PB5"))

df.sp$variable = case_when(df.sp$variable == "VitC_PB1" ~ 0,
                               df.sp$variable == "VitC_PB2" ~ 365.25/4,
                               df.sp$variable == "VitC_PB3" ~ 2*365.25/4,
                               df.sp$variable == "VitC_PB4" ~ 3*365.25/4,
                               df.sp$variable == "VitC_PB5" ~ 365.25)

tmp1 = data.frame(variable =12.1/12*365.25,value2 = 11.5,Treatment_arm = "Vitamin C" )
tmp2 = data.frame(variable =12.1/12*365.25,value2 = 36.5,Treatment_arm = "Vitamin C" )

#This is for plotting purposes only and is noted in the figure legend
df.sp$value2 = df.sp$value
df.sp$value2[df.sp$value>200] = 200
df.sp$value2[df.sp$value2==200 & df.sp$variable == 182.6250] = NA

p1.fig2 = ggplot(df.sp[!is.na(df.sp$value),],aes(x = 12*as.numeric(variable)/365.25,y=value2,color=Treatment_arm,fill=Treatment_arm))  +    geom_hline(aes(yintercept =50),color = colorRampPalette(c("steelblue3","black"))(5)[1],linewidth=1,linetype="dashed") +
 geom_hline(aes(yintercept =23),color=colorRampPalette(c("red3","black"))(5)[1],linewidth=1,linetype="dotted") +
theme_prism() + geom_line(aes(group =  ID),alpha=0.15 ) + stat_smooth(se=T,linewidth=1.5,span=1) + scale_color_manual(values =c("black",pal_nejm()(4)[3])) + scale_fill_manual(values =c("black",pal_nejm()(4)[3])) +scale_y_continuous(expand=c(0,0))+ scale_x_continuous(breaks=c(0,3,6,9,12),expand=c(0,0)) +xlab("Months from randomisation")+ylab("Vitamin C concentration (\U03BCmol/L)")+
  geom_text(data=tmp1,color =colorRampPalette(c("red3","black"))(5)[1],label="Deficient",hjust=0) +
  geom_text(data=tmp2,color =colorRampPalette(c("steelblue3","black"))(5)[1],label="Inadequate",hjust=0) +

  guides(color=guide_legend(title=""),fill=guide_legend(title=""))+coord_cartesian(clip="off")+facet_wrap(~Treatment_arm)+ theme(panel.spacing = unit(2, "lines"))

df.sp$Time = factor(paste0(12*df.sp$variable/365.25,""),levels=seq(0,12,3))

fit = rlmer(log2(value)~Time*Treatment_arm+(1|ID),data=df.sp)
res = data.frame(summary(emmeans(fit,pairwise~Treatment_arm|Time)$contrasts))
res$Label = ifelse(res$p.value <0.0001,"p < 0.0001",paste0("p =  ",signif(round(res$p.value,4),2)))


p2.fig2= ggplot()  +   geom_hline(aes(yintercept =23),color="red3",linewidth=1,linetype="dotted") +
    geom_hline(aes(yintercept =50),color = "steelblue3",linewidth=1,linetype="dashed") +
theme_prism()+ geom_boxplot(data=df.sp,aes(x = Time,y=value,color=Treatment_arm),outlier.size=-1) + geom_quasirandom(data=df.sp,aes(x = Time,y=value,color=Treatment_arm),alpha=0.25,dodge.width = .75)  + scale_color_manual(values =c("black",pal_nejm()(4)[3]))  +xlab("Months from randomisation")+ylab("Vitamin C concentration (\U03BCmol/L)")+
  annotate(geom="text",x = 6,y = 11.5,color = "red3",label="Deficient",hjust=0) +
  annotate(geom="text",x = 6,y = 36.5,color = "steelblue3",label="Inadequate",hjust=0) +
  guides(color=guide_legend(title=""),fill=guide_legend(title=""))+coord_cartesian(clip="off")+scale_y_continuous(breaks=seq(0,300,50)) + geom_text(data=res,aes(label=Label,x=Time,y=260),color="black",fontface="italic")+geom_segment(aes(x = 0.75,xend=1.25,y=240,yend=240))+
  geom_segment(aes(x = 1.75,xend=2.25,y=240,yend=240))+
  geom_segment(aes(x = 2.75,xend=3.25,y=240,yend=240))+
  geom_segment(aes(x = 3.75,xend=4.25,y=240,yend=240))+
  geom_segment(aes(x = 4.75,xend=5.25,y=240,yend=240))

df.sp$Adequacy = case_when( df.sp$value < 23 ~ "Deficient",
                            df.sp$value < 50 ~ "Inadequate",
                            TRUE ~ "Sufficient")

pt = data.frame(prop.table(table(df.sp$Adequacy,df.sp$Time),2))

pt.c = data.frame(prop.table(table(df.sp$Adequacy[df.sp$Treatment_arm=="Vitamin C"],df.sp$Time[df.sp$Treatment_arm=="Vitamin C"]),2))

pt.p =  data.frame(prop.table(table(df.sp$Adequacy[df.sp$Treatment_arm=="Placebo"],df.sp$Time[df.sp$Treatment_arm=="Placebo"]),2))

pt$Group =  "All Patients (n = 109)"
pt.c$Group =  "Vitamin C (n = 55)"
pt.p$Group =  "Placebo (n = 54)"

pt = rbind(pt,pt.c,pt.p)

pt$Var1 = as.character(pt$Var1)

pt$Var1[pt$Var1 == "Deficient"] = "Deficient (< 23 \U03BCmol/L)"
pt$Var1[pt$Var1 == "Inadequate"] = "Inadequate (23-49 \U03BCmol/L)"
pt$Var1[pt$Var1 == "Sufficient"] = "Adequate (\u2265 50 \U03BCmol/L)"


hsize=1.5
pt$hsize=hsize
pt$Freq = round(pt$Freq,2)
pt$Var1 = factor(pt$Var1,levels=c("Adequate (≥ 50 μmol/L)",
                                  "Inadequate (23-49 μmol/L)",
                                  "Deficient (< 23 μmol/L)" ))

p2.pi = ggplot(pt[pt$Var2 == 0,],aes(x =hsize,y = Freq,fill=Var1 )) +
  geom_col() +facet_wrap(~Group)+
  coord_polar(theta = "y",clip="off") +
  xlim(c(0.2, hsize + 0.5))+
  geom_text(aes(label = scales::percent(Freq)),
          position = position_stack(vjust = 0.5))+
  theme_prism(base_family = "") + 
  theme(legend.position = "bottom",
    panel.background = element_rect(fill = "white"),
                        panel.grid = element_blank(),
                        axis.title = element_blank(),
                        axis.ticks = element_blank(),
                        axis.text = element_blank(),
                        axis.line = element_blank(),
        legend.key.spacing.x = grid::unit(1,"cm")) +
  scale_fill_manual(values=c("Adequate (≥ 50 μmol/L)" = "palegreen3","Inadequate (23-49 μmol/L)" = "steelblue3","Deficient (< 23 μmol/L)" = "red3"),breaks=c("Deficient (< 23 μmol/L)"  , 
                             "Inadequate (23-49 μmol/L)",
                              "Adequate (≥ 50 μmol/L)"))+ggtitle("Vitamin C concentration in peripheral blood plasma at inclusion")

cowplot::plot_grid(p2.pi,p2.fig2,p1.fig2,ncol=1,labels=c("a","b","c"),label_size = 15,scale=0.9)

ggsave("~/bbc-secondary/research/JONP_20240430_vitc_trial_VBCS/vitaminc_concentration_figure.pdf",width=10,height=14)

p2.pi
ggsave("~/bbc-secondary/research/JONP_20240430_vitc_trial_VBCS/vitaminc_concentration_figure.a.pdf",width=9,height=14/3)

p2.fig2
ggsave("~/bbc-secondary/research/JONP_20240430_vitc_trial_VBCS/vitaminc_concentration_figure.b.pdf",width=9,height=14/3)

p1.fig2
ggsave("~/bbc-secondary/research/JONP_20240430_vitc_trial_VBCS/vitaminc_concentration_figure.c.pdf",width=9,height=14/3)


```

### Supplemental figure, Vitamin C change over time


**Supplemental figure xx. Change in vitamin C over time.** On average patients treated with vitamin C had a ~2.1 fold increase in plasma vitamin C level between their measures at inclusion and those taken at End of treatment (95% CI 1.6 - 2.8), while those on placebo had no real change. robust linear mixed-effects model with ratio of vitamin C at EOT divided by vitamin C at inclusion.  


```{r, fig.height=5, fig.width = 5.5 }

label(df$VitC_PB1) <- 'Baseline Vitamin C'
label(df$VitC_PB5) <- 'EOT Vitamin C'

my.render.cont <- function(x) {
    with(stats.default(x), 
         c("",
           
          "Mean (SD)" = sprintf("%s (%s)",
                                round_pad(MEAN, 2),
                                round_pad(SD, 2)),
         
          "Median (Min, Max)" = sprintf("%s (%s, %s)",
                                       round_pad(MEDIAN, 2), 
                                       round_pad(Q1, 2), 
                                       round_pad(Q3, 2)))
    )
}

table1::table1(~ VitC_PB1+VitC_PB5 |Treatment_arm,data=df,
               render.continuous=my.render.cont)
               # render.continuous=c(.="Mean (sd)", 
               #                             .="Median [Q1, Q3]"))

fit = rlm(VitC_PB5 - VitC_PB1 ~ Treatment_arm,data=df)

knitr::kable(test(emmeans(fit,"Treatment_arm",type="response")),caption="change in vitamin C")


p1.delta.c = ggplot(df,aes(x =Treatment_arm,y= VitC_PB5 - VitC_PB1, color=Treatment_arm)) + geom_quasirandom() +
  geom_boxplot(fill=NA,outliers=F)+scale_color_manual(values =c("black",pal_nejm()(4)[3]))+theme_prism()+theme(legend.position = "none") + xlab("") + ylab("Change in plasma vitamin C \n(End of treatment - Baseline)") + annotate(geom = "line",x = c(1,2), y = c(135,135)) + annotate(geom="text" , x=1.5,y = 145,label="p < 0.0001 ",fontface="italic")

fit = rlm(log2(VitC_PB5/VitC_PB1) ~ Treatment_arm,data=df)
# knitr::kable(test(emmeans(fit,"Treatment_arm")),caption="change in vitamin C")
knitr::kable(summary(confint(emmeans(fit,pairwise~Treatment_arm,type="response"))))

p2.delta.c = ggplot(df,aes(x =Treatment_arm,y= log2(VitC_PB5/VitC_PB1), color=Treatment_arm)) + geom_quasirandom() +
  geom_boxplot(fill=NA,outliers=F)+scale_color_manual(values =c("black",pal_nejm()(4)[3]))+theme_prism()+theme(legend.position = "none") + xlab("") + ylab("log2(FC) plasma vitamin C \n(End of treatment - Baseline)") + annotate(geom = "line",x = c(1,2), y = c(6.5,6.5)) + annotate(geom="text" , x=1.5,y = 7.02,label="p < 0.0001 ",fontface="italic")

##Keeping the FC plot for manuscript
p2.delta.c

```

### Supplemental figure, plasma versus bone marrow vitamin C

```{r, fig.width = 10,fig.height=5}

df.bm =  subset(df,select = c(ID,Treatment_arm,VitC_BM1,VitC_BM2,VitC_BM5,Age,Sex,Hb1,CCUS))
df.bm =  reshape2::melt(df.bm,id.vars=c("ID","Treatment_arm","Age","Sex","Hb1","CCUS"))
df.bm$variable = factor(df.bm$variable, levels = c("VitC_BM1","VitC_BM2","VitC_BM5"))

df.bm$variable = case_when(df.bm$variable == "VitC_BM1" ~ 0,
                               df.bm$variable == "VitC_BM2" ~ 365.25/4,
                               df.bm$variable == "VitC_BM5" ~ 365.25)

tmp1 = data.frame(variable =12.1/12*365.25,value2 = 11.5,Treatment_arm = "Vitamin C" )
tmp2 = data.frame(variable =12.1/12*365.25,value2 = 36.5,Treatment_arm = "Vitamin C" )

#This is for plotting purposes only and is noted in the figure legend
df.bm$value2 = df.bm$value
df.bm$value2[df.bm$value>200] = 200
df.bm$value2[df.bm$value2==200 & df.bm$variable == 182.6250] = NA
df.sp$Tissue = "Peripheral Blood"
df.bm$Tissue = "Bone Marrow"

df.vitc = rbind(df.sp[,colnames(df.sp) %in% colnames(df.bm)],df.bm)
df.vitc$ID = paste0(df.vitc$ID,df.vitc$Tissue)

ggplot(df.vitc[!is.na(df.vitc$value),],aes(x = 12*as.numeric(variable)/365.25,y=value2,color=Tissue,fill=Tissue)) + theme_prism() + geom_line(aes(group =  ID),alpha=0.15 ) + stat_smooth(se=T,linewidth=1.5,span=1) + scale_color_manual(values =c("#648ad3","#880808")) + scale_fill_manual(values =c("#648ad3","#880808")) +scale_y_continuous(expand=c(0,0))+ scale_x_continuous(breaks=c(0,3,6,9,12),expand=c(0,0)) +xlab("Months from randomisation")+ylab("Vitamin C concentration (\U03BCmol/L)")+
  guides(color=guide_legend(title=""),fill=guide_legend(title=""))+coord_cartesian(clip="off")+facet_wrap(~Treatment_arm)+ theme(panel.spacing = unit(2, "lines"))


```

### Correlate Vitamin C in plasma and bone marrow

```{r, fig.height = 5,fig.width=10}

colnames(df.sp)[colnames(df.sp) == "value"] = "blood"
colnames(df.bm)[colnames(df.bm) == "value"] = "bone"
df.sp = subset(df.sp,select= - c(Tissue))
df.bm = subset(df.bm,select= - c(Tissue))
df.vitc.corr = merge(df.sp,df.bm,by=c("ID","Treatment_arm","variable"))
df.vitc.corr$variable = factor(df.vitc.corr$variable,levels=c(0,91.3125,365.25),labels=c("Baseline","3 months","12 months"))

ggplot(df.vitc.corr,aes(x = blood,y=bone )) + geom_point() +
  facet_wrap(~variable) +theme_prism() +stat_smooth(method="lm",color="black",fill="black") +
  stat_cor(method = "spearman",p.accuracy = 0.0001, r.accuracy = 0.01,
             aes(label = paste(..r.label.., ..p.label.., sep = "~`,`~"))) +
  ylab("Bone marrow vitamin C (μmol/L)") + xlab("Peripheral blood vitamin C (μmol/L)")

```

## Survival {.tabset}
Includes all samples, unadjusted. We also performed a landmark analysis and also analyzed this same data with peto-peto. Both were still significant, therefore an FDR correction would not change our findings here.

### Primary outcome KM (Figure 3)

**Figure 4. Unadjusted Kaplan-Meier curves for overall survival.** Estimated hazard ratio for vitamin C treated patients versus controls based on an unadjusted Cox proportional hazards regression 0.35, 95% CI 0.17, 0.71.

```{r,fig.width=8, fig.height=9}

df$Years_FU_OS = (df$Days_FU_OS/365.25)
df$Months_FU_OS = (df$Days_FU_OS/365.25 * 12)
label(df$Months_FU_OS) = "Follow up (Months)"
label(df$FU_Death) = "Died"
df$FU_event = factor(df$FU_Death,levels=c(0,1),labels=c("Censored","Died"))
label(df$FU_event) = "Event"
table1::table1(~ Months_FU_OS +FU_event |Treatment_arm,data=df,
               render.continuous=c(.="Mean (sd)", 
                                           .="Median [Q1, Q3]"))

fit_os = survfit(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df)

os.p =  ggsurvplot(fit_os,
           legend.labs =gsub("Treatment_arm=","",names(fit_os$strata)),
           palette=c("black",pal_nejm()(4)[c(3)]),
           cumevents=F,
           risk.table="nrisk_cumcensor",
           # cumcensor=T,
           break.time.by=365.25,
           xscale="d_m",
           xlab="Months from randomisation",
           pval.method = F,
           pval=F,legend.title="")
fit_os.cox = coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df)

p.y = min(os.p$data.survplot[os.p$data.survplot$Treatment_arm != "Placebo", ]$surv)
p.c = min(os.p$data.survplot[os.p$data.survplot$Treatment_arm == "Placebo", ]$surv)
p.os = os.p$plot+theme(legend.position = "none") +
  annotate(geom="text",x = 6*365.25,y = p.y,label="Vitamin C", color=pal_nejm()(4)[c(3)],hjust=0) + 
  annotate(geom="text",x = 6*365.25,y = p.c,label="Placebo", color="black",hjust=0) + 
  scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+
  coord_cartesian(clip="off")+theme_prism()+theme(legend.position = "none")+
  geom_hline(yintercept = 0.5,linetype="dotted",linewidth = 1.25) +theme(legend.position = "none",plot.margin = unit(c(0, 1, 0, 0), "inches")) + annotate(geom="text", x = 365,y = 0.25, label=paste0("Log-rank\np =  0.0025\nHR ",round(exp(fit_os.cox$coefficients),2),"; ", "95% CI ",round(exp(confint(fit_os.cox)[1]),2),", ", round(exp(confint(fit_os.cox)[2]),2) ),fontface="italic",hjust=0,size=4.5)+ggtitle("Overall survival")

t.os = os.p$table + theme_prism()+ theme(axis.line = element_blank())+scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+coord_cartesian(clip="off")+theme(text = element_text(face="bold",family=""),axis.text.y = element_text(hjust=1,colour = c(pal_nejm()(4)[c(3)],"black"),margin= margin(0, 15, 0, 7) ),axis.ticks.y = element_blank(),plot.margin = unit(c(0, 0, 0, 0), "inches"))+xlab("")  +ylab("Number at risk\n(Cumulative censoring)") + theme(axis.text.x = element_blank(),axis.ticks.x = element_blank(),plot.title = element_blank())

p.os/t.os/plot_spacer() + plot_layout(heights = c(1,0.25,0.02))

ggsave("~/bbc-secondary/research/JONP_20240430_vitc_trial_VBCS/survival_figure.a.pdf",width=8,height=9.5)

knitr::kable(summary(fit_os.cox)$conf.int,caption="Hazard ratio and CI")

p.fig.3a = p.os/t.os +plot_layout(heights = c(1,0.25))


```

### Cox model (No interaction)


**Supplemntal figure XX. Forest plot for adjusted Cox model.** Cox proportional hazards regression to estimated efficacy of vitamin C on overall survival was adjusted for patient age at inclusion, gender, hemoglobin levels at inclusion, and whether or not they had been diagnosed with CCUS. 

```{r, fig.width=11, figheight=5}

df$Treatment = as.numeric(factor(df$Treatment_arm,levels=c("Placebo","Vitamin C"),labels=c("Placebo","VitC")))

df$VitaminC = as.numeric(factor(df$Treatment_arm,levels=c("Placebo","Vitamin C"),labels=c("Placebo","VitC")))-1

df$Age_mean_centered = (df$Age-mean(df$Age,na.rm=T))

df$VitaminC_Age_Interaction = as.numeric(as.factor(df$Treatment_arm))*df$Age_mean_centered
df$Male = df$Sex

labelled::var_label(df) <- list(
    VitaminC = "Vitamin C group", 
    Age = "Age (years)",
    Hb1 = "Haemoglobin at baseline (g/dL)",
    CCUS = "CCUS diagnosis (yes/no)"
)

fit.cox.simple =  coxph(Surv(Days_FU_OS,FU_Death)~VitaminC+Age +Male + Hb1 + CCUS,data=df)

panels <- list(
  list(width = 0.03),
  list(width = 0.1, display = ~variable, fontface = "bold", heading = "Variable"),
  list(width = 0.1, display = ~level),
  list(width = 0.03, item = "vline", hjust = 0.5),
  list(
    width = 0.55, item = "forest", hjust = 0.5, heading = "Hazard ratio", linetype = "dashed",
    line_x = 0
  ),
  list(width = 0.05, item = "vline", hjust = 0.5),
  list(width = 0.12, display = ~ ifelse(reference, "Reference", sprintf(
    "%0.3f (%0.3f, %0.3f)",
    trans(estimate), trans(conf.low), trans(conf.high)
  )), display_na = NA, heading = "HR (95% CI)"),
    list(width = 0.03, item = "vline", hjust = 0.5),
  list(
    width = 0.05,
    display = ~ ifelse(reference, "", paste0(ifelse(p.value < 0.0001,"P ","p =  "), format.pval(p.value, digits = 1, eps = 0.0001))),
    display_na = NA, hjust = 0.5,fontface="italic"
  ),
  list(width = 0.03)
)

forest_model(fit.cox.simple,panels) 



```


### Cox model (Age interaction)

**Supplemental figure XX. Forest plot for adjusted Cox model in age interaction.** Cox proportional hazards regression to estimated efficacy of vitamin C as a function of age. Age was mean centered and converted to decades to improve hazard ratio scales. Overall vitamin C was still effective, with evidence that it's efficacy decreases with age.

```{r, fig.width=11, figheight=5}

df$age_c = df$Age - mean(df$Age,na.rm=T)
df$age_tx = df$age_c * df$VitaminC
labelled::var_label(df) <- list(
    VitaminC = "Vitamin C group", 
    age_c = "Age (Years)",
    age_tx = "Age x Vitamin C group",
    Hb1 = "Haemoglobin at baseline (g/dL)",
    CCUS = "CCUS diagnosis (yes/no)"
)

fit.cox.age_int =  coxph(Surv(Days_FU_OS,FU_Death)~VitaminC+age_c + age_tx+Male + Hb1 + CCUS,data=df)

forest_model(fit.cox.age_int,panels)


```

### Cox model (TET interaction)

```{r, fig.width=11, figheight=5}

mut_ids = unique(VAFS$ID[VAFS$Hugo_Symbol %in% c("TET2","IDH1","IDH2") & VAFS$Time==1])
tet_ids = unique(VAFS$ID[VAFS$Hugo_Symbol %in% c("TET2") & VAFS$Time==1])
df$TET2 = ifelse(gsub("-","",df$ID) %in% mut_ids,"TET2Mut", "TET2neg")

knitr::kable(table(df$TET2,df$Treatment_arm))

df$TET2.bi = ifelse(df$ID %in% mut_ids,1,0)

df$tet_tx = I(df$TET2 == "TET2Mut") * df$VitaminC


labelled::var_label(df) <- list(
    VitaminC = "Vitamin C group", 
    age_c = "Age (Years)",
    tet_tx = "TET2 x Vitamin C group",
    Hb1 = "Haemoglobin at baseline (g/dL)",
    CCUS = "CCUS diagnosis (yes/no)",
    TET2.bi = "TET2 Mutation (yes/no)"
)

fit.cox.age_int =  coxph(Surv(Days_FU_OS,FU_Death)~VitaminC + age_c + TET2.bi + tet_tx + Male + Hb1 + CCUS,data=df)

forest_model(fit.cox.age_int,panels)

```


### Cox model (Hb1 interaction)

```{r, fig.width=11, figheight=5}
df$Hb1_c = df$Hb1 - mean(df$Hb1,na.rm=T)
df$vitc_hb1 = df$Hb1_c * df$VitaminC
labelled::var_label(df) <- list(
    VitaminC = "Vitamin C group", 
    age_c = "Age (Years)",
    vitc_hb1 = "Haemoglobin at baseline (g/dL) x Vitamin C group",
    Hb1_c = "Haemoglobin at baseline (g/dL)",
    CCUS = "CCUS diagnosis (yes/no)"
)

fit.cox.hb1_int =  coxph(Surv(Days_FU_OS,FU_Death)~VitaminC+age_c + vitc_hb1+Male + Hb1_c + CCUS,data=df)

forest_model(fit.cox.hb1_int,panels)


```



### Cox model (Sex interaction)


```{r, fig.width=11, figheight=5}


df$sex_vitc = df$Male * df$VitaminC
labelled::var_label(df) <- list(
    VitaminC = "Vitamin C group", 
    age_c = "Age (Years)",
    sex_vitc = "Male x Vitamin C group",
    Hb1_c = "Haemoglobin at baseline (g/dL)",
    CCUS = "CCUS diagnosis (yes/no)"
)

fit.cox.sex_int =  coxph(Surv(Days_FU_OS,FU_Death)~VitaminC+age_c + sex_vitc+Male + Hb1_c + CCUS,data=df)

forest_model(fit.cox.sex_int,panels)



```



### Cox model (CCUS interaction)

Forest plot cannot be made due to diverging SE in treatment

```{r, fig.width=11, figheight=5}


df$vitc_ccus = (df$CCUS-1) * df$VitaminC

labelled::var_label(df) <- list(
    VitaminC = "Vitamin C group", 
    age_c = "Age (Years)",
    vitc_ccus = "CCUS x Vitamin C group",
    Hb1_c = "Haemoglobin at baseline (g/dL)",
    CCUS = "CCUS diagnosis (yes/no)"
)

fit.cox.ccus_int =  coxph(Surv(Days_FU_OS,FU_Death)~VitaminC+age_c + vitc_ccus+Male + Hb1_c + CCUS,data=df)
fit.cox.ccus_int

## Forst plot could not be made due to divergence
# forest_model(fit.cox.ccus_int,panels)

```


### Cox model (baseline vitamin C interaction)

```{r, fig.width=11, figheight=5}

df$VitC_PB1_c = df$VitC_PB1 - mean(df$VitC_PB1,na.rm=T)
df$vitc_PB1_int = df$VitC_PB1_c * df$VitaminC
labelled::var_label(df) <- list(
    VitaminC = "Vitamin C group", 
    age_c = "Age (Years)",
    VitC_PB1 = "Baseline vitamin C (μmol/L)",
    vitc_PB1_int = "Baseline vitamin C (μmol/L) x Vitamin C group",
    Hb1_c = "Haemoglobin at baseline (g/dL)",
    CCUS = "CCUS diagnosis (yes/no)"
)

fit.cox.vitc_pb1_int =  coxph(Surv(Days_FU_OS,FU_Death)~VitaminC+age_c + VitC_PB1+Male + vitc_PB1_int + Hb1_c + CCUS,data=df)

forest_model(fit.cox.vitc_pb1_int,panels)

```



### 2-year landmark 

**Maybe Figure?** A 2 year landmark analysis was performed to determine if there was still evidence of vitamin C efficacy after removing individuals who had events prior to 2 years. This landmark analysis ensures the effects of vitamin C had sufficient time to reach efficacy.


```{r,fig.width=8, fig.height=9}

fit_os.L = survfit(Surv(Days_FU_OS-2*365.25,FU_Death)~Treatment_arm,data=df[df$Days_FU_OS>=2*365.25,])

os.p =  ggsurvplot(fit_os.L,
           legend.labs =gsub("Treatment_arm=","",names(fit_os$strata)),
           palette=c("black",pal_nejm()(4)[c(3)]),
           cumevents=F,
           risk.table="nrisk_cumcensor",
           # cumcensor=T,
           break.time.by=365.25,
           xscale="d_m",
           xlab="Months post 2 year landmark",
           pval.method = F,
           pval=F,legend.title="")


p.y = min(os.p$data.survplot[os.p$data.survplot$Treatment_arm != "Placebo", ]$surv)
p.c = min(os.p$data.survplot[os.p$data.survplot$Treatment_arm == "Placebo", ]$surv)
p.os = os.p$plot+theme(legend.position = "none") +
  annotate(geom="text",x = 4*365.25,y = p.y,label="Vitamin C", color=pal_nejm()(4)[c(3)],hjust=0) + 
  annotate(geom="text",x = 4*365.25,y = p.c,label="Placebo", color="black",hjust=0) + 
  scale_x_continuous(limits=c(0,365.25*4), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+
  coord_cartesian(clip="off")+theme_prism()+theme(legend.position = "none")+
  geom_hline(yintercept = 0.5,linetype="dotted",linewidth = 1.25) +theme(legend.position = "none",plot.margin = unit(c(0, 1, 0, 0), "inches")) + annotate(geom="text", x = 365,y = 0.25, label="Log-rank\np =  0.028",fontface="italic",hjust=0,size=4.5)

t.os = os.p$table + theme_prism()+ theme(axis.line = element_blank())+scale_x_continuous(limits=c(0,365.25*4), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+coord_cartesian(clip="off")+theme(text = element_text(face="bold",family=""),axis.text.y = element_text(hjust=1,colour = c(pal_nejm()(4)[c(3)],"black"),margin= margin(0, 15, 0, 7) ),axis.ticks.y = element_blank(),plot.margin = unit(c(0, 0, 0, 0), "inches"))+xlab("") + ggtitle("") +ylab("Number at risk\n(Cumulative censoring)") + theme(axis.text.x = element_blank(),axis.ticks.x = element_blank(),plot.title = element_blank())

p.os/t.os +plot_layout(heights = c(1,0.25))


```


### Per patient protocol
likelihood ratio test p-value is nearly identical to log-rank

```{r,fig.width=8, fig.height=9}

df.pp =  read_excel("/varidata/research/projects/jones/computerome/Vitc_Data/EVI2_per protocol population.xlsx")


df.pp$ID = sapply(strsplit(df.pp$ID,"-"), function(x){ifelse(length(x) >1, paste0(x[[1]],as.numeric(x[[2]])),x)})

df$pp =  ifelse(df$ID %in% df.pp$ID,1,0)
df=df[df$ID != "USC60",]

fit_os = survfit(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$pp == 1,])
os.p =  ggsurvplot(fit_os,
           legend.labs =gsub("Treatment_arm=","",names(fit_os$strata)),
           palette=c("black",pal_nejm()(4)[c(3)]),
           cumevents=F,
           risk.table="nrisk_cumcensor",
           # cumcensor=T,
           break.time.by=365.25,
           xscale="d_m",
           xlab="Months from randomisation",
           pval.method = F,
           pval=F,legend.title="")

p.y = min(os.p$data.survplot[os.p$data.survplot$Treatment_arm != "Placebo", ]$surv)
p.c = min(os.p$data.survplot[os.p$data.survplot$Treatment_arm == "Placebo", ]$surv)
p.os = os.p$plot+theme(legend.position = "none") +
  annotate(geom="text",x = 6*365.25,y = p.y,label="Vitamin C", color=pal_nejm()(4)[c(3)],hjust=0) +
  annotate(geom="text",x = 6*365.25,y = p.c,label="Placebo", color="black",hjust=0) +
  scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+
  coord_cartesian(clip="off")+theme_prism()+theme(legend.position = "none")+
  geom_hline(yintercept = 0.5,linetype="dotted",linewidth = 1.25) +theme(legend.position = "none",plot.margin = unit(c(0, 1, 0, 0), "inches")) 

t.os = os.p$table + theme_prism()+ theme(axis.line = element_blank())+scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+coord_cartesian(clip="off")+theme(text = element_text(face="bold",family=""),axis.text.y = element_text(hjust=1,colour = c(pal_nejm()(4)[c(3)],"black"),margin= margin(0, 15, 0, 7) ),axis.ticks.y = element_blank(),plot.margin = unit(c(0, 0, 0, 0), "inches"))+xlab("") + ggtitle("") +ylab("Number at risk\n(Cumulative censoring)") + theme(axis.text.x = element_blank(),axis.ticks.x = element_blank(),plot.title = element_blank())


HR.no = summary(coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$pp == 1,]))$conf.int

HR.no.anno = paste0("HR ",round(HR.no[1],2), "\n95% CI ",round(HR.no[3],2),", ",round(HR.no[4],2))

p.os = p.os + annotate(geom="text", x = 365.25,y = 0.25, label=HR.no.anno,hjust=0,size=4.5)


p.os/t.os +plot_layout(heights = c(1,0.25))


fit1 = coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$pp == 1,])

fit2 = coxph(Surv(Days_FU_OS,FU_Death)~1,data=df[df$pp == 1,])
knitr::kable(lmtest::lrtest(fit1,fit2),caption = "LRT on treatment")

```


### Subgroups (suppl) {.tabset}
All subgroup analyses include only the hazard ratio with 95% CI as these are exploratory. 


#### TET1 or IDH1/2 at baseline

**Supplemental Figure XX. Overall survival by TET2 or IDH1/2 mutation status.** Hazard ratio and confidence intervals were estimated using Cox proportional hazards models on data stratified by TET2 and IDH1/2 mutation status. Survival curves and estimates are consistent with the primary outcome.

```{r, fig.width=14, fig.height=9}

fit1 = coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm*TET2,data=df)
fit2 = coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm+TET2,data=df)

knitr::kable(lmtest::lrtest(fit1,fit2),caption = "LRT for interaction")

fit_os.no.tet = survfit(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$TET2 == "TET2neg",])

p1 = ggsurvplot(fit_os.no.tet,
           legend.labs =gsub("TET2=","", gsub("Treatment_arm=","",names(fit_os.no.tet$strata))),
           palette=c("black",pal_nejm()(4)[c(3)]),
           cumevents=F,
           risk.table="nrisk_cumcensor",
           break.time.by=365.25,
           xscale="d_m",
           xlab="Months from randomisation",
           pval.method = F,
           pval=F,legend.title="TET2 Negative")

fit_os.tet = survfit(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$TET2 == "TET2Mut",])

p2 = ggsurvplot(fit_os.tet,
           legend.labs =gsub("TET2=","", gsub("Treatment_arm=","",names(fit_os.tet$strata))),
palette=c("black",pal_nejm()(4)[c(3)]),
cumevents=F,
           risk.table="nrisk_cumcensor",
           break.time.by=365.25,
           xscale="d_m",
           xlab="Months from randomisation",
           pval.method = F,
           pval=F,legend.title="TET2 Positive")

p.y1 = min(p1$data.survplot[p1$data.survplot$Treatment_arm != "Placebo", ]$surv)
p.c1 = min(p1$data.survplot[p1$data.survplot$Treatment_arm == "Placebo", ]$surv)

p.y2 = min(p2$data.survplot[p2$data.survplot$Treatment_arm != "Placebo", ]$surv)
p.c2 = min(p2$data.survplot[p2$data.survplot$Treatment_arm == "Placebo", ]$surv)



p1$plot = p1$plot + ggtitle(expression(bold("No" ~ bolditalic("TET2") ~ or ~ bolditalic("IDH1/2") ~ "mutations"))) + theme(legend.position = "none") +theme(legend.position = "none") +
  annotate(geom="text",x = 6*365.25,y = p.y1,label="Vitamin C", color=pal_nejm()(4)[c(3)],hjust=0) + 
  annotate(geom="text",x = 6*365.25,y = p.c1,label="Placebo", color="black",hjust=0) + 
  scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+
  coord_cartesian(clip="off")+theme_prism()+theme(legend.position = "none")+
  geom_hline(yintercept = 0.5,linetype="dotted",linewidth = 1.25) +theme(legend.position = "none",plot.margin = unit(c(0, 1, 0, 0), "inches"))  

HR.no = summary(coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$TET2 == "TET2neg",]))$conf.int

HR.no.anno = paste0("HR ",round(HR.no[1],2), "\n95% CI ",round(HR.no[3],2),", ",round(HR.no[4],2))

p1$plot = p1$plot + annotate(geom="text", x = 365,y = 0.25, label=HR.no.anno,hjust=0,size=4.5)

p2$plot = p2$plot + ggtitle(expression(bold(bolditalic("TET2") ~ or ~ bolditalic("IDH1/2") ~ "mutation detected"))) + theme(legend.position = "none") + theme(legend.position = "none") +theme(legend.position = "none") +
  annotate(geom="text",x = 6*365.25,y = p.y2,label="Vitamin C", color=pal_nejm()(4)[c(3)],hjust=0) + 
  annotate(geom="text",x = 6*365.25,y = p.c2,label="Placebo", color="black",hjust=0) + 
  scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+
  coord_cartesian(clip="off")+theme_prism()+theme(legend.position = "none")+
  geom_hline(yintercept = 0.5,linetype="dotted",linewidth = 1.25) +theme(legend.position = "none",plot.margin = unit(c(0, 1, 0, 0), "inches")) 


HR.tet = summary(coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$TET2 == "TET2Mut",]))$conf.int

HR.tet.anno = paste0("HR ",round(HR.tet[1],2), "\n95% CI ",round(HR.tet[3],2),", ",round(HR.tet[4],2))

p2$plot = p2$plot + annotate(geom="text", x = 365,y = 0.25, label=HR.tet.anno,hjust=0,size=4.5)


p1$table = p1$table +theme_prism()+ theme(axis.line = element_blank())+scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+coord_cartesian(clip="off")+theme(text = element_text(face="bold",family=""),axis.text.y = element_text(hjust=1,colour = c(pal_nejm()(4)[c(3)],"black"),margin= margin(0, 15, 0, 7) ),axis.ticks.y = element_blank(),plot.margin = unit(c(0, 0, 0, 0), "inches"))+xlab("") + ggtitle("") +ylab("Number at risk\n(Cumulative censoring)") + theme(axis.text.x = element_blank(),axis.ticks.x = element_blank(),plot.title = element_blank())

p2$table = p2$table +theme_prism()+ theme(axis.line = element_blank())+scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+coord_cartesian(clip="off")+theme(text = element_text(face="bold",family=""),axis.text.y = element_text(hjust=1,colour = c(pal_nejm()(4)[c(3)],"black"),margin= margin(0, 15, 0, 7) ),axis.ticks.y = element_blank(),plot.margin = unit(c(0, 0, 0, 0), "inches"))+xlab("") + ggtitle("") +ylab("Number at risk\n(Cumulative censoring)") + theme(axis.text.x = element_blank(),axis.ticks.x = element_blank(),plot.title = element_blank())

p1$plot  + p2$plot + p1$table +p2$table +plot_spacer()+plot_spacer() +plot_layout(ncol=2,heights = c(1,0.25,0.02))

ggsave("~/bbc-secondary/research/JONP_20240430_vitc_trial_VBCS/survival_figure_b.pdf",width=18,height=9.5)

p.fig.3b = p1$plot  + p2$plot + p1$table +p2$table +plot_layout(ncol=2,heights = c(1,0.25))
```

### Manuscript Figure

```{r, fig.width=20, fig.height=9}

cowplot::plot_grid(p.fig.3a,p.fig.3b,ncol=2,scale=0.9,labels=c('a','b'),label_size=15,rel_widths = c(0.5,1))

ggsave("~/bbc-secondary/research/JONP_20240430_vitc_trial_VBCS/survival_figure.pdf",width=26,height=9.5)


```

#### LOF TET2

AKA TET2 hypermethylation vs WT signature

```{r, fig.width=14, fig.height=9}

lof.tet2 = read.csv("/varidata/research/projects/jones/computerome/250306_SU2C_Vitamin_C_TET2_meth_signatures_categories.csv")

lof.tet2$Patient_ID = gsub("AAL0","AAL",lof.tet2$Patient_ID)
df$lof.tet2 = rep(NA)
df$lof.tet2[gsub("-","",df$ID) %in% unique(lof.tet2$Patient_ID[lof.tet2$Timepoint == "Baseline" & lof.tet2$TET2_Meth_Signature == "High"])] = "TET2 hypermethylation signature"

df$lof.tet2[gsub("-","",df$ID) %in% unique(lof.tet2$Patient_ID[lof.tet2$Timepoint == "Baseline" & lof.tet2$TET2_Meth_Signature == "Low"])] = "Wildtype"

fit1 = coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm*lof.tet2,data=df)
fit2 = coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm+lof.tet2,data=df)

knitr::kable(lmtest::lrtest(fit1,fit2),caption = "LRT for interaction")

fit_os.no.tet = survfit(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$lof.tet2 == "Wildtype",])

p1 = ggsurvplot(fit_os.no.tet,
           legend.labs =gsub("TET2=","", gsub("Treatment_arm=","",names(fit_os.no.tet$strata))),
           palette=c("black",pal_nejm()(4)[c(3)]),
           cumevents=F,
           risk.table="nrisk_cumcensor",
           break.time.by=365.25,
           xscale="d_m",
           xlab="Months from randomisation",
           pval.method = F,
           pval=F,legend.title="Low TET2")

fit_os.tet = survfit(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$lof.tet2 == "TET2 hypermethylation signature",])

p2 = ggsurvplot(fit_os.tet,
           legend.labs =gsub("TET2=","", gsub("Treatment_arm=","",names(fit_os.tet$strata))),
palette=c("black",pal_nejm()(4)[c(3)]),
cumevents=F,
           risk.table="nrisk_cumcensor",
           break.time.by=365.25,
           xscale="d_m",
           xlab="Months from randomisation",
           pval.method = F,
           pval=F,legend.title="High TET2")

p.y1 = min(p1$data.survplot[p1$data.survplot$Treatment_arm != "Placebo", ]$surv)
p.c1 = min(p1$data.survplot[p1$data.survplot$Treatment_arm == "Placebo", ]$surv)

p.y2 = min(p2$data.survplot[p2$data.survplot$Treatment_arm != "Placebo", ]$surv)
p.c2 = min(p2$data.survplot[p2$data.survplot$Treatment_arm == "Placebo", ]$surv)



p1$plot = p1$plot + ggtitle("TET2 wildtype methylation signature") + theme(legend.position = "none") +theme(legend.position = "none") +
  annotate(geom="text",x = 6*365.25,y = p.y1,label="Vitamin C", color=pal_nejm()(4)[c(3)],hjust=0) + 
  annotate(geom="text",x = 6*365.25,y = p.c1,label="Placebo", color="black",hjust=0) + 
  scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+
  coord_cartesian(clip="off")+theme_prism()+theme(legend.position = "none")+
  geom_hline(yintercept = 0.5,linetype="dotted",linewidth = 1.25) +theme(legend.position = "none",plot.margin = unit(c(0, 1, 0, 0), "inches"))  

HR.no = summary(coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$lof.tet2 == "Wildtype",]))$conf.int

HR.no.anno = paste0("HR ",round(HR.no[1],2), "\n95% CI ",round(HR.no[3],2),", ",round(HR.no[4],2))

p1$plot = p1$plot + annotate(geom="text", x = 365,y = 0.25, label=HR.no.anno,hjust=0,size=4.5)

p2$plot = p2$plot + ggtitle(expression(bold("TET2"^bolditalic(mut) ~"hypermethylation signature"))) + theme(legend.position = "none") + theme(legend.position = "none") +theme(legend.position = "none") +
  annotate(geom="text",x = 6*365.25,y = p.y2,label="Vitamin C", color=pal_nejm()(4)[c(3)],hjust=0) + 
  annotate(geom="text",x = 6*365.25,y = p.c2,label="Placebo", color="black",hjust=0) + 
  scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+
  coord_cartesian(clip="off")+theme_prism()+theme(legend.position = "none")+
  geom_hline(yintercept = 0.5,linetype="dotted",linewidth = 1.25) +theme(legend.position = "none",plot.margin = unit(c(0, 1, 0, 0), "inches") ) 

HR.tet = summary(coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$lof.tet2 == "TET2 hypermethylation signature",]))$conf.int

HR.tet.anno = paste0("HR ",round(HR.tet[1],2), "\n95% CI ",round(HR.tet[3],2),", ",round(HR.tet[4],2))

p2$plot = p2$plot + annotate(geom="text", x = 365,y = 0.25, label=HR.tet.anno,hjust=0,size=4.5)


p1$table = p1$table +theme_prism()+ theme(axis.line = element_blank())+scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+coord_cartesian(clip="off")+theme(text = element_text(face="bold",family=""),axis.text.y = element_text(hjust=1,colour = c(pal_nejm()(4)[c(3)],"black"),margin= margin(0, 15, 0, 7) ),axis.ticks.y = element_blank(),plot.margin = unit(c(0, 0, 0, 0), "inches"))+xlab("") + ggtitle("") +ylab("Number at risk\n(Cumulative censoring)") + theme(axis.text.x = element_blank(),axis.ticks.x = element_blank(),plot.title = element_blank())

p2$table = p2$table +theme_prism()+ theme(axis.line = element_blank())+scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+coord_cartesian(clip="off")+theme(text = element_text(face="bold",family=""),axis.text.y = element_text(hjust=1,colour = c(pal_nejm()(4)[c(3)],"black"),margin= margin(0, 15, 0, 7) ),axis.ticks.y = element_blank(),plot.margin = unit(c(0, 0, 0, 0), "inches"))+xlab("") + ggtitle("") +ylab("Number at risk\n(Cumulative censoring)") + theme(axis.text.x = element_blank(),axis.ticks.x = element_blank(),plot.title = element_blank())

p1$plot  + p2$plot + p1$table +p2$table +plot_layout(ncol=2,heights = c(1,0.25))


```

#### Above/Below median Vitamin C

**Supplemental figure XX. Overall survival by vitamin C above or below median vitamin C at inclusion.** Hazard ratio and confidence intervals were estimated using Cox proportional hazards models on data stratified by patients with plasma vitamin C level below the median measure at inclusion, and greater than or equal to the median. Results are consistent with the primary outcome.  

```{r, fig.width=14, fig.height=9}

df$med_vitc = ifelse(df$VitC_PB1 >= median(df$VitC_PB1,na.rm=T),"above","below")

fit1 = coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm*med_vitc,data=df)
fit2 = coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm+med_vitc,data=df)
knitr::kable(lmtest::lrtest(fit1,fit2),caption = "LRT for interaction")

fit_os.vitc.be = survfit(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$med_vitc == "below",])

p1 = ggsurvplot(fit_os.vitc.be,
           legend.labs =gsub("TET2=","", gsub("Treatment_arm=","",names(fit_os.vitc.be$strata))),
           palette=c("black",pal_nejm()(4)[c(3)]),
           cumevents=F,
           risk.table="nrisk_cumcensor",
           break.time.by=365.25,
           xscale="d_m",
           xlab="Months from randomisation",
           pval.method = F,
           pval=F,legend.title="TET2 Negative")



fit_os.vitc.a = survfit(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$med_vitc == "above",])

p2 = ggsurvplot(fit_os.vitc.a,
           legend.labs =gsub("TET2=","", gsub("Treatment_arm=","",names(fit_os.vitc.a$strata))),
palette=c("black",pal_nejm()(4)[c(3)]),
cumevents=F,
           risk.table="nrisk_cumcensor",
           break.time.by=365.25,
           xscale="d_m",
           xlab="Months from randomisation",
           pval.method = F,
           pval=F,legend.title="TET2 Positive")

p.y1 = min(p1$data.survplot[p1$data.survplot$Treatment_arm != "Placebo", ]$surv)
p.c1 = min(p1$data.survplot[p1$data.survplot$Treatment_arm == "Placebo", ]$surv)

p.y2 = min(p2$data.survplot[p2$data.survplot$Treatment_arm != "Placebo", ]$surv)
p.c2 = min(p2$data.survplot[p2$data.survplot$Treatment_arm == "Placebo", ]$surv)

p1$plot = p1$plot + ggtitle("Below the median vitamin C level at inclusion") + theme(legend.position = "none") +theme(legend.position = "none") +
  annotate(geom="text",x = 6*365.25,y = p.y1,label="Vitamin C", color=pal_nejm()(4)[c(3)],hjust=0) + 
  annotate(geom="text",x = 6*365.25,y = p.c1,label="Placebo", color="black",hjust=0) + 
  scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+
  coord_cartesian(clip="off")+theme_prism()+theme(legend.position = "none")+
  geom_hline(yintercept = 0.5,linetype="dotted",linewidth=1.25) +theme(legend.position = "none",plot.margin = unit(c(0, 1, 0, 0), "inches")) 



HR.a = summary(coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$med_vitc == "below",]))$conf.int
HR.a.anno = paste0("HR ",round(HR.a[1],2), "\n95% CI ",round(HR.a[3],2),", ",round(HR.a[4],2))
p1$plot = p1$plot + annotate(geom="text", x = 365,y = 0.25, label=HR.a.anno,hjust=0,size=4.5)


p2$plot = p2$plot + ggtitle("At or above the median vitamin C level at inclusion") + theme(legend.position = "none") + theme(legend.position = "none") +theme(legend.position = "none") +
  annotate(geom="text",x = 6*365.25,y = p.y2,label="Vitamin C", color=pal_nejm()(4)[c(3)],hjust=0) + 
  annotate(geom="text",x = 6*365.25,y = p.c2,label="Placebo", color="black",hjust=0) + 
  scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+
  coord_cartesian(clip="off")+theme_prism()+theme(legend.position = "none")+
  geom_hline(yintercept = 0.5,linetype="dotted",linewidth=1.25) +theme(legend.position = "none",plot.margin = unit(c(0, 1, 0, 0), "inches")) 

HR.a = summary(coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$med_vitc == "above",]))$conf.int
HR.a.anno = paste0("HR ",round(HR.a[1],2), "\n95% CI ",round(HR.a[3],2),", ",round(HR.a[4],2))
p2$plot = p2$plot + annotate(geom="text", x = 365,y = 0.25, label=HR.a.anno,hjust=0,size=4.5)

p1$table = p1$table +theme_prism()+ theme(axis.line = element_blank())+scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+coord_cartesian(clip="off")+theme(text = element_text(face="bold",family=""),axis.text.y = element_text(hjust=1,colour = c(pal_nejm()(4)[c(3)],"black"),margin= margin(0, 15, 0, 7) ),axis.ticks.y = element_blank(),plot.margin = unit(c(0, 0, 0, 0), "inches"))+xlab("") + ggtitle("") +ylab("Number at risk\n(Cumulative censoring)") + theme(axis.text.x = element_blank(),axis.ticks.x = element_blank(),plot.title = element_blank())


p2$table = p2$table +theme_prism()+ theme(axis.line = element_blank())+scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+coord_cartesian(clip="off")+theme(text = element_text(face="bold",family=""),axis.text.y = element_text(hjust=1,colour = c(pal_nejm()(4)[c(3)],"black"),margin= margin(0, 15, 0, 7) ),axis.ticks.y = element_blank(),plot.margin = unit(c(0, 0, 0, 0), "inches"))+xlab("") + ggtitle("") +ylab("Number at risk\n(Cumulative censoring)") + theme(axis.text.x = element_blank(),axis.ticks.x = element_blank(),plot.title = element_blank())

p1$plot  + p2$plot +p1$table +p2$table +plot_layout(ncol=2,heights = c(1,0.25))


```

#### Diagnostic subgroup

**Supplemental figure XX. Overall survival by diagnostic subgroup at inclusion.** Hazard ratio and confidence intervals were estimated using Cox proportional hazards models on data stratified by diagnosis at inclusion. CCUS appears to be a subgroup that can benefit particularly well from vitamin C supplementation. 

```{r, fig.width=18,fig.height=7.75}



fit1 = coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm*Diagnosis_simple_baseline,data=df)
fit2 = coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm+Diagnosis_simple_baseline,data=df)
knitr::kable(lmtest::lrtest(fit1,fit2),caption = "LRT for interaction")


fit_os.diags = survfit(Surv(Days_FU_OS,FU_Death)~Treatment_arm+Diagnosis_simple_combo,data=df)


p.a = ggsurvplot(fit_os.diags,
           legend.labs =gsub("Diagnosis_simple_combo=","",gsub("Treatment_arm=","",names(fit_os.diags$strata))),
           palette=c("black",pal_nejm()(7)[c(5,2,3,1,7)]),
           cumevents=F,
           risk.table=F,
           break.time.by=365.25,
           xscale="d_m",
           xlab="Months from randomisation",
           pval=T,pval.method = T,
           title= "All patients",legend.title="")

df$Sex.c = ifelse(df$Sex==1,"Male","Female")

fit_os.CCUS = survfit(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$Diagnosis_simple_combo=="CCUS",])

p1 = ggsurvplot(fit_os.CCUS,
           legend.labs =gsub("Treatment_arm=","",names(fit_os.CCUS$strata)),
           palette=c("black",pal_nejm()(7)[c(3)]),
           cumevents=F,
           risk.table="nrisk_cumcensor",
           break.time.by=365.25,
           xscale="d_m",
           xlab="Months from randomisation",
           pval=F,pval.method = F,legend.title="CCUS")



fit_os.mds = survfit(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$Diagnosis_simple_combo=="MDS",])

p2 = ggsurvplot(fit_os.mds,
           legend.labs =gsub("Treatment_arm=","",names(fit_os.mds$strata)),
           palette=c("black",pal_nejm()(7)[c(3)]),
           cumevents=F,
           risk.table="nrisk_cumcensor",
           break.time.by=365.25,
           xscale="d_m",
           xlab="Months from randomisation",
           pval=F,pval.method = F,legend.title="MDS")



fit_os.CMMLplus = survfit(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$Diagnosis_simple_combo=="CMML_MDS_MPN",])

p3 = ggsurvplot(fit_os.CMMLplus,
           legend.labs =gsub("Treatment_arm=","",names(fit_os.CMMLplus$strata)),
           palette=c("black",pal_nejm()(7)[c(3)]),
           cumevents=F,
           risk.table="nrisk_cumcensor",
           break.time.by=365.25,
           xscale="d_m",
           xlab="Months from randomisation",
           pval=F,pval.method = F,legend.title="CMMLplus")

p.c.ccus = min(p1$data.survplot$surv[p1$data.survplot$Treatment_arm == "Vitamin C"])
p.p.ccus = min(p1$data.survplot$surv[p1$data.survplot$Treatment_arm != "Vitamin C"])

p1$plot = p1$plot +theme(legend.position = "none") +
  annotate(geom="text",x = 6*365.25,y = p.c.ccus,label="Vitamin C", color=pal_nejm()(4)[c(3)],hjust=0) + 
  annotate(geom="text",x = 6*365.25,y = p.p.ccus,label="Placebo", color="black",hjust=0) + 
  scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+
  coord_cartesian(clip="off")+theme_prism()+theme(legend.position = "none")+
  geom_hline(yintercept = 0.5,linetype="dotted",linewidth=1.25) +theme(legend.position = "none",plot.margin = unit(c(0, 1, 0, 0), "inches")) + ggtitle("CCUS")


HR.ccus = summary(coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$Diagnosis_simple_combo=="CCUS",]))$conf.int
HR.ccus.anno = paste0("HR ",ifelse(round(HR.ccus[1],2)<0.01,"< 0.01",round(HR.ccus[1],2)), "\n95% CI ",round(HR.ccus[3],2),", ",round(HR.ccus[4],2))
p1$plot = p1$plot + annotate(geom="text", x = 365,y = 0.25, label=HR.ccus.anno,hjust=0,size=4.5)


p.c.ccus = min(p2$data.survplot$surv[p2$data.survplot$Treatment_arm == "Vitamin C"])
p.p.ccus = min(p2$data.survplot$surv[p2$data.survplot$Treatment_arm != "Vitamin C"])

p2$plot = p2$plot +theme(legend.position = "none") +
  annotate(geom="text",x = 6*365.25,y = p.c.ccus,label="Vitamin C", color=pal_nejm()(4)[c(3)],hjust=0) + 
  annotate(geom="text",x = 6*365.25,y = p.p.ccus,label="Placebo", color="black",hjust=0) + 
  scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+
  coord_cartesian(clip="off")+theme_prism()+theme(legend.position = "none")+
  geom_hline(yintercept = 0.5,linetype="dotted",linewidth=1.25) +theme(legend.position = "none",plot.margin = unit(c(0, 1, 0, 0), "inches"))+ ggtitle("MDS") 


HR.MDS = summary(coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$Diagnosis_simple_combo=="MDS",]))$conf.int
HR.MDS.anno = paste0("HR ",round(HR.MDS[1],2), "\n95% CI ",round(HR.MDS[3],2),", ",round(HR.MDS[4],2))
p3$plot = p3$plot + annotate(geom="text", x = 365,y = 0.25, label=HR.MDS.anno,hjust=0,size=4.5)


p.c.ccus = min(p3$data.survplot$surv[p3$data.survplot$Treatment_arm == "Vitamin C"])
p.p.ccus = min(p3$data.survplot$surv[p3$data.survplot$Treatment_arm != "Vitamin C"])

p3$plot = p3$plot +theme(legend.position = "none") +
  annotate(geom="text",x = 6*365.25,y = p.c.ccus,label="Vitamin C", color=pal_nejm()(4)[c(3)],hjust=0) + 
  annotate(geom="text",x = 5*365.25,y = p.p.ccus,label="Placebo", color="black",hjust=0) + 
  scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+
  coord_cartesian(clip="off")+theme_prism()+theme(legend.position = "none")+
  geom_hline(yintercept = 0.5,linetype="dotted",linewidth=1.25) +theme(legend.position = "none",plot.margin = unit(c(0, 1, 0, 0), "inches"))+ ggtitle("MDS/MPN") 


HR.mds.mpn = summary(coxph(Surv(Days_FU_OS,FU_Death)~Treatment_arm,data=df[df$Diagnosis_simple_combo=="CMML_MDS_MPN",]))$conf.int

HR.mds.mpn.anno = paste0("HR ",round(HR.mds.mpn[1],2), "\n95% CI ",round(HR.mds.mpn[3],2),", ",round(HR.mds.mpn[4],2))
p2$plot = p2$plot + annotate(geom="text", x = 365,y = 0.25, label=HR.mds.mpn.anno,hjust=0,size=4.5)


p1$table = p1$table +theme_prism()+ theme(axis.line = element_blank())+scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+coord_cartesian(clip="off")+theme(text = element_text(face="bold",family=""),axis.text.y = element_text(hjust=1,colour = c(pal_nejm()(4)[c(3)],"black"),margin= margin(0, 15, 0, 7) ),axis.ticks.y = element_blank(),plot.margin = unit(c(0, 0, 0, 0), "inches"))+xlab("") + ggtitle("") +ylab("Number at risk\n(Cumulative censoring)") + theme(axis.text.x = element_blank(),axis.ticks.x = element_blank(),plot.title = element_blank())

p2$table = p2$table +theme_prism()+ theme(axis.line = element_blank())+scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+coord_cartesian(clip="off")+theme(text = element_text(face="bold",family=""),axis.text.y = element_text(hjust=1,colour = c(pal_nejm()(4)[c(3)],"black"),margin= margin(0, 15, 0, 7) ),axis.ticks.y = element_blank(),plot.margin = unit(c(0, 0, 0, 0), "inches"))+xlab("") + ggtitle("") +ylab("Number at risk\n(Cumulative censoring)") + theme(axis.text.x = element_blank(),axis.ticks.x = element_blank(),plot.title = element_blank())

p3$table = p3$table +theme_prism()+ theme(axis.line = element_blank())+scale_x_continuous(limits=c(0,365.25*6), breaks = seq(0,5*365.25,365.25),labels = seq(0,60,12),expand=c(0,0))+coord_cartesian(clip="off")+theme(text = element_text(face="bold",family=""),axis.text.y = element_text(hjust=1,colour = c(pal_nejm()(4)[c(3)],"black"),margin= margin(0, 15, 0, 7) ),axis.ticks.y = element_blank(),plot.margin = unit(c(0, 0, 0, 0), "inches"))+xlab("") + ggtitle("") +ylab("Number at risk\n(Cumulative censoring)") + theme(axis.text.x = element_blank(),axis.ticks.x = element_blank(),plot.title = element_blank())

p1$plot + p2$plot + p3$plot +
  p1$table + p2$table + p3$table + plot_layout(ncol=3,heights=c(1,0.4))


```


## Competing risk regression (progression)

For this analysis death is a competing risk of high-risk progression. We are a bit limited in sample size and events; there are only 14 progression events. Power to detect any differences in progression is quite low and we will have difficulty adjusting for any covariates here. We see corroboration with out overall survival findings that death (the dotted lines) are significantly different, however we do not have enough evidence to determine if progression also differs (p =  0.22).

Because competing risks comes from a different package, I can't seem to add a y-axis title to the risk table

```{r,fig.width=11.5, fig.height=9}

df$crevent = ifelse(df$HR_FU_Progression == 1,1,ifelse(df$FU_Death==1,2,0))

df$crevent=factor(df$crevent,levels=c(0,1,2),labels=c("Censor","Higher-risk progression","Death"))
df$Months = 12*df$HR_Days_FU_progression/365.25
df$tx = ifelse(df$Treatment_arm == "Vitamin C", "Vit_c","Placebo")

CI.tx = tidycmprsk::cuminc(Surv( Months ,crevent)~Treatment_arm,data=df)
CI.tx$cmprsk = timepoints(CI.tx$cmprsk,times=seq(0,60,12))

dth = tidycmprsk::crr(Surv( Months ,crevent)~VitaminC,data=df,failcode = "Death")
hrp = tidycmprsk::crr(Surv( Months ,crevent)~VitaminC,data=df,failcode = "Higher-risk progression")

p.ci = ggcuminc(CI.tx,outcome=c("Higher-risk progression","Death"),linewidth=1)+theme_prism()+scale_color_manual(values=c("black",pal_nejm()(4)[3]))+scale_y_continuous(limits=c(0,0.4))+
  annotate(geom="text",x = 65,y = max(CI.tx$tidy$estimate[CI.tx$tidy$strata == "Placebo" & CI.tx$tidy$outcome == "Death"],na.rm=T),label="Death, Placebo",hjust=0,size=4.5)+
  annotate(geom="text",size=4.5,x = 65,y = max(CI.tx$tidy$estimate[CI.tx$tidy$strata == "Placebo" & CI.tx$tidy$outcome == "Higher-risk progression"],na.rm=T)+0.005,label="Higher-risk progression, Placebo",hjust=0)+
  coord_cartesian(clip =  "off")+theme(legend.position = "none",plot.margin = unit(c(0, 1, 0, 0), "inches")) +
  annotate(geom="text",size=4.5,x = 65,y = max(CI.tx$tidy$estimate[CI.tx$tidy$strata == "Vitamin C" & CI.tx$tidy$outcome == "Death"],na.rm=T)-0.005,label="Death, Vitamin C",hjust=0,color=pal_nejm()(4)[3])+
  annotate(geom="text",size=4.5,x = 65,y = max(CI.tx$tidy$estimate[CI.tx$tidy$strata == "Vitamin C" & CI.tx$tidy$outcome == "Higher-risk progression"],na.rm=T),label="Higher-risk progression, Vitamin C",hjust=0,color=pal_nejm()(4)[3]) + ggtitle("Competing risks analysis")+
  xlab("Months from randomisation")+theme(legend.position = "none",plot.margin = unit(c(0, 2.75, 0, 0), "inches")) +scale_linetype_manual(values=c("dashed","solid","dashed","solid"))+ annotate(geom="text", x = 4,y = 0.375, label=paste0("Higher-risk progression\np =  0.22\nHR ",round(exp(hrp$coefs),2),"; 95% CI ", round(exp(tidy(hrp,conf.int=T)$conf.low),2),", ", round(exp(tidy(hrp,conf.int=T)$conf.high),2) ),fontface="italic",hjust=0,size=4.5)+
  annotate(geom="text", x = 4,y = 0.29, label=paste0("Death\np =  ",tidy(dth,conf.int=T)$p.value,"\nHR ",round(exp(dth$coefs),2),"; 95% CI ", round(exp(tidy(dth,conf.int=T)$conf.low),2),", ", round(exp(tidy(dth,conf.int=T)$conf.high),2) ),fontface="italic",hjust=0,size=4.5)+
  scale_x_continuous(limits=c(0,65),breaks = seq(0,60,12),labels = seq(0,60,12),expand=c(0.0,0))+ylab("Cumulative incidence") +coord_cartesian(clip="off")

my_theme = theme_prism()+ theme(axis.line = element_blank(),text = element_text(face="bold",family=""),axis.text.y = element_text(hjust=1,colour = c(pal_nejm()(4)[c(3)],"black"),margin= margin(0, 22, 0,16 ) ),axis.ticks.y = element_blank(),plot.margin = unit(c(0, 0, 0, 0), "inches"),axis.ticks.x = element_blank(),axis.title.x = element_blank(),plot.title=element_blank(),axis.text.x = element_blank())

p.ci = p.ci + add_risktable(times = seq(0,60,12),
                     risktable_stats = "{n.risk} ({cum.censor})",
                     risktable_group = "risktable_stats",
                     theme=my_theme,
                     size = 5) +
 theme(legend.position = "none",plot.margin = unit(c(0, 2.75, 0, 2), "inches"))

p.grobs = ggsurvfit_build(p.ci ,combine_plots = F)
## To get the formatting to match the KM plots, we need to manually modify the grobs, this doesn't impact the data, just alignments of labels
p.grobs[[2]]$layout$clip = rep("off")
p.grobs[[2]]$grobs[[13]][6]$children[[1]]$label = "Number at risk\n(Cumulative censoring)"

tbl.r = ggpubr::as_ggplot(p.grobs[[2]])

ggpubr::as_ggplot(p.grobs[[1]]) +tbl.r+plot_spacer()+plot_layout(ncol=1,heights=c(1,0.25,0.02))&coord_cartesian(clip="off")

ggsave("~/bbc-secondary/research/JONP_20240430_vitc_trial_VBCS/competing_risks.pdf",width=14,height=10)

```


## Global Methylation {.tabset}

**Figure XX. Global methylation changes over time by treatment arm** A) 5-mC/dG and B) 5-hmC/5-mC level for each patient at inclusion and end-of-therapy (EOT) and the change in these measures between inclusion and EOT. Data were analyzed using robust linear mixed-effects models with a random intercept for each patient and an interaction between treatment and time point to assess if measures changed differently over time. Models also included a covariate for batch.


```{r, fig.width=8, fig.height=5}

colnames(df.sp)[7:8] = c("Timepoint","Vitamin_C")
df.hm1 = as.data.frame( read_excel(unzip("/varidata/research/projects/jones/computerome/Vitc_Data.zip","EVI2_data_consolidated030524.xlsx"),sheet=2))
df.hm1 = df.hm1[,c(1,7,8,9:11)]
colnames(df.hm1) = c("ID", "mC","hmC","hmC/mC","mC/dG","hmC/dG")

df.hm2 = as.data.frame( read_excel("/varidata/research/projects/jones/computerome/Vitc_Data/MC00600_QuantCombined 112524 ST edits.xlsx",sheet=6,skip=1))
df.hm2 = df.hm2[,c(1,5,7,9:11)]
colnames(df.hm2) = c("ID", "hmC","mC","hmC/dG","mC/dG","hmC/mC")
df.hm1$Batch=1
df.hm2$Batch = 2
df.hm = rbind(df.hm1,df.hm2)

##These were mistakenly changed to 5 as EOT, they are not and we should model these as their respective times
df.hm$ID[match(c("1-5","12-5","33-5","25-5","44-5","78-5"),df.hm$ID )] = 
 c("1-2","12-2","33-2","25-2","44-2","78-4")

df.hm$Time = as.numeric(gsub(".*-","",df.hm$ID))
df.hm$Time = factor(df.hm$Time,levels=seq(1:5),labels=c(0,3,6,9,12))
df.hm$ID = gsub("AAL0","AAL",gsub("-.*","",df.hm$ID))
df.sp$ID = gsub("AAL0","AAL",gsub("-","",df.sp$ID))
df.merge = left_join(df.hm,df.sp,by= c("ID","Time"))
colnames(df.merge) = make.names(colnames(df.merge))
df.merge= df.merge[!is.na(df.merge$Time),]
df.merge$Timepoint  = ifelse(df.merge$Time == 0, "Baseline","EOT")

knitr::kable(unique(lof.tet2$Patient_ID[which(! lof.tet2$Patient_ID %in% df.merge$ID)]), caption = "IDs not in Mass Spec data")

knitr::kable(unique(df.merge$ID[which(! df.merge$ID %in% lof.tet2$Patient_ID)]), caption = "IDs not in methylation data")

```


```{r, fig.width=8, fig.height=5}

svr = df$ID[df$VitC_PB1 <=11.4]
df$vitc_group =  case_when(df$VitC_PB1 <23 ~ "Deficient", 
                            df$VitC_PB1 <50 ~ "Inadequate",
                            TRUE ~ "Adequate")

###loop through all subgroups 
subgrp.glbl = function(outcome,subgroup){
  EOT = subset(df.merge[df.merge$Timepoint == "EOT",], select=c(outcome,"Treatment_arm","ID","Sex","Vitamin_C"))
  Base = subset(df.merge[df.merge$Timepoint == "Baseline",], select=c(outcome,"Treatment_arm","ID","Sex","Vitamin_C"))
  colnames(EOT)[1] = "EOT"
  colnames(Base)[1] = "Base"
  
  colnames(EOT)[5] = "EOT.Vitc"
  colnames(Base)[5] = "Base.Vitc"
  
  wide = left_join(EOT,Base)
  
  if(subgroup == "TET2"){
    wide$grp = ifelse( wide$ID %in% mut_ids,"TET2 or IDH1/2 mutation","No TET2 or IDH1/2 mutation")
    cols = c("black","#6F99ADFF")
    
  } 
  
  if(subgroup == "lof.TET2"){
      wide = wide[wide$ID %in% lof.tet2$Patient_ID,]
      wide$grp = ifelse( wide$ID %in% lof.tet2$Patient_ID[lof.tet2$TET2_Meth_Signature == "Low" | is.na(lof.tet2$TET2_Meth_Signature) ] ,"No or Low", ifelse(wide$ID %in% lof.tet2$Patient_ID[lof.tet2$TET2_Meth_Signature == "Intermediate"],"Intermediate","High")) 
      wide=wide[wide$grp!="Intermediate",]
      wide$grp = factor(wide$grp,levels=c("No or Low","High"),
                        labels = c("TET2 WT signature","TET2 hypermethylation signature"))
    cols = c("black","#6F99ADFF")
  } 
  

  if(subgroup == "vitc_med"){
    wide$grp = ifelse(wide$ID %in% df$ID[df$med_vitc == "above"],"Above median","Below median")  
    wide$grp = factor(wide$grp,levels=c("Below median","Above median"))
    cols = c("#FFDC91FF","#E18727FF")
  }
  
  if(subgroup == "vitc_severe"){
    wide$grp = ifelse(wide$ID %in% svr ,"Severe (< 11.4 μmol/L)","All others")  
    wide$grp = factor(wide$grp,levels=c("Severe (< 11.4 μmol/L)","All others"))
    cols = c("grey","#E18727FF")
  }
  
  if(subgroup == "vitc_ccus"){
        wide$grp = ifelse(wide$ID %in% df$ID[df$Diagnosis_simple_combo == "CCUS"],"CCUS",
                                         ifelse(wide$ID %in% df$ID[df$Diagnosis_simple_combo == "MDS"],"MDS",
                                         ifelse(wide$ID %in% df$ID[df$Diagnosis_simple_combo == "CMML_MDS_MPN"],"MDS/MPN",NA)))
    wide=wide[!is.na(wide$grp),]    
    wide$grp = factor(wide$grp,
                      levels= c("CCUS","MDS","MDS/MPN" ))
    
        cols = values=c("CCUS" =  viridis::viridis(4)[1] ,
                        "MDS" = viridis::viridis(4)[2],
                        "MDS/MPN" = viridis::viridis(4)[3])
    
  }
  
  
  # autoplot(lm(as.numeric( EOT.mC.dG) - as.numeric(Base.mC.dG ) ~ Treatment_arm*TET2,  data=mc.dg.wide),1:6)
  
  if(subgroup=="All"){
    wide$grp="All"
  }
  print(table(wide$Treatment_arm,wide$grp))

  yl = case_when(outcome == "mC.dG" ~ "5-mC/dG",
                   outcome == "hmC.dG" ~ "5-hmC/dG",
                   outcome == "hmC.mC"~"5-hmC/5-mC")
    
  if(subgroup == "All"){
        p.c.b = ggplot(wide,aes(x = Base.Vitc,y = as.numeric(Base),color=Treatment_arm,fill = Treatment_arm )) + 
      theme_minimal()+
      geom_point(alpha=0.5) +
      stat_smooth(method="lm") + 
      stat_cor(method = "pearson",p.accuracy = 0.0001, r.accuracy = 0.001,
               aes(label = paste(..rr.label.., ..p.label.., sep = "~`,`~"))) +
      theme_prism() + scale_color_manual(values=c("black","#E18727FF"))+theme(legend.position="none")+xlab("")+scale_fill_manual(values=c("black","#E18727FF"))+
        ylab(paste0("Baseline ",yl)) +xlab("Baseline Vitamin C")
        print(p.c.b)
        
  } else {
    fit = rlm(as.numeric( EOT) - as.numeric(Base) ~ Treatment_arm*grp ,  data=wide)
    res = data.frame(summary(confint(emmeans(fit,pairwise~Treatment_arm|grp)$contrasts)))

    ymin = signif(0.975*min(as.numeric(c(wide$Base,wide$EOT)),na.rm=T),2)
    ymax = signif(1.025*max(as.numeric(c(wide$Base,wide$EOT)),na.rm=T),2)
    
    ybar = signif(1.1*quantile(as.numeric(c(wide$Base,wide$EOT)),na.rm=T,p=0.975),2)
    ybar = signif(quantile(as.numeric(c(wide$Base,wide$EOT)),na.rm=T,p=0.975),2) + 0.1*(ymin-ymax)

    fit.a = rlm(as.numeric(Base) ~ grp+Treatment_arm ,  data=wide[wide$grp != "Intermediate",])
    p.1 = summary(emmeans(fit.a,pairwise~grp)$contrasts)
    # label1 = ifelse(p.1$p.value<0.0001, "p < 0.0001", paste0("p = ",signif(p.1$p.value,2)))
    ci.1 = confint(emmeans(fit.a,pairwise~grp)$contrasts)
    label1 = paste0("Mean \U0394 ",-1*signif(round(ci.1$estimate,4),2),"\n95% CI ",
                    -1*signif(round(ci.1$asymp.UCL,4),2),", ",
                    -1*signif(round(ci.1$asymp.LCL,4),2))
    
    print(p.1)
    summary(fit.a)
    
    fit.b = rlm(as.numeric(EOT) ~ grp+Treatment_arm ,  data=wide[wide$grp != "Intermediate",])
    p.2 = summary(emmeans(fit.b,pairwise~grp)$contrasts)
    # label2 = ifelse(p.2$p.value<0.0001, "p < 0.0001", paste0("p = ",signif(p.2$p.value,2)))
    ci.2 = confint(emmeans(fit.b,pairwise~grp)$contrasts)
    label2 = paste0("Mean \U0394 ",-1*signif(round(ci.2$estimate,4),2),"\n95% CI ",
                    -1*signif(round(ci.2$asymp.UCL,4),2),", ",
                    -1*signif(round(ci.2$asymp.LCL,4),2))
    
    
    print(p.2)
    summary(fit.b)
    p.e.a = ggplot(wide[wide$grp != "Intermediate",],aes(x = grp,y = as.numeric(EOT),color=grp )) + 
      geom_quasirandom() +
      geom_boxplot(fill=NA,outlier.size = -1) +
      theme_prism(axis_text_angle = 45) + scale_color_manual(values=cols)+theme(legend.position="none")+xlab("")+
        ylab(paste0("End ",yl)) +scale_y_continuous(limits=c(ymin,max(ymax,ybar*1.15))) 
    
    
    p.b.a = ggplot(wide[wide$grp != "Intermediate",],aes(x = grp,y = as.numeric(Base),color=grp )) + 
      geom_quasirandom() +
      geom_boxplot(fill=NA,outlier.size = -1) +
      theme_prism(axis_text_angle = 45) + scale_color_manual(values=cols)+theme(legend.position="none")+xlab("")+
        ylab(paste0("Baseline ",yl)) +scale_y_continuous(limits=c(ymin,max(ymax,ybar*1.15))) 
    
  
     
    
    if(subgroup != "vitc_ccus"){
     p.b.a = p.b.a + annotate(geom = "text",x=1.5,y = ybar*1.1,label = label1,fontface="italic",size=5) + theme(axis.text.x = element_text(angle=30,hjust=1))+
      annotate(geom = "line", y= ybar*1.05, x=c(1,2))
  
     p.e.a = p.e.a + annotate(geom = "text",x=1.5,y = ybar*1.1,label = label2,fontface="italic",size=5) + theme(axis.text.x = element_text(angle=30,hjust=1))+
      annotate(geom = "line", y= ybar*1.05, x=c(1,2))
  
    } else{
    p.b.a = p.b.a + 
        annotate(geom = "text",x=1.5,y = ybar*1.1,label = label1[1],fontface="italic",size=5) + theme(axis.text.x = element_text(angle=30,hjust=1))+
      annotate(geom = "line", y= ybar*1.05, x=c(1,1.9)) +
      
        annotate(geom = "text",x=2.5,y = ybar*1.1,label = label1[3],fontface="italic",size=5) + theme(axis.text.x = element_text(angle=30,hjust=1))+
      annotate(geom = "line", y= ybar*1.05, x=c(2.1,3)) +
      
        annotate(geom = "text",x=2,y = ybar*1.2,label = label1[2],fontface="italic",size=5) + theme(axis.text.x = element_text(angle=30,hjust=1))+
      annotate(geom = "line", y= ybar*1.15, x=c(1,3))+
        scale_y_continuous(limits=c(ymin,max(ymax,ybar*1.125))) 
      
      p.e.a = p.e.a + 
        annotate(geom = "text",x=1.5,y = ybar*1.1,label = label2[1],fontface="italic",size=5) + theme(axis.text.x = element_text(angle=30,hjust=1))+
      annotate(geom = "line", y= ybar*1.05, x=c(1,1.9)) +
      
        annotate(geom = "text",x=2.5,y = ybar*1.1,label = label2[3],fontface="italic",size=5) + theme(axis.text.x = element_text(angle=30,hjust=1))+
      annotate(geom = "line", y= ybar*1.05, x=c(2.1,3)) +
      
        annotate(geom = "text",x=2,y = ybar*1.2,label = label2[2],fontface="italic",size=5) + theme(axis.text.x = element_text(angle=30,hjust=1))+
      annotate(geom = "line", y= ybar*1.15, x=c(1,3))+
        scale_y_continuous(limits=c(ymin,max(ymax,ybar*1.125))) 
    }  
  
    print(p.b.a+p.e.a)
  
    p.b = ggplot(wide,aes(x = Treatment_arm,y = as.numeric(Base),color=Treatment_arm )) + 
      facet_wrap(~grp,ncol=3) +theme_minimal(15)+
      geom_quasirandom() +
      geom_boxplot(fill=NA,outlier.size = -1) +
      theme_prism(axis_text_angle = 45) + scale_color_manual(values=c("black","#E18727FF"))+theme(legend.position="none")+xlab("")+
        ylab(paste0("Baseline ",yl)) +scale_y_continuous(limits=c(ymin,ymax))
    
    p.e =ggplot(wide,aes(x = Treatment_arm,y = as.numeric(EOT),color=Treatment_arm )) + 
      facet_wrap(~grp,ncol=3) +theme_minimal(15)+
      geom_quasirandom() +
      geom_boxplot(fill=NA,outlier.size = -1) +
      theme_prism(axis_text_angle = 45) + scale_color_manual(values=c("black","#E18727FF"))+theme(legend.position="none")+xlab("")+
        ylab(paste0("End ",yl)) +scale_y_continuous(limits=c(ymin,ymax))
    
    print(p.b + p.e)
    
    p.c.b = ggplot(wide,aes(x = Base.Vitc,y = as.numeric(Base),color=Treatment_arm,fill = Treatment_arm )) + 
      facet_wrap(~grp,ncol=3) +theme_minimal()+
      geom_point(alpha=0.5) +
      stat_smooth(method="lm") + 
      stat_cor(method = "pearson",p.digits = 2, r.accuracy = 0.01,
               aes(label =..rr.label..)) +
      theme_prism() + scale_color_manual(values=c("black","#E18727FF"))+theme(legend.position="none")+xlab("")+scale_fill_manual(values=c("black","#E18727FF"))+
        ylab(paste0("Baseline ",yl)) +xlab("Baseline Vitamin C")
    
    p.c.e = ggplot(wide,aes(x = EOT.Vitc,y = as.numeric(EOT),color=Treatment_arm,fill=Treatment_arm )) + 
      facet_wrap(~grp,ncol=3) +theme_minimal()+
      geom_point(alpha=0.5) +
      stat_smooth(method="lm") + 
      stat_cor(method = "pearson",p.digits = 2, r.accuracy = 0.01,
               aes(label =..rr.label..)) +
      theme_prism() + scale_color_manual(values=c("black","#E18727FF"))+theme(legend.position="none")+xlab("")+scale_fill_manual(values=c("black","#E18727FF"))+
        ylab(paste0("End ",yl)) +xlab("End Vitamin C")
    
    print(p.c.b)
    print(p.c.e)
    
    p = ggplot(wide,aes(x = Treatment_arm,y =as.numeric(EOT) - as.numeric(Base),color=Treatment_arm )) + 
      facet_wrap(~grp,ncol=3) +theme_minimal()+
      geom_quasirandom() +
      geom_boxplot(fill=NA,outlier.size = -1) +
      theme_prism(axis_text_angle = 45) + scale_color_manual(values=c("black","#E18727FF"))+theme(legend.position="none")+xlab("")+
      ylab(paste0("Delta ",yl))
    
    
    res.p = res
    res.p$Treatment_arm = "Placebo"
    
    res.c = res
    res.c$Treatment_arm = "Vitamin C"
    res.sig = rbind(res.p,res.c)  
    value = max(as.numeric(wide$EOT) - as.numeric(wide$Base),na.rm=T)
    res.sig$x = as.numeric(as.factor(res.sig$Treatment_arm))
    
    # res$pv = paste0("p = ",signif(res$p.value,2))
    res$CI = paste0("Mean \U0394 ",-1*signif(round(res$estimate,4),2),"\n95% CI ",
                    -1*signif(round(res$asymp.UCL,4),2),", ",
                    -1*signif(round(res$asymp.LCL,4),2))
    res$x = 1.5
    p = p + geom_line(data=res.sig,aes(x = x, y = value*1.1),color="black") +
      geom_text(data=res,aes(x = x, y= value*1.175 ,label = CI),fontface="italic",color="black")
    
    print(p +plot_spacer() + plot_layout(widths=c(1,0.75)))
  }    
}

## for manuscript figures
delta.only = function(outcome,subgroup){
  EOT = subset(df.merge[df.merge$Timepoint == "EOT",], select=c(outcome,"Treatment_arm","ID","Sex","Vitamin_C"))
  Base = subset(df.merge[df.merge$Timepoint == "Baseline",], select=c(outcome,"Treatment_arm","ID","Sex","Vitamin_C"))
  colnames(EOT)[1] = "EOT"
  colnames(Base)[1] = "Base"
  
  colnames(EOT)[5] = "EOT.Vitc"
  colnames(Base)[5] = "Base.Vitc"
  
  wide = left_join(EOT,Base)
  
  if(subgroup == "TET2"){
    wide$grp = ifelse( wide$ID %in% mut_ids,"TET2 or IDH1/2 mutation","No TET2 or IDH1/2 mutation")
    cols = c("black","#6F99ADFF")
    
  } 
  
  if(subgroup == "lof.TET2"){
      wide = wide[wide$ID %in% lof.tet2$Patient_ID,]
      wide$grp = ifelse( wide$ID %in% lof.tet2$Patient_ID[lof.tet2$TET2_Meth_Signature == "Low" | is.na(lof.tet2$TET2_Meth_Signature) ] ,"No or Low", ifelse(wide$ID %in% lof.tet2$Patient_ID[lof.tet2$TET2_Meth_Signature == "Intermediate"],"Intermediate","High")) 
      wide=wide[wide$grp!="Intermediate",]
      wide$grp = factor(wide$grp,levels=c("No or Low","High"),
                        labels = c("TET2 WT signature","TET2 hypermethylation signature"))
    cols = c("black","#6F99ADFF")
  } 
  

  if(subgroup == "vitc_med"){
    wide$grp = ifelse(wide$ID %in% df$ID[df$med_vitc == "above"],"Above median","Below median")  
    wide$grp = factor(wide$grp,levels=c("Below median","Above median"))
    cols = c("#FFDC91FF","#E18727FF")
  }
  
  if(subgroup == "vitc_severe"){
    wide$grp = ifelse(wide$ID %in% svr ,"Severe (< 11.4 μmol/L)","All others")  
    wide$grp = factor(wide$grp,levels=c("Severe (< 11.4 μmol/L)","All others"))
    cols = c("grey","#E18727FF")
  }
  
  if(subgroup == "vitc_ccus"){
        wide$grp = ifelse(wide$ID %in% df$ID[df$Diagnosis_simple_combo == "CCUS"],"CCUS",
                                         ifelse(wide$ID %in% df$ID[df$Diagnosis_simple_combo == "MDS"],"MDS",
                                         ifelse(wide$ID %in% df$ID[df$Diagnosis_simple_combo == "CMML_MDS_MPN"],"MDS/MPN",NA)))
    wide=wide[!is.na(wide$grp),]    
    wide$grp = factor(wide$grp,
                      levels= c("CCUS","MDS","MDS/MPN" ))
    
        cols = values=c("CCUS" =  viridis::viridis(4)[1] ,
                        "MDS" = viridis::viridis(4)[2],
                        "MDS/MPN" = viridis::viridis(4)[3])
    
  }
  

  fit = rlm(as.numeric( EOT) - as.numeric(Base) ~ Treatment_arm*grp ,  data=wide)
  
  # print(summary(fit))
  res = data.frame(summary(confint(emmeans(fit,pairwise~Treatment_arm|grp)$contrasts)))
  yl = case_when(outcome == "mC.dG" ~ "5-mC/dG",
                 outcome == "hmC.dG" ~ "5-hmC/dG",
                 outcome == "hmC.mC"~"5-hmC/5-mC")
  
  ymin = signif(0.975*min(as.numeric(c(wide$Base,wide$EOT)),na.rm=T),2)
  ymax = signif(1.025*max(as.numeric(c(wide$Base,wide$EOT)),na.rm=T),2)
  
  ybar = signif(1.1*quantile(as.numeric(c(wide$Base,wide$EOT)),na.rm=T,p=0.975),2)
  ybar = signif(quantile(as.numeric(c(wide$Base,wide$EOT)),na.rm=T,p=0.975),2) + 0.1*(ymin-ymax)

  fit.a = rlm(rank(as.numeric(Base)) ~ grp+Treatment_arm ,  data=wide[wide$grp != "Intermediate",])
  p.1 = summary(emmeans(fit.a,pairwise~grp)$contrasts)
  # label1 = ifelse(p.1$p.value<0.0001, "p < 0.0001", paste0("p = ",signif(p.1$p.value,2)))
   ci.1 = confint(emmeans(fit.a,pairwise~grp)$contrasts)
    label1 = paste0("Mean \U0394 ",-1*signif(round(ci.1$estimate,4),2),"\n95% CI ",
                    -1*signif(round(ci.1$asymp.UCL,4),2),", ",
                    -1*signif(round(ci.1$asymp.LCL,4),2))
    
  fit.b = rlm(rank(as.numeric(EOT)) ~ grp+Treatment_arm ,  data=wide[wide$grp != "Intermediate",])
  p.2 = summary(emmeans(fit.b,pairwise~grp)$contrasts)
  # label2 = ifelse(p.2$p.value<0.0001, "p < 0.0001", paste0("p = ",signif(p.2$p.value,2)))
  ci.2 = confint(emmeans(fit.b,pairwise~grp)$contrasts)
  label2 = paste0("Mean \U0394 ",-1*signif(round(ci.2$estimate,4),2),"\n95% CI ",
                    -1*signif(round(ci.2$asymp.UCL,4),2),", ",
                    -1*signif(round(ci.2$asymp.LCL,4),2))
    
  p = ggplot(wide,aes(x = Treatment_arm,y =as.numeric(EOT) - as.numeric(Base),color=Treatment_arm )) + 
    facet_wrap(~grp,ncol=3) +theme_minimal()+
    geom_quasirandom() +
    geom_boxplot(fill=NA,outlier.size = -1) +
    theme_prism(axis_text_angle = 45) + scale_color_manual(values=c("black","#E18727FF"))+theme(legend.position="none")+xlab("")+
    ylab(paste0("Delta ",yl))
  
  
  res.p = res
  res.p$Treatment_arm = "Placebo"
  
  res.c = res
  res.c$Treatment_arm = "Vitamin C"
  res.sig = rbind(res.p,res.c)  
  value = max(as.numeric(wide$EOT) - as.numeric(wide$Base),na.rm=T)
  res.sig$x = as.numeric(as.factor(res.sig$Treatment_arm))
  # res$pv = paste0("p = ",signif(res$p.value,2))
  res$CI = paste0("Mean \U0394 ",-1*signif(round(res$estimate,4),2),"\n95% CI ",
                    -1*signif(round(res$asymp.UCL,4),2),", ",
                    -1*signif(round(res$asymp.LCL,4),2))
  res$x = 1.5
  p = p + geom_line(data=res.sig,aes(x = x, y = value*1.1),color="black") +
    geom_text(data=res,aes(x = x, y= value*1.3 ,label = CI),fontface="italic",color="black")+geom_hline(yintercept=0,linetype="dotted",linewidth=0.75)
  
  return(p )
      
}

#This is for the FDR, to recreate these p-values the function can be rerun and the p-values printed.
saved_pvals = data.frame(outcome = rep(c("mC.dG","hmC.mC"),each=5),
                         subgroup = rep(c("TET2","vitc_med",rep("vitc_ccus",3) ),2),
                         p = c(0.1991,
                               0.1066,
                               0.0078, 0.6186, 0.0797,
                               0.00000008861223,
                               0.5790,
                               0.7838, 0.1405, 0.0837
                               ))

saved_pvals$FDR = p.adjust(saved_pvals$p,method="BH")


base.only = function(outcome,subgroup){
  EOT = subset(df.merge[df.merge$Timepoint == "EOT",], select=c(outcome,"Treatment_arm","ID","Sex","Vitamin_C"))
  Base = subset(df.merge[df.merge$Timepoint == "Baseline",], select=c(outcome,"Treatment_arm","ID","Sex","Vitamin_C"))
  colnames(EOT)[1] = "EOT"
  colnames(Base)[1] = "Base"
  
  colnames(EOT)[5] = "EOT.Vitc"
  colnames(Base)[5] = "Base.Vitc"
  
  wide = left_join(EOT,Base)
  
  if(subgroup == "TET2"){
    wide$grp = ifelse( wide$ID %in% mut_ids,"TET2 or IDH1/2 mutation","No TET2 or IDH1/2 mutation")
    cols = c("black","#6F99ADFF")
    
  } 

  if(subgroup == "vitc_med"){
    wide$grp = ifelse(wide$ID %in% df$ID[df$med_vitc == "above"],"Above median","Below median")  
    wide$grp = factor(wide$grp,levels=c("Below median","Above median"))
    cols = c("#FFDC91FF","#E18727FF")
  }
  

  if(subgroup == "vitc_ccus"){
        wide$grp = ifelse(wide$ID %in% df$ID[df$Diagnosis_simple_combo == "CCUS"],"CCUS",
                                         ifelse(wide$ID %in% df$ID[df$Diagnosis_simple_combo == "MDS"],"MDS",
                                         ifelse(wide$ID %in% df$ID[df$Diagnosis_simple_combo == "CMML_MDS_MPN"],"MDS/MPN",NA)))
    wide=wide[!is.na(wide$grp),]    
    wide$grp = factor(wide$grp,
                      levels= c("CCUS","MDS","MDS/MPN" ))
    
        cols = values=c("CCUS" =  viridis::viridis(4)[1] ,
                        "MDS" = viridis::viridis(4)[2],
                        "MDS/MPN" = viridis::viridis(4)[3])
    
  }
  
  

  fit = rlm(as.numeric( EOT) - as.numeric(Base) ~ Treatment_arm*grp ,  data=wide)
  res = data.frame(summary(emmeans(fit,pairwise~Treatment_arm|grp)$contrasts))
  yl = case_when(outcome == "mC.dG" ~ "5-mC/dG",
                 outcome == "hmC.dG" ~ "5-hmC/dG",
                 outcome == "hmC.mC"~"5-hmC/5-mC")
  
  ymin = signif(0.975*min(as.numeric(c(wide$Base,wide$EOT)),na.rm=T),2)
  ymax = signif(1.025*max(as.numeric(c(wide$Base,wide$EOT)),na.rm=T),2)
  
  ybar = max(as.numeric(wide$Base),na.rm=T)
  
  fit.a = rlm(as.numeric(Base) ~ grp+Treatment_arm ,  data=wide[wide$grp != "Intermediate",])
  joint_tests(fit.a)
  #p-vals are getting FDR adjusted across all subgroups
  p.1 = summary(emmeans(fit.a,pairwise~grp)$contrasts,adjust="none")
  print(p.1)
  
  label1 = ifelse(saved_pvals[saved_pvals$outcome == outcome & saved_pvals$subgroup==subgroup,]$FDR <0.0001, "FDR p < 0.0001", paste0("FDR p = ",signif(saved_pvals[saved_pvals$outcome == outcome & saved_pvals$subgroup==subgroup,]$FDR,2)))
  

  p.b.a = ggplot(wide[wide$grp != "Intermediate",],aes(x = grp,y = as.numeric(Base),color=grp )) + 
    geom_quasirandom() +
    geom_boxplot(fill=NA,outlier.size = -1) +
    theme_prism(axis_text_angle = 45) + scale_color_manual(values=cols)+theme(legend.position="none")+xlab("")+
      ylab(paste0("Baseline ",yl)) 
  
  if(subgroup != "vitc_ccus"){
   p.b.a = p.b.a + annotate(geom = "text",x=1.5,y = ybar*1.15,label = label1,fontface="italic",size=5) + theme(axis.text.x = element_text(angle=40,hjust=1))+
    annotate(geom = "line", y= ybar*1.05, x=c(1,2)) +
     scale_y_continuous(limits=c(ymin,ymax*1.3))

  } else{
  p.b.a = p.b.a + 
      annotate(geom = "text",x=1.5,y = ybar*1.14,label = label1[1],fontface="italic",size=5) + theme(axis.text.x = element_text(angle=40,hjust=1))+
    annotate(geom = "line", y= ybar*1.075, x=c(1,1.9)) +
    
      annotate(geom = "text",x=2.5,y = ybar*1.14,label = label1[3],fontface="italic",size=5) + theme(axis.text.x = element_text(angle=40,hjust=1))+
    annotate(geom = "line", y= ybar*1.075, x=c(2.1,3)) +
    
      annotate(geom = "text",x=2,y = ybar*1.33,label = label1[2],fontface="italic",size=5) + theme(axis.text.x = element_text(angle=40,hjust=1))+
    annotate(geom = "line", y= ybar*1.25, x=c(1,3))
  }  

  return(p.b.a )
      
}



```

### Delta 5-mC and 5-hmC/5-mC {.tabset}
#### TET2

```{r,fig.width=12,fig.height=5}

cowplot::plot_grid(delta.only("mC.dG","TET2"), delta.only("hmC.mC","TET2"),ncol=2,
                   labels = c("a","b"), label_size = 15)


```


#### baseline vitamin C

```{r,fig.width=12,fig.height=5}

cowplot::plot_grid(delta.only("mC.dG","vitc_med"), delta.only("hmC.mC","vitc_med"),ncol=2,
                   labels = c("a","b"))


```

#### Diagnosis

```{r,fig.width=14,fig.height=5.5}

cowplot::plot_grid(delta.only("mC.dG","vitc_ccus"), delta.only("hmC.mC","vitc_ccus"),ncol=2,
                   labels = c("a","b"))


```


### Baseline 5-mC and 5-hmC/5-mC {.tabset}

The raw p-values should be printing in each section

```{r}

knitr::kable(saved_pvals,caption="Saved p-values and FDR adjustments")


```

#### TET2

```{r,fig.width=10,fig.height=5.75}

cowplot::plot_grid(base.only("mC.dG","TET2"), base.only("hmC.mC","TET2"),ncol=2,
                   labels = c("a","b"))


```


#### baseline vitamin C

```{r,fig.width=10,fig.height=5.75}

cowplot::plot_grid(base.only("mC.dG","vitc_med"), base.only("hmC.mC","vitc_med"),ncol=2,
                   labels = c("a","b"))


```

#### Diagnosis

```{r,fig.width=17,fig.height=6}

cowplot::plot_grid(base.only("mC.dG","vitc_ccus"), base.only("hmC.mC","vitc_ccus"),ncol=2,
                   labels = c("a","b"))

```


### 5-mC/dG {.tabset}

#### All corr

```{r, fig.width=7,fig.height=7}

subgrp.glbl("mC.dG","All")

```

#### TET2 

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("mC.dG","TET2")

```

#### LOF TET2 

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("mC.dG","lof.TET2")

```

#### Median Vitamin C

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("mC.dG","vitc_med")

```


#### Vitamin C Severe

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("mC.dG","vitc_severe")

```

#### Diagnosis (CCUS)

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("mC.dG","vitc_ccus")

```




### 5-hmC/dG {.tabset}

#### All corr

```{r, fig.width=7,fig.height=7}

subgrp.glbl("hmC.dG","All")

```

#### TET2 

```{r, fig.width=12.3, fig.height=7}
subgrp.glbl("hmC.dG","TET2")

```

#### LOF TET2 

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("hmC.dG","lof.TET2")

```

#### Median Vitamin C

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("hmC.dG","vitc_med")

```


#### Vitamin C Severe

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("hmC.dG","vitc_severe")

```


#### Diagnosis (CCUS)

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("hmC.dG","vitc_ccus")

```


### 5-hmC/5-mC {.tabset}

#### All corr

```{r, fig.width=7,fig.height=7}

subgrp.glbl("hmC.mC","All")

```

#### TET2 

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("hmC.mC","TET2")

```

#### LOF TET2 

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("hmC.mC","lof.TET2")

```

#### Median Vitamin C

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("hmC.mC","vitc_med")

```


#### Vitamin C Severe

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("hmC.mC","vitc_severe")

```


#### Diagnosis (CCUS)

```{r, fig.width=12.3, fig.height=7}

subgrp.glbl("hmC.mC","vitc_ccus")

```


### Manuscript Figure

```{r, fig.width=11, fig.height=5}

df.merge$TET2 = ifelse(df.merge$ID %in% mut_ids,1,0)
library(ggfortify)

# Need a function for all the subgroup analyses
EOT = subset(df.merge[df.merge$Timepoint == "EOT",], select=c(mC.dG,Treatment_arm,ID))
Base = subset(df.merge[df.merge$Timepoint == "Baseline",], select=c(mC.dG,Treatment_arm,ID))
colnames(EOT)[1] = "EOT.mC.dG"
colnames(Base)[1] = "Base.mC.dG"

mc.dg.wide = left_join(EOT,Base)

fit = rlmer(as.numeric(mC.dG)  ~ Treatment_arm*Timepoint + Batch+ (1|ID),data=df.merge)

confint.rlmerMod <- function(object,parm,level=0.95) {
     beta <- fixef(object)
     if (missing(parm)) parm <- names(beta)
     se <- sqrt(diag(vcov(object)))
     z <- qnorm((1+level)/2)
     ctab <- cbind(beta-z*se,beta+z*se)
     colnames(ctab) <- stats:::format_perc(c((1-level)/2,(1+level)/2),
                                           digits=3)
     return(ctab[parm,])
}


knitr::kable(cbind(summary(fit)$coeff,confint(fit)),caption = "rlmer estimates 5mC/dG")

knitr::kable(summary(emmeans(fit,pairwise~Timepoint|Treatment_arm)),caption = "Time comparisons by treatment 5mC/dG")

knitr::kable(joint_tests(fit),caption = "anova on 5mC/dG model")
df.merge$mC.dG = as.numeric(df.merge$mC.dG)
df.merge$hmC.mC = as.numeric(df.merge$hmC.mC)

table1(~mC.dG+hmC.mC|Timepoint+Treatment_arm,data=df.merge)

p =  round(joint_tests(fit)$p.value[4],2)
df.merge$Timepoint = factor(df.merge$Timepoint,levels=c("Baseline","EOT"))
p1.tp =  ggplot(df.merge,aes(x = Time,y=as.numeric(mC.dG),color=Treatment_arm)) + geom_point(alpha=0.25) +
  geom_line(aes(group=ID),alpha=0.25) + facet_wrap(~Treatment_arm) +
  theme_prism(axis_text_angle = 45) + scale_color_manual(values=c("black","#E18727FF"))+theme(legend.position="none")+xlab("")+ylab('5-mC/dG')


p1.mc = ggplot(df.merge,aes(x = Timepoint,y=as.numeric(mC.dG),color=Treatment_arm)) + geom_point(alpha=0.25) +
  geom_line(aes(group=ID),alpha=0.25) + facet_wrap(~Treatment_arm) +
  theme_prism(axis_text_angle = 45) + scale_color_manual(values=c("black","#E18727FF"))+theme(legend.position="none")+xlab("")+ylab('5-mC/dG')


p2.mc = ggplot(mc.dg.wide,aes(x = Treatment_arm, y = as.numeric(EOT.mC.dG) - as.numeric(Base.mC.dG ),color=Treatment_arm)) +
  geom_quasirandom() + geom_boxplot(fill=NA,outliers=F) +
  stat_smooth(method="rlm") +
  xlab("") +
  ylab("Delta 5-mC/dG\n(Post - Pre)")+
  theme_prism(axis_text_angle = 45)+theme(legend.title = element_blank())+stat_cor()+
  scale_color_manual(values=c("black","#E18727FF")) +
  annotate(geom = "line",x = c(1,2), y= c(0.018,0.018) ,color="black") +
  annotate(geom="text",x = 1.5,y = 0.02,label=paste0("p =  ",p),fontface="italic")+theme(legend.position = "none")+geom_hline(yintercept=0,linetype="dotted")

pmc = (p2.mc)


fit = rlmer(as.numeric(hmC.mC)  ~ Treatment_arm*Timepoint + Batch + (1|ID),data=df.merge )

knitr::kable(cbind(summary(fit)$coeff,confint(fit)),caption = "rlmer estimates 5hmC/5mC")
knitr::kable(summary(emmeans(fit,pairwise~Timepoint|Treatment_arm)),caption = "Time comparisons by treatment 5hmC/5mC")

knitr::kable(joint_tests(fit),caption = "anova on 5hmC/5mC model")
table1(~mC.dG+hmC.mC|Timepoint+Treatment_arm,data=df.merge)

p =  round(joint_tests(fit)$p.value[4],2)

p1.tp.hmC.mC = ggplot(df.merge,aes(x = Time,y=as.numeric(hmC.mC),color=Treatment_arm)) + geom_point(alpha=0.25) +
  geom_line(aes(group=ID),alpha=0.25) + facet_wrap(~Treatment_arm) +
  theme_prism(axis_text_angle = 45) + scale_color_manual(values=c("black","#E18727FF"))+theme(legend.position="none")+xlab("")+ylab('5-mC/dG')


p1.hmC.mC = ggplot(df.merge,aes(x = Timepoint,y=as.numeric(hmC.mC),color=Treatment_arm)) + geom_point(alpha=0.25) +
  geom_line(aes(group=ID),alpha=0.25) + facet_wrap(~Treatment_arm) +
  theme_prism(axis_text_angle = 45) + scale_color_manual(values=c("black","#E18727FF"))+theme(legend.position="none")+xlab("")+ylab('5-hmC/5-mC')


EOT = subset(df.merge[df.merge$Timepoint == "EOT",], select=c(hmC.mC,Treatment_arm,ID))
Base = subset(df.merge[df.merge$Timepoint == "Baseline",], select=c(hmC.mC,Treatment_arm,ID))
colnames(EOT)[1] = "EOT.hmC.mC"
colnames(Base)[1] = "Base.hmC.mC"

hmC.mC.wide = left_join(EOT,Base)

p2.hmC.mC = ggplot(hmC.mC.wide,aes(x = Treatment_arm, y = as.numeric(EOT.hmC.mC) - as.numeric(Base.hmC.mC ),color=Treatment_arm)) +
  geom_quasirandom() + geom_boxplot(fill=NA,outliers=F) +
  stat_smooth(method="rlm") +
  xlab("") +
  ylab("Delta 5-hmC/5-mC\n(Post - Pre)")+
  theme_prism(axis_text_angle = 45)+theme(legend.title = element_blank())+stat_cor()+
  scale_color_manual(values=c("black","#E18727FF")) +
  annotate(geom = "line",x = c(1,2), y= c(0.0075,0.0075) ,color="black") +
  annotate(geom="text",x = 1.5,y = 0.0075*1.1,label=paste0("p =  ",p),fontface="italic")+theme(legend.position = "none")+geom_hline(yintercept=0,linetype="dotted")

fit = rlmer(as.numeric(hmC.mC)  ~ Treatment_arm*Timepoint + Batch +  (1|ID),data=df.merge )
phmC.mC = (p2.hmC.mC)


cowplot::plot_grid(pmc ,phmC.mC,
                     labels=c("a","b"),ncol=2,label_size = 15)


```


## Cytokines {.tabset}
### Primary Analysis, all patients {.tabset}

#### Suppl all cytokines

**Supplemental Figure XX. log2 fold-change in cytokine measures for all individuals.** Data were analyzed using robust linear mixed-effects models with a random intercept for each patient and an interaction between treatment and timepoint to assess if measures changed differently over time. 


```{r, fig.height=11,fig.width=11}

cyto = read.table(unzip("/varidata/research/projects/jones/computerome/Vitc_Data.zip","20240417_EVI_MSD.tsv"),sep="\t",header=T)

cyto$ID = gsub("-.*","",cyto$Source_Tube_ID)
cyto$ID[which(grepl("USC",fixed=T,cyto$Source_Tube_ID))] = substr(cyto$Source_Tube_ID[which(grepl("USC",fixed=T,cyto$Source_Tube_ID))],1,6)
cyto$ID[which(grepl("AAL",fixed=T,cyto$Source_Tube_ID))] = substr(cyto$Source_Tube_ID[which(grepl("AAL",fixed=T,cyto$Source_Tube_ID))],1,6)

cyto$ID = sapply(strsplit(cyto$ID,"-"), function(x){ifelse(length(x) >1, paste0(x[[1]],as.numeric(x[[2]])),x)})


cyto$Time = as.numeric(substr(gsub(" .*","",gsub(".*-","",cyto$Source_Tube_ID)),1,1))
cyto$Timepoint = ifelse(cyto$Time ==1,"Baseline","EOT")
cyto = cyto[order(cyto$Time,cyto$ID),]

cyto.m = merge(cyto,subset(df,select=c(ID, Treatment_arm,Age_inclusion_years, Sex,CCUS,Hb1)))

agg = aggregate(calc_conc_imputed~Timepoint +ID+assay+Treatment_arm,cyto.m,function(x) {mean(x,na.rm=T)})
agg.sub = subset(agg,select=c(ID,calc_conc_imputed,assay,Treatment_arm,Timepoint))

cyto.wide = reshape2::dcast(agg.sub, ID + Timepoint +Treatment_arm  ~ assay,value.var="calc_conc_imputed")
grps=NULL
diff=NULL
time=NULL
for(i in unique(cyto.m$assay)){
  fit = rlmer(log2(calc_conc_imputed) ~ Timepoint*Treatment_arm  +(1|ID),data=cyto.m[cyto.m$assay==i,])
  
  grps = rbind(grps,data.frame(assay = i,summary(emmeans(fit,pairwise~Treatment_arm|Timepoint)$contrasts)))
  time = rbind(time,data.frame(assay = i,summary(emmeans(fit,pairwise~Timepoint|Treatment_arm)$contrasts)))

  diff = rbind(diff,data.frame(assay = i,p =  emmeans::joint_tests(fit)[3,6],est = -1*summary(fit)$coeff[4,1],se = summary(fit)$coeff[4,2]))

}

grps$FDR = p.adjust(grps$p.value,method="BH")
time$FDR = p.adjust(time$p.value,method="BH")
time$FoldChange = 1/(2^time$estimate)
knitr::kable(time,caption = "Significant changes over time by treatment")

diff$FDR = p.adjust(diff$p,method="BH")

knitr::kable(diff[diff$FDR < 0.05,],caption= "Cytokines that changed differently over time for Placebo versus Vitamin C")

sig.t = time[time$FDR<0.05,]
sig.t$Label = case_when(sig.t$FDR<0.0001 ~"***",
                        sig.t$FDR<0.01 ~"**",
                        TRUE~"*")

sig.t$y = rep(0)
for(i in 1:nrow(sig.t)){
  sig.t$y[i] = log2(max(agg$calc_conc_imputed[agg$assay ==sig.t$assay[i]],na.rm=T))
}

agg$Timepoint = factor(agg$Timepoint,levels=c("Baseline","EOT"))
agg.pr = agg[agg$Timepoint=="Baseline",]
agg.post = subset(agg[agg$Timepoint=="EOT",],select=c(calc_conc_imputed, assay,ID))
agg.w = left_join(agg.pr,agg.post,by=c("ID","assay"))
agg.w$delta = log2(agg.w$calc_conc_imputed.y) - log2(agg.w$calc_conc_imputed.x)

sig.d=diff
options(scipen=999)


sig.d$Label = ifelse(sig.d$FDR<0.0001 ,"FDR p < 0.0001",paste0("FDR p =  ",signif(round(sig.d$FDR,4),2)))

sig.d$y = rep(0)
for(i in 1:nrow(sig.d)){
  sig.d$y[i] = max(agg.w$delta[agg.w$assay ==sig.d$assay[i]],na.rm=T)*1.275
}

sig.d$y = 5

agg.w$assay = gsub("_"," ",agg.w$assay)
time$assay = gsub("_"," ",time$assay)
```

```{r, fig.height=21,fig.width=14}

lvls=c("IL-1α", "IL-1β", "IL-6", "TNF-α", "IL-12p70", "IL-17",
 "IL-4", "IL-10", "TGF-β1",
 "GM-CSF", "G-CSF", "M-CSF",
 "IL-8 HA", "IP-10", "RANTES", "SDF-1α",
 "IL-7", "IFN-γ", "VEGF cyto")

lbls = c("IL-1α", "IL-1β", "IL-6", "TNF-α", "IL-12p70", "IL-17A",
 "IL-4", "IL-10", "TGF-β1",
 "GM-CSF", "G-CSF", "M-CSF",
 "IL-8", "CXCL10 (IP-10)", "RANTES (CCL5)", "SDF-1α",
 "IL-7", "IFN-γ", "VEGF")

sig.d$assay = factor(sig.d$assay,levels=lvls,labels=lbls)
time$assay = factor(time$assay,levels=lvls,labels=lbls)
agg.w$assay = factor(agg.w$assay,levels=lvls,labels=lbls)

grps = list(pro.inflm = c("IL-1α", "IL-1β", "IL-6", "TNF-α", "IL-12p70", "IL-17A"),
ant.inflm = c("IL-4", "IL-10", "TGF-β1"),
m.grow = c("GM-CSF", "G-CSF", "M-CSF"),
chemok = c("IL-8", "CXCL10 (IP-10)", "RANTES (CCL5)", "SDF-1α"),
reg.cyt = c("IL-7", "IFN-γ", "VEGF"))

grp.nms = c("Pro-inflammatory cytokines",
          "Anti-inflammatory cytokines",
          "Myeloid growth factors",
          "Chemokines",
          "Immune regulatory cytokines")
p=list()
for(i in 1:length(grps)){
  p[[i]] = ggplot(agg.w[agg.w$assay %in% grps[[i]],],aes(x = Treatment_arm, y = delta,color= Treatment_arm)) +
    facet_wrap2(vars(assay),scales="free",ncol=6, trim_blank = FALSE)+geom_hline(yintercept = 0,linewidth=1,color="grey",linetype="dashed") +
    theme_prism() + geom_quasirandom()+ geom_boxplot(fill=NA,outlier.size=-1)+ stat_smooth(method="lm",linewidth=1.25)+scale_color_manual(values=c("black","#E18727FF"))+theme(strip.background = element_blank())+ylab(expression(~ log2(frac("Cytokine concentration post","Cytokine concentration pre")))) +geom_text(data=sig.d[sig.d$assay %in% grps[[i]],],aes(x=1.5,y=y,label=Label),color="black",size=4,fontface="italic")+geom_segment(data=sig.d[sig.d$assay %in% grps[[i]],],aes(x = 1,xend=2,y =0.8*y,yend=0.8*y ),color="black")+theme(legend.title = element_blank(),legend.position = "none")+xlab("")+
    ggtitle(grp.nms[i])
}

cowplot::plot_grid(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],ncol=1)


```


```{r, fig.height=9,fig.width=12}

ggplot(cyto.m,aes(x = Timepoint, y = log2(calc_conc_imputed), color=Treatment_arm)) + facet_wrap(~assay,scales="free_y") +
  geom_point(size=1.5)+
  geom_line(aes(group=ID),alpha=0.3,size=1) +
  theme_prism()+scale_color_manual(values=c("black","#E18727FF"))+theme(strip.background = element_blank())

```

#### Cytokine figure

**Supplemental Figure XX. log2 fold-change in cytokine measures that changed significantly.** Data were analyzed using robust linear mixed-effects models with a random intercept for each patient and an interaction between treatment and timepoint to assess if measures changed differently over time. Larger points indicate the mean log2(fold-change) with 95% CI error bars. Scales were limited to +/-1.5 log2(fold-change) to better characterize dispersion, all data points are shown in supplemental figure XX. P-values were false discovery rate adjusted using Benjamini-Hochberg. 


```{r,fig.height=4.5,fig.width=12}
sig.d = sig.d[sig.d$FDR<0.05,]

diff$assay = factor(gsub("_"," ",diff$assay),levels=lvls,labels=lbls)

time.h = time
agg.w.h = agg.w

time = time[time$assay %in% diff$assay[diff$FDR<0.05 ],]
agg.w = agg.w[agg.w$assay %in% diff$assay[diff$FDR<0.05 ],]
sig.d$Treatment_arm = "Placebo"

agg.w$abs_delta =  agg.w$calc_conc_imputed.y - agg.w$calc_conc_imputed.x

time$estimate = time$estimate*-1

ggplot(time,aes(x = Treatment_arm, y = estimate,fill=Treatment_arm,color=Treatment_arm)) + geom_point(size=2.5)+ facet_wrap(~assay,scales="free_x",ncol=6)+geom_hline(yintercept = 0,linewidth=1,color="grey",linetype="dashed",linewidth=1.25) +
  theme_prism() + scale_fill_manual(values=c("black","#E18727FF"))+
  scale_color_manual(values=c("black","#E18727FF")) +theme(strip.background = element_blank(),plot.caption =  element_text(hjust = 0, margin = margin(0, 0, 0, 0, "pt")))+ylab("log2(FC)")+geom_text(data=sig.d,aes(x=1.5,y=1.5,label=Label,group=1),fill="black",size=4,fontface="italic")+geom_segment(data=sig.d,aes(x = 1,xend=2,y =0.925*1.5,yend=0.925*1.5 ,group=1),color="black")+theme(legend.title = element_blank(),legend.position = "none")+xlab("") + geom_errorbar(aes(ymin = estimate - 1.96*SE,ymax= estimate + 1.96*SE),width=0.25,linewidth=0.75) +geom_quasirandom(data=agg.w,aes(x = Treatment_arm, y = delta,color=Treatment_arm),alpha=0.25) + coord_cartesian(ylim=c(-1.5,1.5))

ggsave("~/bbc-secondary/research/JONP_20240430_vitc_trial_VBCS/cytokine_figure_a.pdf",width=18,height=6)


```



```{r}

agg.df = aggregate(abs_delta ~ assay+Treatment_arm,agg.w,median)
colnames(agg.df)[3] = "Median change"
agg.df$IQR  = aggregate(abs_delta ~ assay+Treatment_arm,agg.w,IQR)$abs_delta
time$FoldChange = exp(time$estimate)
time$FC.CI = paste0(round(exp(time$estimate - 1.96*time$SE),2),", ",round(exp(time$estimate + 1.96*time$SE),2))

cyto.res = merge(subset(time,  select=c(assay,Treatment_arm,FoldChange,FC.CI)),agg.df,by=c("assay","Treatment_arm"))
cyto.res = cyto.res[order(cyto.res$assay,cyto.res$Treatment_arm),]
cyto.res$assay = as.character(cyto.res$assay)
cyto.res$assay[duplicated( cyto.res$assay)] = ""
knitr::kable(cyto.res)


```

```{r}
cyto.mm = merge(cyto.m,df,by=c("ID"))
cyto.sub=NULL

cyto.mm$Timepoint = factor(cyto.mm$Timepoint,levels=c("Baseline","EOT"))
cyto.time = NULL
for(i in unique(cyto.m$assay)){
  fit = rlmer(log2(calc_conc_imputed) ~ Timepoint*Treatment_arm.x +(1|ID),data=cyto.mm[cyto.mm$assay==i & cyto.mm$TET2 == "TET2neg",])
  cyto.sub = rbind(cyto.sub,data.frame(Subgroup =  "TET2 neg",assay = i,p =  emmeans::joint_tests(fit)[3,6],est=-1*summary(fit)$coeff[4,1],se = summary(fit)$coeff[4,2]))

  cyto.time = rbind(cyto.time,data.frame(assay = i,Subgroup =  "TET2 neg",summary(emmeans(fit,pairwise~Timepoint|Treatment_arm.x)$contrasts)))


}

for(i in unique(cyto.m$assay)){
  fit = rlmer(log2(calc_conc_imputed) ~ Timepoint*Treatment_arm.x +(1|ID),data=cyto.mm[cyto.mm$assay==i & cyto.mm$TET2 == "TET2Mut",])
  cyto.sub = rbind(cyto.sub,data.frame(Subgroup =  "TET2Mut",assay = i,p =  emmeans::joint_tests(fit)[3,6],est=-1*summary(fit)$coeff[4,1],se = summary(fit)$coeff[4,2]))

  cyto.time = rbind(cyto.time,data.frame(assay = i,Subgroup =  "TET2Mut",summary(emmeans(fit,pairwise~Timepoint|Treatment_arm.x)$contrasts)))

}

for(i in unique(cyto.m$assay)){
  fit = rlmer(log2(calc_conc_imputed) ~ Timepoint*Treatment_arm.x +(1|ID),data=cyto.mm[cyto.mm$assay==i & cyto.mm$med_vitc == "above",])
  cyto.sub = rbind(cyto.sub,data.frame(Subgroup =  "Above Med Vit C",assay = i,p =  emmeans::joint_tests(fit)[3,6],est=-1*summary(fit)$coeff[4,1],se = summary(fit)$coeff[4,2]))
  cyto.time = rbind(cyto.time,data.frame(assay = i,Subgroup =  "Above Med Vit C",summary(emmeans(fit,pairwise~Timepoint|Treatment_arm.x)$contrasts)))

}

for(i in unique(cyto.m$assay)){
  fit = rlmer(log2(calc_conc_imputed) ~ Timepoint*Treatment_arm.x +(1|ID),data=cyto.mm[cyto.mm$assay==i & cyto.mm$med_vitc == "below",])
  cyto.sub = rbind(cyto.sub,data.frame(Subgroup =  "below Med Vit C",assay = i,p =  emmeans::joint_tests(fit)[3,6],est=-1*summary(fit)$coeff[4,1],se = summary(fit)$coeff[4,2]))
  cyto.time = rbind(cyto.time,data.frame(assay = i,Subgroup =  "below Med Vit C",summary(emmeans(fit,pairwise~Timepoint|Treatment_arm.x)$contrasts)))

}

for(i in unique(cyto.m$assay)){
  fit = rlmer(log2(calc_conc_imputed) ~ Timepoint*Treatment_arm.x +(1|ID),data=cyto.mm[cyto.mm$assay==i & cyto.mm$Diagnosis_simple_combo == "CCUS",])
  cyto.sub = rbind(cyto.sub,data.frame(Subgroup =  "CCUS",assay = i,p =  emmeans::joint_tests(fit)[3,6],est=-1*summary(fit)$coeff[4,1],se = summary(fit)$coeff[4,2]))
  cyto.time = rbind(cyto.time,data.frame(assay = i,Subgroup =  "CCUS",summary(emmeans(fit,pairwise~Timepoint|Treatment_arm.x)$contrasts)))
}

for(i in unique(cyto.m$assay)){
  fit = rlmer(log2(calc_conc_imputed) ~ Timepoint*Treatment_arm.x +(1|ID),data=cyto.mm[cyto.mm$assay==i & cyto.mm$Diagnosis_simple_combo == "MDS",])
  cyto.sub = rbind(cyto.sub,data.frame(Subgroup =  "MDS",assay = i,p =  emmeans::joint_tests(fit)[3,6],est=-1*summary(fit)$coeff[4,1],se = summary(fit)$coeff[4,2]))
  cyto.time = rbind(cyto.time,data.frame(assay = i,Subgroup =  "MDS",summary(emmeans(fit,pairwise~Timepoint|Treatment_arm.x)$contrasts)))
}

for(i in unique(cyto.m$assay)){
  fit = rlmer(log2(calc_conc_imputed) ~ Timepoint*Treatment_arm.x +(1|ID),data=cyto.mm[cyto.mm$assay==i & cyto.mm$Diagnosis_simple_combo == "CMML_MDS_MPN",])
  cyto.sub = rbind(cyto.sub,data.frame(Subgroup =  "CMML_MDS_MPN",assay = i,p =  emmeans::joint_tests(fit)[3,6],est=-1*summary(fit)$coeff[4,1],se = summary(fit)$coeff[4,2]))
  cyto.time = rbind(cyto.time,data.frame(assay = i,Subgroup =  "CMML_MDS_MPN",summary(emmeans(fit,pairwise~Timepoint|Treatment_arm.x)$contrasts)))
}

#We aren't reporting these, but we'll calculate them in case any future readers want them
cyto.sub$FDR = p.adjust(cyto.sub$p,method="BH")

```

#### Volcano plots of each subroup by assay

```{r, fig.height=11,fig.width=11}

cyto.sub$assay[cyto.sub$assay == "IL-8_HA"] = "IL-8 HA"
cyto.sub$assay[cyto.sub$assay == "VEGF_cyto"] = "VEGF cyto"
cyto.sub$assay = factor(cyto.sub$assay,levels=lvls,labels=lbls)

cyto.time$assay[cyto.time$assay == "IL-8_HA"] = "IL-8 HA"
cyto.time$assay[cyto.time$assay == "VEGF_cyto"] = "VEGF cyto"
cyto.time$assay = factor(cyto.time$assay,levels=lvls,labels=lbls)

cyto.sub = cyto.sub[!cyto.sub$Subgroup %in% c("Deficient","Inadequate","Adequate"),]

diff$Subgroup =  "All"

cyto.sub = rbind(cyto.sub,diff)
           
cyto.sub$Significant = case_when(cyto.sub$FDR < 0.05 & cyto.sub$est<0~"Significantly higher in vitamin C patients",
                          cyto.sub$FDR < 0.05 & cyto.sub$est>0~"Significantly lower in vitamin C patients",
                          TRUE ~ "Not Significant")

cyto.sub$Subgroup =  factor(cyto.sub$Subgroup,levels=c("All",
                                                      "TET2 neg",
                                                      "TET2Mut",
                                                      "below Med Vit C",
                                                      "Above Med Vit C",
                                                      "CCUS",
                                                      "MDS",
                                                      "CMML_MDS_MPN"),
                                             labels=c("All patients",
                                                    "No TET2 or IDH1/2 mutations",
                                                    "TET2 or IDH1/2 mutation",
                                                    "Below median vitamin C",
                                                    "Above median vitamin C",
                                                    "CCUS",
                                                    "MDS",
                                                    "MDS/MPN")
)

ggplot(cyto.sub, aes(x = est, y = -log10(FDR),color=Subgroup )) + 
        geom_point() + 
        theme_prism(axis_text_angle=45) +
        scale_color_viridis_d(option="turbo")+    
        xlab("log2(FC)") +
  facet_wrap(~assay,scales = "free") +
  scale_x_continuous(limits=c(-1,1)) + theme(legend.position = "bottom")+ geom_hline(aes(yintercept = 1.30,linetype = "FDR < 0.05"),color="red",alpha=0.25) + 
  scale_linetype_manual(values=c(2)) +
  scale_y_continuous(limits=c(0,5))

```

#### Subgroup plots {.tabset}

All FDR p-values less than 0.1 are shown for comparing across subgroups.

```{r}

cyto.time$Subgroup =  factor(cyto.time$Subgroup,levels=rev(c("All",
                                                      "TET2 neg",
                                                      "TET2Mut",
                                                      "below Med Vit C",
                                                      "Above Med Vit C",
                                                      "CCUS",
                                                      "MDS",
                                                      "CMML_MDS_MPN")),
                                             labels=rev(c("All patients",
                                                    "No TET2 or IDH1/2 mutations",
                                                    "TET2 or IDH1/2 mutation",
                                                    "Below median vitamin C",
                                                    "Above median vitamin C",
                                                    "CCUS",
                                                    "MDS",
                                                    "MDS/MPN"))
)

plot.cyto.sub = function(subgroup){
  
  assays = unique(cyto.sub$assay[cyto.sub$Subgroup %in% subgroup & cyto.sub$FDR<0.05])
  time = cyto.time[cyto.time$assay %in% assays & cyto.time$Subgroup %in% subgroup,]
  agg.w = agg.w.h[agg.w.h$assay %in% assays,]
  agg.w$assay = factor(agg.w$assay,levels = unique(agg.w$assay))
  sig.d = cyto.sub[cyto.sub$Subgroup %in% subgroup & cyto.sub$assay %in% assays,]
  sig.d$Treatment_arm = rep("Placebo")
  sig.d$Label = ifelse(sig.d$FDR<0.0001 ,"FDR p < 0.0001",paste0("FDR p =  ",signif(round(sig.d$FDR,4),2)))
  sig.d = sig.d[sig.d$FDR < 0.1,]
  agg.w$abs_delta =  agg.w$calc_conc_imputed.y - agg.w$calc_conc_imputed.x
  
  time$estimate = time$est*-1
  
  colnames(time)[which(colnames(time) == "Treatment_arm.x")] = "Treatment_arm"
  
  if("No TET2 or IDH1/2 mutations" %in% subgroup ){
    agg.w = left_join(agg.w,subset(df,select=c(ID,TET2)),by="ID")
  
    agg.w$Subgroup = factor(agg.w$TET2,levels=c("TET2neg","TET2Mut"),
                                               labels=c("No TET2 or IDH1/2 mutations",
                                                      "TET2 or IDH1/2 mutation"))
  }
  
  if("CCUS" %in% subgroup ){
    agg.w = left_join(agg.w,subset(df,select=c(ID,Diagnosis_simple_combo)),by="ID")
    agg.w = agg.w[!is.na(agg.w$Diagnosis_simple_combo),]
    agg.w$Subgroup = factor(agg.w$Diagnosis_simple_combo,levels=c("CCUS",
                                                      "MDS",
                                                      "CMML_MDS_MPN"),
                                               labels=c("CCUS",
                                                    "MDS",
                                                    "MDS/MPN"))
  }
  
  if("Below median vitamin C" %in% subgroup ){
    agg.w = left_join(agg.w,subset(df,select=c(ID,med_vitc)),by="ID")
    agg.w$Subgroup = factor(agg.w$med_vitc,levels=c("below", "above"),
                                               labels=c("Below median vitamin C",
                                                    "Above median vitamin C"))
  }
  
  print(table(agg.w$Treatment_arm,agg.w$assay,agg.w$Subgroup))
  
 
  p = ggplot(time,aes(x = Treatment_arm, y = estimate,fill=Treatment_arm,color=Treatment_arm)) + geom_point(size=2.5)+
    facet_nested_wrap(vars(Subgroup,assay),scales="free_x",ncol=length(assays),trim_blank = FALSE)+
      geom_hline(yintercept = 0,linewidth=1,color="grey",linetype="dashed",linewidth=1.25) +
    theme_prism() + scale_fill_manual(values=c("black","#E18727FF"))+
    scale_color_manual(values=c("black","#E18727FF")) +theme(strip.background = element_blank(),plot.caption =  element_text(hjust = 0, margin = margin(0, 0, 0, 0, "pt")))+ylab("log2(FC)") +
    # geom_text(data=sig.d,aes(x=1.5,y=1.5,label=Label,group=1),fill="black",size=4,fontface="italic")+geom_segment(data=sig.d,aes(x = 1,xend=2,y =0.925*1.5,yend=0.925*1.5 ,group=1),color="black")+
    theme(legend.title = element_blank(),legend.position = "none")+
    xlab("") +
    geom_errorbar(aes(ymin = estimate - 1.96*SE,ymax= estimate + 1.96*SE),width=0.25,linewidth=0.75) +
    geom_quasirandom(data=agg.w,aes(x = Treatment_arm, y = delta,color=Treatment_arm),alpha=0.25) + coord_cartesian(ylim=c(-1.5,1.5))

  return(p)
  
  
}


plot.cyto.sub = function(subgroup){
  
  assays = unique(cyto.sub$assay[cyto.sub$Subgroup %in% subgroup & cyto.sub$FDR<0.05])
  time = cyto.time[cyto.time$assay %in% assays & cyto.time$Subgroup %in% subgroup,]
  agg.w = agg.w.h[agg.w.h$assay %in% assays,]
  agg.w$assay = factor(agg.w$assay,levels = unique(agg.w$assay))
  sig.d = cyto.sub[cyto.sub$Subgroup %in% subgroup & cyto.sub$assay %in% assays,]
  sig.d$Treatment_arm = rep("Placebo")
  sig.d$Label = ifelse(sig.d$FDR<0.0001 ,"FDR p < 0.0001",paste0("FDR p =  ",signif(round(sig.d$FDR,4),2)))
  sig.d = sig.d[sig.d$FDR < 0.1,]
  agg.w$abs_delta =  agg.w$calc_conc_imputed.y - agg.w$calc_conc_imputed.x
  
  time$estimate = time$est*-1
  
  colnames(time)[which(colnames(time) == "Treatment_arm.x")] = "Treatment_arm"
  
  if("No TET2 or IDH1/2 mutations" %in% subgroup ){
    agg.w = left_join(agg.w,subset(df,select=c(ID,TET2)),by="ID")
  
    agg.w$Subgroup = factor(agg.w$TET2,levels=c("TET2neg","TET2Mut"),
                                               labels=c("No TET2 or IDH1/2 mutations",
                                                      "TET2 or IDH1/2 mutation"))
  }
  
  if("CCUS" %in% subgroup ){
    agg.w = left_join(agg.w,subset(df,select=c(ID,Diagnosis_simple_combo)),by="ID")
    agg.w = agg.w[!is.na(agg.w$Diagnosis_simple_combo),]
    agg.w$Subgroup = factor(agg.w$Diagnosis_simple_combo,levels=c("CCUS",
                                                      "MDS",
                                                      "CMML_MDS_MPN"),
                                               labels=c("CCUS",
                                                    "MDS",
                                                    "MDS/MPN"))
  }
  
  if("Below median vitamin C" %in% subgroup ){
    agg.w = left_join(agg.w,subset(df,select=c(ID,med_vitc)),by="ID")
    agg.w$Subgroup = factor(agg.w$med_vitc,levels=c("below", "above"),
                                               labels=c("Below median vitamin C",
                                                    "Above median vitamin C"))
  }
  
  print(knitr::kable(table(agg.w$Treatment_arm,agg.w$assay,agg.w$Subgroup)))
  
 
  p = ggplot(time,aes(x = Treatment_arm, y = estimate,fill=Treatment_arm,color=Treatment_arm)) + geom_point(size=2.5)+
    facet_nested_wrap(vars(Subgroup,assay),scales="free_x",ncol=length(assays),trim_blank = FALSE)+
      geom_hline(yintercept = 0,linewidth=1,color="grey",linetype="dashed",linewidth=1.25) +
    theme_prism() + scale_fill_manual(values=c("black","#E18727FF"))+
    scale_color_manual(values=c("black","#E18727FF")) +theme(strip.background = element_blank(),plot.caption =  element_text(hjust = 0, margin = margin(0, 0, 0, 0, "pt")))+ylab("log2(FC)") +
    # geom_text(data=sig.d,aes(x=1.5,y=1.5,label=Label,group=1),fill="black",size=4,fontface="italic")+geom_segment(data=sig.d,aes(x = 1,xend=2,y =0.925*1.5,yend=0.925*1.5 ,group=1),color="black")+
    theme(legend.title = element_blank(),legend.position = "none")+
    xlab("") +
    geom_errorbar(aes(ymin = estimate - 1.96*SE,ymax= estimate + 1.96*SE),width=0.25,linewidth=0.75) +
    geom_quasirandom(data=agg.w,aes(x = Treatment_arm, y = delta,color=Treatment_arm),alpha=0.25) + coord_cartesian(ylim=c(-1.5,1.5))

  return(p)
  
  
}



```

##### All subgroups

```{r, fig.height=18,fig.width=18}

p=list()
for(i in 1:length(grps)){
  
p[[i]] = ggplot(cyto.sub[cyto.sub$assay  %in% grps[[i]],],aes(x = -1*est , y = Subgroup)) +
  geom_vline(xintercept=0,color="red",linetype="dotted",size=0.75) +
  geom_point(size=2) + facet_wrap2(vars(assay),scales="free_x",ncol=6, trim_blank = FALSE,remove_labels = "all")+
  theme_minimal(15) + geom_errorbar(aes(xmin = -1*(est-1.96*se),xmax=-1*(est+1.96*se)),width=0.15)+
  xlab("Delta-delta on log2 transformed cytokines")+ theme(legend.title = element_blank(),legend.position = "none")+ylab("")+
    ggtitle(grp.nms[i]) 
  
    
  
}

 
cowplot::plot_grid(p[[1]],p[[2]],p[[3]],p[[4]],p[[5]],ncol=1)

```



##### TET2

```{r, fig.width = 16,fig.height=8}

plot.cyto.sub(c("No TET2 or IDH1/2 mutations","TET2 or IDH1/2 mutation"))
```


##### Median Vitamin C

```{r, fig.width = 16,fig.height=8}

plot.cyto.sub(c("Below median vitamin C","Above median vitamin C"))

```

##### Diagnosis

```{r, fig.width = 16,fig.height=12}

plot.cyto.sub(c("CCUS","MDS","MDS/MPN"))

```



#### WGCNA

Cannot get good scale-free topology WGCNA is a no go. Surprisingly little correlation among the cyotkines


```{r}

library(corrplot)
library(WGCNA)
M = cor(cyto.wide[,4:ncol(cyto.wide)],use='pairwise.complete.obs')
corrplot(M, method = 'square', order = 'FPC', type = 'upper', diag = FALSE)


datExpr = cyto.wide[,4:ncol(cyto.wide)]
SubCytoNames=colnames(cyto.wide[,4:ncol(cyto.wide)])


powers = c(c(1:10), seq(from = 12, to=20, by=2));

sft=pickSoftThreshold(datExpr,dataIsExpr = TRUE,powerVector = powers,corFnc = cor,corOptions = list(use = 'p'),networkType = "signed hybrid")

# Plot the results
sizeGrWindow(9, 5)
par(mfrow = c(1,2));
cex1 = 0.9;

# Scale-free topology fit index as a function of the soft-thresholding power
plot(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit, signed hybrid hybrid R^2",type="n", main = paste("Scale independence"));
text(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],labels=powers,cex=cex1,col="red");

# Red line corresponds to using an R^2 cut-off
abline(h=0.80,col="red")

# Mean connectivity as a function of the soft-thresholding power
plot(sft$fitIndices[,1], sft$fitIndices[,5],xlab="Soft Threshold (power)",ylab="Mean Connectivity", type="n",main = paste("Mean connectivity"))
text(sft$fitIndices[,1], sft$fitIndices[,5], labels=powers, cex=cex1,col="red")

```


#### PCA

Majority of variance is neither time nor treatment

```{r, fig.height=5,fig.width=10}

cyto.wide.c = na.omit(cyto.wide)
pc.a = prcomp(log2(cyto.wide.c[,4:ncol(cyto.wide.c)]+1),scale=T)
pc.df = as.data.frame(pc.a$x)
pc.df$Treatment = cyto.wide.c$Treatment_arm
pc.df$Timepoint = cyto.wide.c$Timepoint
pc.df$ID = cyto.wide.c$ID

pc.1 = ggplot(pc.df,aes(x = PC1,y=PC2,color=Treatment,shape=Timepoint)) + geom_point() +
  theme_prism() + scale_color_manual(values =c("black",pal_nejm()(4)[3])) + stat_ellipse()

pc.2 = ggplot(pc.df,aes(x = PC3,y=PC4,color=Treatment,shape=Timepoint)) + geom_point() +
  theme_prism() + scale_color_manual(values =c("black",pal_nejm()(4)[3])) + stat_ellipse()

pc.3 = ggplot(pc.df,aes(x = PC5,y=PC6,color=Treatment,shape=Timepoint)) + geom_point() +
  theme_prism() + scale_color_manual(values =c("black",pal_nejm()(4)[3])) + stat_ellipse()

pc.1 + pc.2 + pc.3

```

